{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "3c2b2e94",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "id": "fcf4e506",
   "metadata": {
    "slideshow": {
     "slide_type": "slide"
    }
   },
   "source": [
    "# We'll start at 2:10ish PM \n",
    "\n",
    "Waiting for others to join the call :)\n",
    "\n",
    "https://www.youtube.com/watch?v=l6rS8Dv5g-8"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "ecf8788f",
   "metadata": {
    "slideshow": {
     "slide_type": "slide"
    }
   },
   "source": [
    "<center><img src='img/cdy.png' style='width:500px; float: left; margin: 0px 30px 15px 0px'></center>\n",
    "\n",
    "# Classification models\n",
    "## Class 22 - Data Science Curriculum \n",
    "\n",
    "<br>\n",
    "\n",
    "#### Women Building Change scholarship program 2023 üáßüáÆ\n",
    "May 15, 2023\n",
    "\n",
    "\n",
    "\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "52d145e2",
   "metadata": {
    "slideshow": {
     "slide_type": "slide"
    }
   },
   "source": [
    "# ‚è™ Recap last class\n",
    "\n",
    "- Performance metrics for classification models\n",
    "    - Accuracy\n",
    "    - Precision \n",
    "    - Recall\n",
    "    - F1-score \n",
    "    - Confusion matrix"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "59660346",
   "metadata": {
    "slideshow": {
     "slide_type": "slide"
    }
   },
   "source": [
    "# üöÄ Today's agenda\n",
    "\n",
    "- Classification models\n",
    "    - Logistic regression \n",
    "    - Naive Bayes\n",
    "    - Support Vector Machines (SVM)\n",
    "    - Decision Trees"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "b113ea0b",
   "metadata": {
    "slideshow": {
     "slide_type": "slide"
    }
   },
   "source": [
    "# üó∫Ô∏èüìç Guide map \n",
    "\n",
    "<center>\n",
    "    <img src='img/classification.png' style='height:550px; float: center; margin: 0px 0px 0px 0px'>\n",
    "</center>\n",
    "    \n",
    "\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "19105db5",
   "metadata": {
    "slideshow": {
     "slide_type": "slide"
    }
   },
   "source": [
    "# Classification models\n",
    "\n",
    "There are three types of classification models:\n",
    "\n",
    "- Binary\n",
    "- Multiclass\n",
    "- Multilabel\n",
    "\n",
    "<center><img src='img/classf2.png' style='height:400px; float: center; margin: 0px 0px 0px 0px'></center>"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "ef1c99ba",
   "metadata": {
    "slideshow": {
     "slide_type": "slide"
    }
   },
   "source": [
    "# Some performance metrics for classification models\n",
    "- **Accuracy**: $$\\dfrac{TP+TN}{TP+TN+FP+FN} = \\dfrac{\\text{correctly classified}}{\\text{number of observations}}$$\n",
    "<br>\n",
    "- **Recall**: $$\\dfrac{TP}{TP+FN} = \\dfrac{\\text{TP}}{\\text{what it's actually positive}}$$\n",
    "<br>\n",
    "- **Precision**: $$\\dfrac{TP}{TP+FP} = \\dfrac{\\text{TP}}{\\text{what got classified as positive}}$$\n",
    "<br>\n",
    "- **F1**: $$2\\cdot \\dfrac{\\text{precision}\\cdot\\text{recall}}{\\text{precision}+\\text{recall}}$$"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "a1f44957",
   "metadata": {
    "slideshow": {
     "slide_type": "slide"
    }
   },
   "source": [
    "# Recall vs Precision\n",
    "\n",
    "- **Precision** is the ratio of correctly predicted positive observations to the total predicted positives. It's more focused on minimizing false positives. Precision is used when the cost of False Positives is high. \n",
    "\n",
    "    - Pharmaceutical drug approval: In pharmaceutical research, precision is highly valued because a false positive (assuming a drug is effective when it is not) can lead to a significant waste of time and resources in later phases of drug testing and potentially harmful outcomes in patients.\n",
    "\n",
    "    - Marketing campaign: When you are running a targeted marketing campaign, precision would be a priority. You want to ensure that your resources are spent on leads that are most likely to convert to customers (true positives). A high number of false positives would mean wasted resources on leads that do not convert.\n",
    "\n",
    "- **Recall**, on the other hand, is the ratio of correctly predicted positive observations to all observations in actual class - yes. It's more focused on minimizing false negatives. Recall would be the metric to focus on when the cost of False Negatives is high. \n",
    "\n",
    "    - Fraud detection: If you're trying to catch fraudulent activity, such as credit card fraud or insurance fraud, it's typically more important to catch all potential fraud cases (high recall) even if that means you investigate some false alarms (lower precision).\n",
    "\n",
    "    - Emergency disaster warnings: For emergency alerts, such as tornado or hurricane warnings, a high recall is critical. It is better to issue some false alarms (lower precision) than to miss a disaster that could result in loss of life."
   ]
  },
  {
   "cell_type": "markdown",
   "id": "cf265861",
   "metadata": {
    "slideshow": {
     "slide_type": "slide"
    }
   },
   "source": [
    "<center><img src=\"img/conf_matrix3.png\" style='height:800px; float: center; margin: 0px 0px 0px 0px'></center>"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "1a14f7a9",
   "metadata": {
    "slideshow": {
     "slide_type": "slide"
    }
   },
   "source": [
    "# Break for 5 minutes!\n",
    "See you here at 2:46 PM :)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "214905f1",
   "metadata": {
    "slideshow": {
     "slide_type": "slide"
    }
   },
   "source": [
    "# ‚ùáÔ∏è Logistic regression"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "722794dd",
   "metadata": {
    "slideshow": {
     "slide_type": "slide"
    }
   },
   "source": [
    "<br>\n",
    "<center><img src='img/rm2.png' style='height:500px; float: center; margin: 0px 0px 0px 0px'></center>"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "786d1e80",
   "metadata": {
    "slideshow": {
     "slide_type": "slide"
    }
   },
   "source": [
    "# Logistic regression"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "4abd239c",
   "metadata": {
    "slideshow": {
     "slide_type": "fragment"
    }
   },
   "source": [
    "- It's the most famous Machine Learning model after Linear regression"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "433593cf",
   "metadata": {
    "slideshow": {
     "slide_type": "fragment"
    }
   },
   "source": [
    "- While you use linear regression to predict **continuous** values, logistic regression is <u>typically</u> used to classify **discrete** values"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "52f3dfe4",
   "metadata": {
    "slideshow": {
     "slide_type": "fragment"
    }
   },
   "source": [
    "- Predictions are mapped between 0 and 1, which means that they can be interpreted as probabilities"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "ff7cf4f9",
   "metadata": {
    "slideshow": {
     "slide_type": "slide"
    }
   },
   "source": [
    "# Logistic regression"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "bb30b4be",
   "metadata": {
    "slideshow": {
     "slide_type": "fragment"
    }
   },
   "source": [
    "### üëç Pros \n",
    "\n",
    "- The predictions have an elegant probabilistic representation \n",
    "- The model can be regularized to avoid overfitting"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "f4d68eba",
   "metadata": {
    "slideshow": {
     "slide_type": "fragment"
    }
   },
   "source": [
    "### üëé Cons \n",
    "\n",
    "- Tends to underperform when there are multiple decision lines or these are non-linear \n",
    "- Not flexible enough to capture more complex relationships"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "d953740f",
   "metadata": {
    "slideshow": {
     "slide_type": "slide"
    }
   },
   "source": [
    "# ‚ùáÔ∏è Support Vector Machines (SVM)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "f16682a0",
   "metadata": {
    "slideshow": {
     "slide_type": "slide"
    }
   },
   "source": [
    "# Support Vector Machines (SVM)\n",
    "\n",
    "<br>\n",
    "\n",
    "<center><img src='img/svm1.png' style='height:400px; float: center; margin: 0px 0px 0px 0px'></center>"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "7a796ef5",
   "metadata": {
    "slideshow": {
     "slide_type": "slide"
    }
   },
   "source": [
    "# SVM\n",
    "\n",
    "How would you separate this 1-dimensional data into two classes?\n",
    "<br>\n",
    "\n",
    "<center><img src='img/svm2.png' style='height:400px; float: center; margin: 0px 0px 0px 0px'></center>"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "0da05968",
   "metadata": {
    "slideshow": {
     "slide_type": "fragment"
    }
   },
   "source": [
    "A point"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "edcbe6d0",
   "metadata": {
    "slideshow": {
     "slide_type": "slide"
    }
   },
   "source": [
    "# SVM\n",
    "\n",
    "How would you separate this 2-dimensional data into two classes?\n",
    "<br>\n",
    "\n",
    "<center><img src='img/svm3.png' style='height:400px; float: center; margin: 0px 0px 0px 0px'></center>"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "0c4237e4",
   "metadata": {
    "slideshow": {
     "slide_type": "fragment"
    }
   },
   "source": [
    "A line"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "9f79e685",
   "metadata": {
    "slideshow": {
     "slide_type": "slide"
    }
   },
   "source": [
    "## How decision boundaries scale with dimensionality?"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "066b21a6",
   "metadata": {
    "slideshow": {
     "slide_type": "fragment"
    }
   },
   "source": [
    "- A point in 1 dimension"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "14764afc",
   "metadata": {
    "slideshow": {
     "slide_type": "fragment"
    }
   },
   "source": [
    "- A line (or curve) in 2 dimensions"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "c0f4c7a8",
   "metadata": {
    "slideshow": {
     "slide_type": "fragment"
    }
   },
   "source": [
    "- A plane in 3 dimensions"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "a90a508b",
   "metadata": {
    "slideshow": {
     "slide_type": "fragment"
    }
   },
   "source": [
    "- A _______ in higher dimensions"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "621abdb4",
   "metadata": {
    "slideshow": {
     "slide_type": "fragment"
    }
   },
   "source": [
    "    - Answer: Hyperplane (or manifold)\n",
    "   "
   ]
  },
  {
   "cell_type": "markdown",
   "id": "cd017b05",
   "metadata": {
    "slideshow": {
     "slide_type": "slide"
    }
   },
   "source": [
    "## Learn a decision bound\n",
    "\n",
    "<center><img src='img/svm8.png' style='height:400px; float: center; margin: 0px 0px 0px 0px'></center>"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "4b60f52b",
   "metadata": {
    "slideshow": {
     "slide_type": "fragment"
    }
   },
   "source": [
    "- Find a boundary that divides the classes from each other (training)\n",
    "- Use that boundary to predict class membership for new data (inference)\n",
    "   "
   ]
  },
  {
   "cell_type": "markdown",
   "id": "15172038",
   "metadata": {
    "slideshow": {
     "slide_type": "slide"
    }
   },
   "source": [
    "## Which line represents the best separation?\n",
    "<br>\n",
    "<center><img src='img/svm4.png' style='height:400px; float: center; margin: 0px 0px 0px 0px'></center>"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "0e0b6432",
   "metadata": {
    "slideshow": {
     "slide_type": "slide"
    }
   },
   "source": [
    "## Margin\n",
    "<br>\n",
    "<center><img src='img/svm4b.png' style='height:400px; float: center; margin: 0px 0px 0px 0px'></center>\n",
    "\n",
    "- Rather than drawing a zero-width lune between the classes, draw a margin around each one of some width"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "b94ea2a5",
   "metadata": {
    "slideshow": {
     "slide_type": "slide"
    }
   },
   "source": [
    "## Maximize the margin\n",
    "<br>\n",
    "<center><img src='img/svm5.png' style='height:400px; float: center; margin: 0px 0px 0px 0px'></center>\n",
    "\n",
    "- The margin is up to the nearest point of each class"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "fa404bca",
   "metadata": {
    "slideshow": {
     "slide_type": "slide"
    }
   },
   "source": [
    "## Maximize the margin\n",
    "\n",
    "<br>\n",
    "<center><img src='img/svm9.gif' style='height:400px; float: center; margin: 0px 0px 0px 0px'></center>\n",
    "<br>\n",
    "\n",
    "- SVMs use a mechanism called kernel that essentially calculates the distance between two observations. With this, the model finds an optimal decision boundary (hyperplane) to separate the classes\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "279b251d",
   "metadata": {
    "slideshow": {
     "slide_type": "slide"
    }
   },
   "source": [
    "## SVM\n",
    "\n",
    "<br>\n",
    "<center><img src='img/svm7.png' style='height:400px;'></center>\n",
    "<br>\n",
    "\n",
    "- An SVM with a linear kernel is very similar to logistic regression, so it is generally used with nonlinear kernels with nonlinear data\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "de83ed6b",
   "metadata": {
    "slideshow": {
     "slide_type": "slide"
    }
   },
   "source": [
    "# SVM"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "df3e748f",
   "metadata": {
    "slideshow": {
     "slide_type": "fragment"
    }
   },
   "source": [
    "### üëç Pros \n",
    "\n",
    "- It can model non-linear relationships\n",
    "- There are many kernels to choose from\n",
    "- Good at avoiding overfitting"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "288c6306",
   "metadata": {
    "slideshow": {
     "slide_type": "fragment"
    }
   },
   "source": [
    "### üëé  Cons \n",
    "\n",
    "- Intense with memory usage\n",
    "- Difficult to finetune, difficult to choose kernel\n",
    "- It doesn't do well with large data sets"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "16320e28",
   "metadata": {
    "slideshow": {
     "slide_type": "slide"
    }
   },
   "source": [
    "# ‚ùáÔ∏è Naive-Bayes"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "79352ee5",
   "metadata": {
    "slideshow": {
     "slide_type": "slide"
    }
   },
   "source": [
    "# Naive-Bayes for classification \n",
    "\n",
    "<br>\n",
    "\n",
    "<center><img src='img/nb.jpg' style='height:400px; float: center; margin: 0px 0px 0px 0px'></center>\n",
    "<br>\n",
    "\n",
    "- Model based on Bayes' theorem which explains conditional probability. The model is updated through your training data\n",
    "- It is called \"naive\" because it assumes conditional independence (all attributes are independent)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "6c500b28",
   "metadata": {
    "slideshow": {
     "slide_type": "slide"
    }
   },
   "source": [
    "# Naive-Bayes"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "bb9c8604",
   "metadata": {
    "slideshow": {
     "slide_type": "fragment"
    }
   },
   "source": [
    "### üëç Pros \n",
    "\n",
    "- Although the conditional independence assumption is rarely true, it has surprisingly good results in practice\n",
    "- The model is easy to implement"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "7c62c703",
   "metadata": {
    "slideshow": {
     "slide_type": "fragment"
    }
   },
   "source": [
    "### üëé  Cons \n",
    "\n",
    "- Due to its simplicity, although it has good results, other models generally have better results"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "df961c5b",
   "metadata": {
    "slideshow": {
     "slide_type": "slide"
    }
   },
   "source": [
    "# ‚ùáÔ∏è Decision trees"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "d25ce79a",
   "metadata": {
    "slideshow": {
     "slide_type": "slide"
    }
   },
   "source": [
    "# Decision trees\n",
    "\n",
    "<br>\n",
    "<center>\n",
    "    <img src='img/dt.gif' style='height:650px; float: center; margin: 0px 0px 0px 0px'>\n",
    "</center>"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "8aa25ba9",
   "metadata": {
    "slideshow": {
     "slide_type": "fragment"
    }
   },
   "source": [
    "- It is an algorithm that can be used for both classification and regression\n",
    "- The goal is to create a model that predicts the value of the target variable by learning simple decision rules"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "5b07ebd0",
   "metadata": {
    "slideshow": {
     "slide_type": "slide"
    }
   },
   "source": [
    "# Decision trees\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "ecc50a0c",
   "metadata": {
    "slideshow": {
     "slide_type": "fragment"
    }
   },
   "source": [
    "### üëç Pros \n",
    "\n",
    "- The predictions have an elegant probabilistic representation \n",
    "- The model can be regularized to avoid overfitting"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "c2ea44ba",
   "metadata": {
    "slideshow": {
     "slide_type": "fragment"
    }
   },
   "source": [
    "### üëé Cons\n",
    "\n",
    "- Prone to overfitting\n",
    "- Unstable, but that can be solved with an ensemble model"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "d3603898",
   "metadata": {
    "slideshow": {
     "slide_type": "slide"
    }
   },
   "source": [
    "# ‚ùáÔ∏è Coding time! \n",
    "\n",
    "<br>\n",
    "<center><img src='img/excited.gif' style='height:200px; float: center; margin: 0px 0px 0px 0px'></center>\n",
    "<br>"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "6e194a46",
   "metadata": {
    "slideshow": {
     "slide_type": "slide"
    }
   },
   "source": [
    "# üó∫Ô∏èüìç Guide map \n",
    "\n",
    "<br>\n",
    "<center>\n",
    "    <img src='img/pipeline.png' style='height:550px; float: center; margin: 0px 0px 0px 0px'>\n",
    "</center>\n",
    "    \n",
    "\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "9c8299e5",
   "metadata": {
    "slideshow": {
     "slide_type": "slide"
    }
   },
   "source": [
    "## Classification models\n",
    "### Steps for a classification model:\n",
    "\n",
    "1. Acquire labeled data\n",
    "2. Divide the data set into two:\n",
    "    - Training and Test\n",
    "3. Feature Engineering on Training dataset (Pre-processing, TF-IDF, etc...)\n",
    "4. Train the model \n",
    "5. Obtain performance measures on test dataset and improve model\n",
    "6. Put into production"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 63,
   "id": "e666a793",
   "metadata": {
    "scrolled": false,
    "slideshow": {
     "slide_type": "slide"
    }
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'Deoxyshikonin Inhibits Viability andGlycolysis by Suppressing theAktmTOR Pathway in Acute MyeloidLeukemia CellsHuijuan Wu \\xa0 Hongmian Zhao \\xa0 and Li Chen  Telemedicine and Connected Health Center Huaihe Hospital of Henan University Kaifeng China  Department ofHematology Huaihe Hospital of Henan University Kaifeng ChinaDeoxyshikonin was reported to exhibit an antitumor effect in colorectal cancer Howeverno studies are available to illustrate the effect of deoxyshikonin on acute myeloid leukemiaAML The effects of deoxyshikonin on viability apoptosis caspase37 activity andcytochrome Cyt C expression were evaluated by Cell Counting Kit8 assay √Ø¬¨\\x82owcytometry analysis caspase37 activity assay and western blot analysis respectivelyGlucose consumption and lactate production were measured to determine the glycolysislevel The expression of pyruvate kinase M2 PKM2 was detected by quantitativerealtime polymerase chain reaction and western blot analysis The results showed thatdeoxyshikonin inhibited cell viability and increased the apoptotic rate the caspase37activity and the Cyt C protein level in AML cells in a dosedependent manner Additionallydeoxyshikonin concentrationdependently decreased glucose consumptionlactateproduction and PKM2 expression in AML cells Deoxyshikonin inactivated the proteinkinase B Aktmammalian target of the rapamycin mTOR pathway The activation ofthe AktmTOR pathway reversed the effects of deoxyshikonin on viability apoptosisand glycolysis in AML cells In  deoxyshikonin dampened the viability and theglycolysis of AML cells by suppressing PKM2 via inactivation of the AktmTOR signalingEdited byCyrus KhandanpourUniversity Hospital M¬ºnster GermanyReviewed byPeng YangShanxi University ChinaHaili WuShanxi University ChinaHongyu ZhouKunming Medical University ChinaCorrespondenceHuijuan Wuhjwu297163com\\xa0These authors share √Ø¬¨\\x81rst authorshipKeywords deoxyshikonin glycolysis PKM2 the AktmTOR signaling acute myeloid leukemiaSpecialty sectionThis  was submitted toHematologic Malignanciesa section of the journalFrontiers in OncologyReceived  March Accepted  June Published  August CitationWu H Zhao H and Chen L Deoxyshikonin Inhibits Viability andGlycolysis by Suppressing theAktmTOR Pathway in Acute MyeloidLeukemia CellsFront Oncol  103389fonc202001253INTRODUCTIONAcute myeloid leukemia AML the most prevalent form of acute leukemia in adults is anaggressive malignancy derived from hemopoietic progenitor cells and with poor survival rate andfrequent relapse posing a threat to the health and life of a√Ø¬¨ected patients  AML is characterizedby rapid growth impaired apoptosis and abnormal clonal accumulation of hematopoietic stemcells in the bone marrow due to various genetic and epigenetic changes eventually leading to bonemarrow failure  There were an estimated  new cases diagnosed with AML and mortalities due to AML according to the statistics in   Presently the e√Ø¬¨\\x83cacy of the standardchemotherapy for AML patients remains suboptimal owing to drug resistance and high clinicalrelapse rate  In this regard searching for novel antileukemia drugs is urgently required toe√Ø¬¨ectively improve the outcome of patients with AMLShikonin 58dihydroxy2[1S1hydroxy4methylpent3en1yl]naphthalene14 dionea naturally occurring naphthoquinone extracted from the oriental traditional medical herbFrontiers in Oncology  wwwfrontiersinAugust   Volume    0cWu et alDeoxyshikonin Inhibits AML Cellsand suppressed glycolysis in AML cells Mechanistically the antiAML e√Ø¬¨ect of deoxyshikonin was mediated via repressing PKM2by inactivation of the AktmTOR pathwayMATERIALS AND METHODSCell Cultivation and TreatmentAML cell lines including THP1 and HL60 were purchasedfrom American Tissue Culture Collection ATCC ManassasVA USA and routinely cultured in Roswell Park MemorialInstitute1640 medium HyClone South Logan UT USAconjugated with  heatinactivated fetal bovine serum ExCellBio Shanghai China and  penicillinstreptomycin SigmaAldrich St Louis MO USA The cells were fostered at \\x97¬¶Cin a drippy environment √Ø¬¨\\x82ushed with  air plus  CO2Deoxyshikonin purity  Tauto Biotech Shanghai Chinawas dissolved in dimethyl sulfoxide SigmaAldrich St LouisMO USA as a reserve solution and diluted into di√Ø¬¨erentconcentrations      and  ¬µgml The THP1and HL60 cells were exposed to diverse doses of deoxyshikoninfor  h The Aktoverexpressing plasmid pcDNAAkt andthe empty vectorpcDNA control were obtained fromRibobio Guangzhou China Transfection was performed usingLipofectamine  reagent Invitrogen Carlsbad CA USACell Counting Kit8 AssayCell Counting Kit8 CCK8 Beyotime Shanghai China wastaken to evaluate the viability of AML cells THP1 and HL60cells were seeded into 96well plates at  \\x97  cellswell anddealt with various concentrations of deoxyshikonin     and  ¬µgml for  h or incubated with  ¬µgmldeoxyshikonin in the presence or the absence of  ¬µM 740YP Tocris Bioscience Shanghai China an activator of theAktmTOR signaling pathway After treatment for  h  ¬µl ofCCK8 solution was added to each well followed by incubationfor another  h The optical density of each well was recorded at awavelength of  nm using a microplate reader Thermo FisherScienti√Ø¬¨\\x81c Waltham MA USAFlow Cytometry Analysis for ApoptosisAfter the treatments as aforementioned annexin V FITCand propidium iodide kitKeyGen Nanjing China wasimplemented to analyze the apoptosis of THP1 and HL60 cellsCell apoptotic rate was detected by means of a √Ø¬¨\\x82ow cytometerFACScan BD Biosciences San Diego CA USAtheastreatmentsCaspase37 Activity AssayFollowingaforementioned ApoONEHomogeneous CaspaseGlo  Assay kit Promega MadisonWI USA was adopted to measure the caspase37 activityof THP1 and HL60 cells referring to the manufacturer\\x99sdescription Finally M2000 In√Ø¬¨\\x81nite Pro instrument TecanTrading AG Maennedorf Switzerland was used to determinethe luminescenceFIGURE   Chemical structure of deoxyshikoninLithospermum erythrorhizon Sieb et Zucc has been extensivelyused for the treatment of many diseases including burnssore throats HIV1 infection and macular eruption  Currently clinical and pharmacological propertiesstudieshave demonstrated that shikonin and its derivatives exhibitvarious biological activities such as immune regulation andantithrombotic anti\\x82ammatory antioxidative and antiglycolytic activities   An increasing number of researchesreveal that shikonin derivatives have garnered much researchinterest due to its limited toxicity and stronger antitumoractivities in miscellaneous cancers  Interestingly a previousinvestigation proved that deoxyshikonin its chemical structureis shown in Figure  a derivative of shikonin exhibited an antitumor e√Ø¬¨ect in colorectal cancer  However no studies areavailable to illustrate the e√Ø¬¨ect of deoxyshikonin on AMLAerobic glycolysis also known as the Warburg e√Ø¬¨ect is wellrecognized as a metabolic pathway in the rapidly proliferatingcancer cells for the regeneration of energy and the biosynthesisof macromolecules even in the presence of su√Ø¬¨\\x83cient oxygen Aerobic glycolysis has been extensively accepted as an importantcharacteristic of tumor cells including AML eventually resultingin increased glucose consumption and lactate production  Pyruvate kinase PK is a ratelimiting glycolytic enzymePyruvate kinase M1 PKM1 is expressed in normal di√Ø¬¨erentiatedtissues whereas pyruvate kinase M2 PKM2 is expressedin cancer cellsleading to increased glycolysis  As weall know the protein kinase B Aktmammalian target ofrapamycin mTOR pathway widely existing in cells is one ofthe most important survival signaling pathways that participatein the regulation of diverse physiological processes such as cellgrowth apoptosis and metabolism  It has been reportedthat the AktmTOR pathway is a positive regulator of PKM2expression   Previous studies suggested that shikonininhibited glycolysis by suppression of PKM2 expression   Therefore we hypothesized that deoxyshikonin inhibitedviability and glycolysis suppressing pyruvate kinase M2 via theAktmTOR pathway in acute myeloid leukemia cellsIn the present study we assessed the e√Ø¬¨ects and the underlyingmechanisms of deoxyshikonin on viability apoptosis glycolysisand PKM2 expression in AML cells These results revealed thatdeoxyshikonin treatment inhibited viability induced apoptosisFrontiers in Oncology  wwwfrontiersinAugust   Volume    0cWu et alDeoxyshikonin Inhibits AML CellsDetermination of Glucose Consumptionand Lactate ProductionAfter the treatments as aforementioned the culture medium ofTHP1 and HL60 cells was collected for the measurement ofglucose and lactate levels using a glucose uptake colorimetricassay kit SigmaAldrich and a lactic acid assay kit JianchengBioengineering Institute Nanjing China respectivelyPKM2 Activity AssayPKM2 activity was detected by the lactate dehydrogenasecoupled assay as described earlier  Five microliters of wholecelllysate was utilized in the assay The absorbance at awavelength of  nm was measured using a microplate readerThermo Fisher Scienti√Ø¬¨\\x81cicecold radioimmunoprecipitation assayWestern Blot AnalysisThe treated THP1 and HL60 cells were collected and rinsedwith phosphatebu√Ø¬¨ered salinefollowed by the additionoflysis bu√Ø¬¨erBeyotime containing  mM phenylmethylsulfonyl √Ø¬¨\\x82uorideSigmaAldrich A total of  ¬µg of protein samples wasfractionated on  sodium dodecyl sulfatepolyacrylamide gelelectrophoresis prior to electrotransfer onto polyvinylidenedi√Ø¬¨\\x82uoride membranes Millipore Bedford MA USA Afterblocking with  defatted milkTrisbased saline with Tween atroom temperature for  h the membranes were immunoblottedovernight at \\x97¬¶C with corresponding primary antibodiesagainst cytochrome C Cyt C Cell Signaling TechnologyInc Danvers MA USA phosphorylated Akt pAkt Ser473Abcam Cambridge MA USA AktAbcam glycogensynthase kinase3√é¬≤ GSK3√é¬≤ Abcam phosphorylated GSK3√é¬≤pGSK3√é¬≤ Ser9 Abcam mTOR Abcam phosphorylatedmTOR pmTOR Ser2448 Abcam p70 ribosomal S6 kinasep70S6K Cell Signaling TechnologyInc phosphorylatedp70S6K pp70S6K Thr389 Cell Signaling Technology Inceukaryotic translation initiation factor 4Ebinding protein 4EBP1 Cell Signaling Technology Inc phosphorylated4EBP1 p4EBP1 Thr70 Cell Signaling TechnologyIncPKM2 Abcam and √é¬≤actin Abcam and then incubated withhorseradish peroxidaseconjugated secondary antibodies CellSignaling Technology Inc for  h at room temperature Lastlyan enhanced chemiluminescence kit Amersham PharmaciaPiscataway NJ USA was implemented to examine the antigen\\x93antibody complexes √é¬≤Actin was used as a loading control Theprotein bands were visualized by VersaDoc imaging systemBioRad Hercules CA USAQuantitative RealTime PCRRNAiso Plus TaKaRa Dalian China was used to extracttotal RNA from treated THP1 and HL60 cells and theconcentration of extracted RNA was measured using a NanoDrop spectrophotometer Thermo Fisher Scienti√Ø¬¨\\x81c Reversetranscription was carried out using the PrimeScript RT ReagentKit Takara Dalian China SYBR Green Taq Mix TaKaRawas then implemented to detect PKM2 mRNA expressionon the StepOnePlus qPCR system Thermo Fisher Scienti√Ø¬¨\\x81cwith √é¬≤actin as an endogenous control The thermocyclingconditions were displayed as follows \\x97¬¶C for  s followedby  cycles of \\x97¬¶C for  s \\x97¬¶C for  s and \\x97¬¶C for  sThe primers were as follows PKM2 ¬≤GCTG CCAT CTACCACT TGC3¬≤ forward and ¬≤CCAG ACTT GGTG AGGACGAT T3¬≤ reverse GAPDH ¬≤ATGT CGTG GAGT CTACTGGC3¬≤ forward and ¬≤TGAC CTTG CCCA CAGC CTTG¬≤ reverse The \\x88\\x92\\\\x01\\\\x01Ct method was taken to quantify theexpression level of PKM2 mRNAStatisticsAll data are shown as mean ¬± standard deviation of threeindependent experiments Statistical assays were determinedusing SPSS  software IBM Corp Armonk NY USA withStudent\\x99s ttest or oneway ANOVA followed by Dunnett\\x99s test P  were regarded as statistically signi√Ø¬¨\\x81cantRESULTSDeoxyshikonin Inhibited the Viability ofAML CellsTo clarify the antitumor activity of deoxyshikonin in AMLcells CCK8 was taken to evaluate cell viability after THP and HL60 cells were exposed to a series of deoxyshikoninconcentrations      and  ¬µgml for  h Asshown in Figures 2AB cell viability was signi√Ø¬¨\\x81cantly declinedin a concentrationdependent manner in THP1 and HL60 cellsin response to deoxyshikoninDeoxyshikonin Enhanced the Apoptosis ofAML CellsThe e√Ø¬¨ect of deoxyshikonin on the apoptotic consequences ofAML cells was investigated by annexin VFITC apoptosis assayAs a result deoxyshikonin treatment led to a concentrationdependent increase of apoptotic rate in THP1 Figure 3Aand HL60 cells Figure 3B In line with the results of the√Ø¬¨\\x82ow cytometry analysis an elevation of caspase37 activityin deoxyshikonintreated THP1 Figure 3C and HL60 cellsFigure 3D was observed The mitochondrial protein Cyt Cis known as the initiating factor of mitochondrial apoptosispathway The expression of apoptotic marker Cyt C in THP1and HL60 cells was further detected by western blot analysisThe results demonstrated that deoxyshikonin concentrationdependently increased Cyt C protein level in THP1 Figure 3Eand HL60 cells Figure 3F relative to the control group Theseresults suggested that deoxyshikonin facilitated the apoptosis ofAML cellsDeoxyshikonin Suppressed Glycolysis inAML CellsTo further characterize the e√Ø¬¨ect of deoxyshikonin on glycolysisin AML cells we measured glucose consumption and lactateproduction in AML cells These results demonstrated thatdeoxyshikonin exposure decreased glucoseconsumptionFigures 4AB and lactate production Figures 4CD in THP1Frontiers in Oncology  wwwfrontiersinAugust   Volume    0cWu et alDeoxyshikonin Inhibits AML CellsFIGURE   Deoxyshikonin dosedependently inhibited the viability of acute myeloid leukemia cells THP1 A and HL60 B cells were administrated with variousdoses of deoxyshikonin      and  ¬µgml for  h and cell viability was then assessed by CCK8 assay P   compared to the control groupand HL60 cells in a dosedependent manner We concluded thatdeoxyshikonin suppressed glycolysis in AML cellsenzymeratelimitingan importantDeoxyshikonin Decreased the ExpressionLevel of PKM2 in AML CellsA previous study reported that shikonin suppressed tumoraerobic glycolysis through suppressing the activity of PKM2in regulatingcellular glycolysis  Accordingly we supposed whetherdeoxyshikonin had an inhibitory e√Ø¬¨ect on PKM2 expression Asexpected the quantitative realtime polymerase chain reactionqRTPCR results showed that the mRNA levels of PKM2were suppressed following the addition of deoxyshikonin in adosedependent manner in THP1 Figure 5A and HL60 cellsFigure 5B The western blot results showed that deoxyshikonintreatment inhibited the protein levels of PKM2 in a dosedependent manner in THP1 Figure 5C and HL60 cellsFigure 5D We also found that deoxyshikonin suppressedPKM2 activity in THP1 Figure 5E and HL60 cells Figure 5FThese results suggested that deoxyshikonin decreased theexpression level of PKM2 in AML cellsDeoxyshikonin Inactivated the AktmTORPathway in AML CellsThe aberrant AktmTOR signaling has been demonstrated tobe associated with the tumorigenesis of miscellaneous cancersincluding AML  We further determined the \\x82uence ofdeoxyshikonin on the AktmTOR pathway in AML cells Asdemonstrated by western blot analysis deoxyshikonin treatmentrestricted the phosphorylation of Akt GSK3√é¬≤ mTOR p70S6Kand 4EBP1 in a concentrationdependent manner but causedno noticeable change on the total protein levels of Akt GSK3√é¬≤ mTOR p70S6K and 4EBP1 in THP1 and HL60 cellsFigures 6A\\x93C indicating that deoxyshikonin inactivated theAktmTOR pathway in AML cellsActivation of the AktmTOR PathwayReversed the Effects of Deoxyshikonin onViability and Apoptosis of AML CellsTo √Ø¬¨\\x81gure out whether the AktmTOR pathway was involved inmediating the antitumor e√Ø¬¨ects of deoxyshikonin on AML cellsTHP1 and HL60 cells were treated with  ¬µgml deoxyshikoninin the presence or the absence of 740YP for  h The 740YP treatment alone resulted in a notable enhancement of pAkt and pmTOR expressions but produced little alternationon Akt and mTOR protein levels in THP1 Figure 7A andHL60 cells Figure 7B suggesting the activation of AktmTORsignaling by 740YP The CCK8 assay presented thatthedeoxyshikonin treatmentinduced viability reduction in THP Figure 7C and HL60 cells Figure 7D was e√Ø¬¨ectivelyameliorated following the addition of 740YP The increaseof apoptotic rate in deoxyshikonintreated THP1 Figure 7Eand HL60 cells Figure 7F was signi√Ø¬¨\\x81cantly abolished aftercotreatment with deoxyshikonin and 740YP Moreover caspase activity was enhanced in THP1 Figure 7G and HL60 cellsFigure 7H in response to deoxyshikonin which was attenuatedfollowing the addition of 740YP Furthermore the protein levelof Cyt C in the deoxyshikonin  740YP cotreatment groupin THP1 Figure 7I and HL60 cells Figure 7J was reducedwhen compared with that of the deoxyshikonin treatment groupCollectively these results suggested that the activation of theAktmTOR pathway reversed the e√Ø¬¨ects of deoxyshikonin on theviability and the apoptosis of AML cellsActivation of the AktmTOR PathwayReversed the Effects of Deoxyshikonin onGlycolysis and PKM2 Expression in AMLCellsThe 740YP treatment overturned the reduction of glucoseconsumption Figure 8A and lactate production Figure 8Bmediated by deoxyshikonin in THP1 and HL60 cells ThedeoxyshikoninsuppressedPKM2 mRNA expression in THP1 and HL60 cells whichsigni√Ø¬¨\\x81cantlytreatmentaloneFrontiers in Oncology  wwwfrontiersinAugust   Volume    0cWu et alDeoxyshikonin Inhibits AML CellsFIGURE   Deoxyshikonin concentrationdependently promoted the apoptosis of acute myeloid leukemia cells THP1 and HL60 cells were treated with increasingdoses of deoxyshikonin    and  ¬µgml followed by detection of the apoptotic rate caspase37 activity and Cyt C protein level by annexin VFITC apoptosisassay AB caspase37 activity assay CD and western blot analysis EF respectively P   compared to the control groupFrontiers in Oncology  wwwfrontiersinAugust   Volume    0cWu et alDeoxyshikonin Inhibits AML CellsFIGURE   Deoxyshikonin concentrationdependently suppressed glycolysis in acute myeloid leukemia cells After THP1 and HL60 cells were administrated withincreasing doses of deoxyshikonin    and  ¬µgml for  h glucose consumption AB and lactate production CD were then measured P  compared to control the groupwas restored by the combined treatment of deoxyshikoninand 740YP Figure 8C Moreover 740YP resisted thedeoxyshikonininduced decrease of PKM2 protein levelFigures 8DE and PKM2 activity Figure 8F in THP1 andHL60 cells These results suggested that the activation of theAktmTOR pathway reversed the e√Ø¬¨ects of deoxyshikonin onglycolysis and PKM2 expression in AML cells To con√Ø¬¨\\x81rmthe abovementioned results THP1 and HL60 cells weretransfected with Aktoverexpressing plasmid pcDNAAkt toactivate the AktmTOR pathway As shown in Figures 9ABthe ratios of pAktAkt and pmTORmTOR were increased h after transfection in THP1 and HL60 cells The CCK8assay showed that deoxyshikonin treatmentcaused viabilityreduction in THP1 and HL60 cells was attenuated aftertransfection with pcDNAAkt Figure 9C The increase ofapoptotic rate in deoxyshikonintreated THP1 and HL60 cellswas signi√Ø¬¨\\x81cantly abolished after transfection with pcDNAAktimpairedthe deoxyshikonin treatmentcaused decrease of glucoseconsumption Figure 9E and lactate production Figure 9Fin THP1 and HL60 cells The deoxyshikonin treatmentinhibited the expression of PKM2 mRNA Figure 9G andprotein Figures 9HI and the activity of PKM2 Figure 9Jwhereas these e√Ø¬¨ects were attenuated after transfection withpcDNAAkt These results con√Ø¬¨\\x81rmed that the activation ofFigure 9D Transfection with pcDNAAktthe AktmTOR pathway reversed the e√Ø¬¨ects of deoxyshikoninon viability apoptosis glycolysis and PKM2 expression inAML cellsDISCUSSIONIn spite of signi√Ø¬¨\\x81cant improvements in therapeutic interventionsof AML the prognosis of patients su√Ø¬¨ering from AML remainsunfavorable and the 5year survival rate of AML patients islower than  accompanied by a high mortality rate  Forthe high mortality there is a great need to identify e√Ø¬¨ectivealternative therapeutic agents speci√Ø¬¨\\x81cally targeting AML It iscommonly reckoned that natural products have the potentialto induce apoptosis in cancer cells including AML and maytherefore be essential sources for anticancer drugs becauseof their extensive biological activities and limited side e√Ø¬¨ects  Shikonin and its derivatives the predominant typeof naphthoquinone derivatives extracted from the root ofLithospermum erythrorhizon Sieb et Zucc have been welldocumented to possess a wide range of pharmacologicalactivities including antitumor activity by suppression of cellproliferation For instance shikonin potently depressed theviability and the metastasis of triplenegative breast cancercells by reversing the epithelialtomesenchymal transition viaFrontiers in Oncology  wwwfrontiersinAugust   Volume    0cWu et alDeoxyshikonin Inhibits AML CellsFIGURE   Deoxyshikonin decreased the expression level and the activity of PKM2 in acute myeloid leukemia cells THP1 and HL60 cells were exposed toincreasing doses of deoxyshikonin    and  ¬µgml for  h A\\x93D The mRNA and protein levels of PKM2 were determined by qRTPCR and western blotrespectively EF PKM2 activity was detected by the lactate dehydrogenasecoupled assay P   compared to the control groupFrontiers in Oncology  wwwfrontiersinAugust   Volume    0cWu et alDeoxyshikonin Inhibits AML CellsFIGURE   Deoxyshikonin inactivated the AktmTOR pathway in AML cells A\\x93C THP1 and HL60 cells were treated with different doses of deoxyshikonin   and  ¬µgml for  h and the protein levels of pAkt Akt pGSK3√é¬≤ GSK3√é¬≤ pmTOR mTOR pp70S6K p70S6K p4EBP1 and 4EBP1 were measured bywestern blot analysis P   compared to the control groupglycogen synthase kinase 3√é¬≤regulated repression of √é¬≤cateninsignaling  Additionally √é¬≤dimethylacrylshikonin suppressedcell viability and induced mitochondriadependent apoptosisin human lung adenocarcinoma cells via the activation of thep38 signaling pathway  Acetylshikonin another shikoninderivative signi√Ø¬¨\\x81cantly inhibited the anchorageindependentgrowth of pancreatic cancer cells by suppressing the nuclearfactorkappa B signaling pathway  More importantly itwas previously demonstrated that deoxyshikonin exhibited antiproliferative and proapoptotic activities in colorectal cancercells through the phosphoinositide 3kinase PI3KAktmTORpathway  Nevertheless whether deoxyshikonin showed anantitumor activity in AML remained far from being addressedTo our knowledgetime to demonstratethat deoxyshikonin dampened the viability of AML cellsin a dosedependent manner Meanwhile we found thatexposure to deoxyshikonin led to a concentrationdependentthis is the √Ø¬¨\\x81rstincrease of the apoptotic rate caspase37 activity and CytC protein levelin AML cells The increase of caspase activity is an important indicator of apoptosis These resultssuggested that deoxyshikonin exerted an antitumor activityin AML cells The e√Ø¬¨ects of deoxyshikonin on the viabilityand the apoptosis of normal bone marrow stromal HS5cells were also evaluated in this study The results showedthat deoxyshikonin at  ¬µgml did not a√Ø¬¨ect HS5 cellviability and apoptosis Supplementary Figure  suggestingthat deoxyshikonin was selectively toxic to cancer cells but notto normal cellsIt has been proven that disruption of aerobic glycolysisrestricts cancer carcinogenesis suggesting that elevated aerobicglycolysis facilitates tumor development and oncogenesis Our study provided evidence that deoxyshikonin inhibitedglycolysis in AML cells as demonstrated by decreased glucoseconsumption and lactate production Moreover we found thatFrontiers in Oncology  wwwfrontiersinAugust   Volume    0cWu et alDeoxyshikonin Inhibits AML CellsFIGURE   Activation of the AktmTOR pathway reversed the effects of deoxyshikonin on the viability and the apoptosis of acute myeloid leukemia cells AB Theprotein levels of pAkt Akt pmTOR and mTOR in the THP1 and HL60 cells treated with 740YP for  h were detected by western blot analysis THP1 and HL60cells were treated with  ¬µgml deoxyshikonin in the presence or the absence of  ¬µM 740YP for  h followed by assessment of cell viability CD apoptosisEF caspase37 activity GH and Cyt C protein level IJ by CCK8 assay √Ø¬¨\\x82ow cytometry analysis caspase37 activity assay and western blot analysisrespectively P   compared to the control group P   compared to the deoxyshikonin treatment groupdeoxyshikonin inhibited the expression of PKM2 in AML cellsPKM2 a critical ratelimiting enzyme of aerobic glycolysis isproposed to play a crucial role in glycolysis process and cancerprogression  The robust expression of PKM2 has beenobserved in various human cancers which facilitates cancercell proliferation and growth  These √Ø¬¨\\x81ndings togetherrevealed that deoxyshikonin inhibited glycolysis in AML cells bysuppressing PKM2It has been demonstrated that the AktmTOR signalingnetwork is constitutively activated and associated with thedevelopment of several types of cancers including AML  Overactivation of the AktmTOR signaling is involvedin the elevated aerobic glycolysis of cancer cellstherebycontributing to cancer cell survival and growth  Thusthe AktmTOR pathway may be regarded as a promisingtherapeutic target for cancer treatment  To elucidate themolecular mechanism underlying the antitumor e√Ø¬¨ects ofdeoxyshikonin we detected the \\x82uence of deoxyshikoninon the AktmTOR signaling in AML cells It was shownthat deoxyshikonin impeded the activation of the AktmTORsignaling in AML cells In the restoration assay activation ofthe AktmTOR signaling by 740YP or pcDNAAkt plasmidabolished the antitumor e√Ø¬¨ect of deoxyshikonin in AML cellsTaken togetherthese results suggested that deoxyshikoninFrontiers in Oncology  wwwfrontiersinAugust   Volume    0cWu et alDeoxyshikonin Inhibits AML CellsFIGURE   Activation of the AktmTOR pathway reversed the effects of deoxyshikonin on glycolysis and PKM2 expression in acute myeloid leukemia cells THP1and HL60 cells were exposed to  ¬µgml deoxyshikonin or together with  ¬µM 740YP for  h AB Glucose consumption and lactate production in thesupernatants of treated THP1 and HL60 cells were measured C The mRNA level of PKM2 in treated THP1 and HL60 cells was estimated by quantitative realtimepolymerase chain reaction DE The protein level of PKM2 in the treated THP1 and HL60 cells was estimated by western blot F PKM2 activity was detected bythe lactate dehydrogenasecoupled assay P   compared to the control group P   compared to the deoxyshikonin treatment groupFrontiers in Oncology  wwwfrontiersinAugust   Volume    0cWu et alDeoxyshikonin Inhibits AML CellsFIGURE   Transfection with pcDNAAkt reversed the effects of deoxyshikonin on viability apoptosis glycolysis and PKM2 expression in acute myeloid leukemiacells AB The Aktoverexpressing plasmid pcDNAAkt and empty vector pcDNA control were transfected into THP1 and HL60 cells After  h the protein levelsof pAkt Akt pmTOR and mTOR were detected by western blot analysis Transfected THP1 and HL60 cells were exposed to  ¬µgml deoxyshikonin for  hfollowed by assessment of cell viability C apoptosis D glucose consumption E and lactate production F as well as PKM2 mRNA G protein HI and activityJ P   compared to the pcDNA group P   compared to the pcDNA  deoxyshikonin groupFrontiers in Oncology  wwwfrontiersinAugust   Volume    0cWu et alDeoxyshikonin Inhibits AML Cellsdampened the viability and the glycolysis of AML cells bysuppressing PKM2 via inactivation of the AktmTOR signalingThe main defect ofthis study is that experiments wereonly performed on two AML cell lines THP1 and HL60and no primary AML cells were tested The heterogeneityof AML is therefore not taken into account Future studiesshould explore the role of deoxyshikonin using primaryAML cellsCONCLUSIONTo sum up our study provided the √Ø¬¨\\x81rst evidence thatdeoxyshikonin exerted antitumor and antiglycolytic activitiesin AML cells by suppressing PKM2 via inactivation of theAktmTOR signaling Our study provided novel insights intothe antitumor and antiglycolytic activities of deoxyshikonin inAML Deoxyshikonin may be a promising anticancer candidateagent in AML cellsREFERENCES Kavanagh S Murphy T Law A Yehudai D Ho JM Chan S et al Emergingtherapies for acute myeloid leukemia translating biology into the clinic JCIInsight    101172jciinsight95679 Papaemmanuil E Gerstung M Bullinger L Gaidzik VI Paschka PRoberts ND et al Genomic classi√Ø¬¨\\x81cation and prognosis in acute myeloidleukemia N EnglJ Med  \\x93  101056NEJMoa1 Siegel RL Miller KD Jemal A Cancer statistics  CA Cancer J Clin \\x93  103322caac21442 Krug U Berdel WE Gale RP Haferlach C Schnittger S MullerTidow C etal Increasing intensity of therapies assigned at diagnosis does not improvesurvival of adults with acute myeloid leukemia Leukemia  \\x93 101038leu201625 Gao H Liu L Qu ZY Wei FX Wang SQ Chen G et al Antiadenovirus activities ofshikonin a component of Chinese herbalmedicine in vitro Biol Pharm Bull  \\x93  101248bpb Li W Zhang C Ren A Li T Jin R Li G et al Shikonin suppressesskin carcinogenesis via inhibiting cell proliferation PLoS ONE 10e0126459  101371journalpone0126459 Andujar I Rios JL Giner RM Recio MC Pharmacological properties ofshikonin  a review of literature since  Planta Med  \\x93 101055s00331350934 Li W Liu J Zhao Y PKM2 inhibitor shikonin suppresses TPAinducedmitochondrial malfunction and proliferation of skin epidermal JB6 cells MolCarcinog  \\x93  101002mc21988 Boulos JC Rahama M Hegazy MF E√Ø¬¨erth T Shikonin derivatives \\x93for cancer prevention and therapy Cancer Lett 101016jcanlet201904033 Zhu Y Zhong Y Long X Zhu Z Zhou Y Ye H et al Deoxyshikoninisolated from Arnebia euchroma inhibits colorectal cancer by downregulating the PI3KAktmTOR pathway PharmBiol  \\x93  Liberti MV Locasale JW The Warburg e√Ø¬¨ect how does it bene√Ø¬¨\\x81t cancer cellsDATA AVAILABILITY STATEMENTAll datasets generated for this study are included in theSupplementary MaterialAUTHOR CONTRIBUTIONSHW conducted the experiments and participated in theconception and the design of the study HZ conducted theexperiments and performed the analysis LC contributed toanalyzing the data and drafting the manuscript All authorscontributed to the  and approved the submitted versionSUPPLEMENTARY MATERIALThe Supplementary Materialonline202001253fullsupplementarymaterialfor this  can be foundhttpswwwfrontiersins103389foncat Xu D Liang J Lin J Yu C PKM2 a potential regulator of rheumatoid arthritisvia glycolytic and nonglycolytic pathways Front Immunol   103389√Ø¬¨\\x81mmu201902919 Gong T Cui L Wang H Wang H Han N Knockdown of KLF5 suppresseshypoxiainduced resistance to cisplatin in NSCLC cells by regulating HIF1√é¬±dependent glycolysis through inactivation of the PI3KAktmTOR pathway JTransl Med    101186s1296701815432 Nemazanyy I Espeillac C Pende M Panasyuk G Role of PI3K mTOR andAkt2 signalling in hepatic tumorigenesis via the control of PKM2 expressionBiochem Soc Trans  \\x93  101042BST20130034 Zheng YL Li L Jia YX Zhang BZ Li JC Zhu YH et al LINC01554mediated glucose metabolism reprogramming suppresses tumorigenicityin hepatocellular carcinoma via downregulating PKM2 expression andinhibiting AktmTOR signaling pathway Theranostics  \\x93 107150thno28992 Zhao X Zhu Y Hu J Jiang L Li L Jia S et al Shikonin inhibits tumor growthin mice by suppressing pyruvate kinase M2mediated aerobic glycolysis SciRep    101038s4159801831615y Tang J Ren YG Zhao J Long F Chen J Jiang Z Shikonin enhancessensitization of ge√Ø¬¨\\x81tinib against wildtype EGFR nonsmall cell lung cancervia inhibition PKM2stat3cyclinD1 signal pathway Life Sci  \\x93 101016jlfs201805012 Popescu NC Cheng SY Chromosomal localization of the gene for a humancytosolic thyroid hormone binding protein homologous to the subunitof pyruvate kinase subtype M2 Somat Cell Mol Genet  \\x93 101007BF01233100 Martelli AMF McCubreyEvangelisti C ChiariniJA Thephosphatidylinositol 3kinaseAktmTOR signaling network as a therapeutictarget in acute myelogenous leukemia patients Oncotarget  \\x93 1018632oncotarget114 Yang D Zhang X Zhang X Xu Y The progress and current status ofimmunotherapy in acute myeloid leukemia Ann Hematol  \\x93 101007s002770173148x Wang X Feng Y Chinese medicines induce cell death the molecular andcellular mechanisms for cancer therapy Biomed Res Int   Trends Biochem Sci  \\x93  101016jtibs201512001 Wang ZY Chen Z Acute promyelocytic leukemia from highly fatal to highly Hanahan D Weinberg RA Hallmarks of cancer the next generation Cellcurable Blood  \\x93  101182blood200707102798 \\x93  101016jcell201102013 Herst PM Howma'"
      ]
     },
     "execution_count": 63,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df['note'][1]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "e51ec64e",
   "metadata": {
    "slideshow": {
     "slide_type": "slide"
    }
   },
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "\n",
    "df= pd.read_csv(\"alldata_1_for_kaggle.csv\", encoding='ISO-8859-1')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "91a235bb",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Thyroid_Cancer    2810\n",
       "Colon_Cancer      2580\n",
       "Lung_Cancer       2180\n",
       "Name: 0, dtype: int64"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df['0'].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "60479b9c",
   "metadata": {},
   "outputs": [],
   "source": [
    "df = df[df.columns[1:]]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "3428c4f7",
   "metadata": {},
   "outputs": [],
   "source": [
    "df.columns = ['cancer_type', \"note\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "id": "a9c14aed",
   "metadata": {},
   "outputs": [],
   "source": [
    "df = df[df.cancer_type.isin(['Thyroid_Cancer', 'Lung_Cancer'])].sample(1000).reset_index(drop=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "id": "53e3f06d",
   "metadata": {},
   "outputs": [],
   "source": [
    "X = df[['note']]\n",
    "y = df[['cancer_type']]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "id": "096cc26d",
   "metadata": {},
   "outputs": [],
   "source": [
    "from sklearn.model_selection import train_test_split\n",
    "\n",
    "X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.2, random_state=42)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "id": "5acc0ff5",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Requirement already satisfied: spacy in /Users/vmarquez/opt/anaconda3/lib/python3.9/site-packages (3.5.3)\n",
      "Requirement already satisfied: murmurhash<1.1.0,>=0.28.0 in /Users/vmarquez/opt/anaconda3/lib/python3.9/site-packages (from spacy) (1.0.7)\n",
      "Requirement already satisfied: requests<3.0.0,>=2.13.0 in /Users/vmarquez/opt/anaconda3/lib/python3.9/site-packages (from spacy) (2.26.0)\n",
      "Requirement already satisfied: cymem<2.1.0,>=2.0.2 in /Users/vmarquez/opt/anaconda3/lib/python3.9/site-packages (from spacy) (2.0.6)\n",
      "Requirement already satisfied: setuptools in /Users/vmarquez/opt/anaconda3/lib/python3.9/site-packages (from spacy) (62.2.0)\n",
      "Requirement already satisfied: spacy-legacy<3.1.0,>=3.0.11 in /Users/vmarquez/opt/anaconda3/lib/python3.9/site-packages (from spacy) (3.0.12)\n",
      "Requirement already satisfied: catalogue<2.1.0,>=2.0.6 in /Users/vmarquez/opt/anaconda3/lib/python3.9/site-packages (from spacy) (2.0.7)\n",
      "Requirement already satisfied: pathy>=0.10.0 in /Users/vmarquez/opt/anaconda3/lib/python3.9/site-packages (from spacy) (0.10.1)\n",
      "Requirement already satisfied: thinc<8.2.0,>=8.1.8 in /Users/vmarquez/opt/anaconda3/lib/python3.9/site-packages (from spacy) (8.1.10)\n",
      "Requirement already satisfied: pydantic!=1.8,!=1.8.1,<1.11.0,>=1.7.4 in /Users/vmarquez/opt/anaconda3/lib/python3.9/site-packages (from spacy) (1.8.2)\n",
      "Requirement already satisfied: wasabi<1.2.0,>=0.9.1 in /Users/vmarquez/opt/anaconda3/lib/python3.9/site-packages (from spacy) (0.9.1)\n",
      "Requirement already satisfied: spacy-loggers<2.0.0,>=1.0.0 in /Users/vmarquez/opt/anaconda3/lib/python3.9/site-packages (from spacy) (1.0.2)\n",
      "Requirement already satisfied: tqdm<5.0.0,>=4.38.0 in /Users/vmarquez/opt/anaconda3/lib/python3.9/site-packages (from spacy) (4.62.3)\n",
      "Requirement already satisfied: preshed<3.1.0,>=3.0.2 in /Users/vmarquez/opt/anaconda3/lib/python3.9/site-packages (from spacy) (3.0.6)\n",
      "Requirement already satisfied: srsly<3.0.0,>=2.4.3 in /Users/vmarquez/opt/anaconda3/lib/python3.9/site-packages (from spacy) (2.4.3)\n",
      "Requirement already satisfied: numpy>=1.15.0 in /Users/vmarquez/opt/anaconda3/lib/python3.9/site-packages (from spacy) (1.20.3)\n",
      "Requirement already satisfied: langcodes<4.0.0,>=3.2.0 in /Users/vmarquez/opt/anaconda3/lib/python3.9/site-packages (from spacy) (3.3.0)\n",
      "Requirement already satisfied: typer<0.8.0,>=0.3.0 in /Users/vmarquez/opt/anaconda3/lib/python3.9/site-packages (from spacy) (0.4.1)\n",
      "Requirement already satisfied: smart-open<7.0.0,>=5.2.1 in /Users/vmarquez/opt/anaconda3/lib/python3.9/site-packages (from spacy) (5.2.1)\n",
      "Requirement already satisfied: packaging>=20.0 in /Users/vmarquez/opt/anaconda3/lib/python3.9/site-packages (from spacy) (21.3)\n",
      "Requirement already satisfied: jinja2 in /Users/vmarquez/opt/anaconda3/lib/python3.9/site-packages (from spacy) (2.11.3)\n",
      "Requirement already satisfied: pyparsing!=3.0.5,>=2.0.2 in /Users/vmarquez/opt/anaconda3/lib/python3.9/site-packages (from packaging>=20.0->spacy) (3.0.4)\n",
      "Requirement already satisfied: typing-extensions>=3.7.4.3 in /Users/vmarquez/opt/anaconda3/lib/python3.9/site-packages (from pydantic!=1.8,!=1.8.1,<1.11.0,>=1.7.4->spacy) (4.4.0)\n",
      "Requirement already satisfied: urllib3<1.27,>=1.21.1 in /Users/vmarquez/opt/anaconda3/lib/python3.9/site-packages (from requests<3.0.0,>=2.13.0->spacy) (1.26.14)\n",
      "Requirement already satisfied: charset-normalizer~=2.0.0 in /Users/vmarquez/opt/anaconda3/lib/python3.9/site-packages (from requests<3.0.0,>=2.13.0->spacy) (2.0.4)\n",
      "Requirement already satisfied: idna<4,>=2.5 in /Users/vmarquez/opt/anaconda3/lib/python3.9/site-packages (from requests<3.0.0,>=2.13.0->spacy) (3.2)\n",
      "Requirement already satisfied: certifi>=2017.4.17 in /Users/vmarquez/opt/anaconda3/lib/python3.9/site-packages (from requests<3.0.0,>=2.13.0->spacy) (2021.10.8)\n",
      "Requirement already satisfied: confection<1.0.0,>=0.0.1 in /Users/vmarquez/opt/anaconda3/lib/python3.9/site-packages (from thinc<8.2.0,>=8.1.8->spacy) (0.0.4)\n",
      "Requirement already satisfied: blis<0.8.0,>=0.7.8 in /Users/vmarquez/opt/anaconda3/lib/python3.9/site-packages (from thinc<8.2.0,>=8.1.8->spacy) (0.7.9)\n",
      "Requirement already satisfied: click<9.0.0,>=7.1.1 in /Users/vmarquez/opt/anaconda3/lib/python3.9/site-packages (from typer<0.8.0,>=0.3.0->spacy) (7.1.2)\n",
      "Requirement already satisfied: MarkupSafe>=0.23 in /Users/vmarquez/opt/anaconda3/lib/python3.9/site-packages (from jinja2->spacy) (1.1.1)\n",
      "\u001b[33mWARNING: You are using pip version 22.0.4; however, version 23.1.2 is available.\n",
      "You should consider upgrading via the '/Users/vmarquez/opt/anaconda3/bin/python -m pip install --upgrade pip' command.\u001b[0m\u001b[33m\n",
      "\u001b[0m"
     ]
    }
   ],
   "source": [
    "! pip install -U spacy"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "id": "ea0e55a0",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Collecting en-core-web-sm==3.5.0\n",
      "  Downloading https://github.com/explosion/spacy-models/releases/download/en_core_web_sm-3.5.0/en_core_web_sm-3.5.0-py3-none-any.whl (12.8 MB)\n",
      "\u001b[2K     \u001b[90m‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ\u001b[0m \u001b[32m12.8/12.8 MB\u001b[0m \u001b[31m2.1 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m00:01\u001b[0m00:01\u001b[0m\n",
      "\u001b[?25hRequirement already satisfied: spacy<3.6.0,>=3.5.0 in /Users/vmarquez/opt/anaconda3/lib/python3.9/site-packages (from en-core-web-sm==3.5.0) (3.5.3)\n",
      "Requirement already satisfied: numpy>=1.15.0 in /Users/vmarquez/opt/anaconda3/lib/python3.9/site-packages (from spacy<3.6.0,>=3.5.0->en-core-web-sm==3.5.0) (1.20.3)\n",
      "Requirement already satisfied: srsly<3.0.0,>=2.4.3 in /Users/vmarquez/opt/anaconda3/lib/python3.9/site-packages (from spacy<3.6.0,>=3.5.0->en-core-web-sm==3.5.0) (2.4.3)\n",
      "Requirement already satisfied: spacy-legacy<3.1.0,>=3.0.11 in /Users/vmarquez/opt/anaconda3/lib/python3.9/site-packages (from spacy<3.6.0,>=3.5.0->en-core-web-sm==3.5.0) (3.0.12)\n",
      "Requirement already satisfied: packaging>=20.0 in /Users/vmarquez/opt/anaconda3/lib/python3.9/site-packages (from spacy<3.6.0,>=3.5.0->en-core-web-sm==3.5.0) (21.3)\n",
      "Requirement already satisfied: typer<0.8.0,>=0.3.0 in /Users/vmarquez/opt/anaconda3/lib/python3.9/site-packages (from spacy<3.6.0,>=3.5.0->en-core-web-sm==3.5.0) (0.4.1)\n",
      "Requirement already satisfied: setuptools in /Users/vmarquez/opt/anaconda3/lib/python3.9/site-packages (from spacy<3.6.0,>=3.5.0->en-core-web-sm==3.5.0) (62.2.0)\n",
      "Requirement already satisfied: smart-open<7.0.0,>=5.2.1 in /Users/vmarquez/opt/anaconda3/lib/python3.9/site-packages (from spacy<3.6.0,>=3.5.0->en-core-web-sm==3.5.0) (5.2.1)\n",
      "Requirement already satisfied: spacy-loggers<2.0.0,>=1.0.0 in /Users/vmarquez/opt/anaconda3/lib/python3.9/site-packages (from spacy<3.6.0,>=3.5.0->en-core-web-sm==3.5.0) (1.0.2)\n",
      "Requirement already satisfied: preshed<3.1.0,>=3.0.2 in /Users/vmarquez/opt/anaconda3/lib/python3.9/site-packages (from spacy<3.6.0,>=3.5.0->en-core-web-sm==3.5.0) (3.0.6)\n",
      "Requirement already satisfied: jinja2 in /Users/vmarquez/opt/anaconda3/lib/python3.9/site-packages (from spacy<3.6.0,>=3.5.0->en-core-web-sm==3.5.0) (2.11.3)\n",
      "Requirement already satisfied: pathy>=0.10.0 in /Users/vmarquez/opt/anaconda3/lib/python3.9/site-packages (from spacy<3.6.0,>=3.5.0->en-core-web-sm==3.5.0) (0.10.1)\n",
      "Requirement already satisfied: thinc<8.2.0,>=8.1.8 in /Users/vmarquez/opt/anaconda3/lib/python3.9/site-packages (from spacy<3.6.0,>=3.5.0->en-core-web-sm==3.5.0) (8.1.10)\n",
      "Requirement already satisfied: langcodes<4.0.0,>=3.2.0 in /Users/vmarquez/opt/anaconda3/lib/python3.9/site-packages (from spacy<3.6.0,>=3.5.0->en-core-web-sm==3.5.0) (3.3.0)\n",
      "Requirement already satisfied: wasabi<1.2.0,>=0.9.1 in /Users/vmarquez/opt/anaconda3/lib/python3.9/site-packages (from spacy<3.6.0,>=3.5.0->en-core-web-sm==3.5.0) (0.9.1)\n",
      "Requirement already satisfied: cymem<2.1.0,>=2.0.2 in /Users/vmarquez/opt/anaconda3/lib/python3.9/site-packages (from spacy<3.6.0,>=3.5.0->en-core-web-sm==3.5.0) (2.0.6)\n",
      "Requirement already satisfied: murmurhash<1.1.0,>=0.28.0 in /Users/vmarquez/opt/anaconda3/lib/python3.9/site-packages (from spacy<3.6.0,>=3.5.0->en-core-web-sm==3.5.0) (1.0.7)\n",
      "Requirement already satisfied: catalogue<2.1.0,>=2.0.6 in /Users/vmarquez/opt/anaconda3/lib/python3.9/site-packages (from spacy<3.6.0,>=3.5.0->en-core-web-sm==3.5.0) (2.0.7)\n",
      "Requirement already satisfied: pydantic!=1.8,!=1.8.1,<1.11.0,>=1.7.4 in /Users/vmarquez/opt/anaconda3/lib/python3.9/site-packages (from spacy<3.6.0,>=3.5.0->en-core-web-sm==3.5.0) (1.8.2)\n",
      "Requirement already satisfied: requests<3.0.0,>=2.13.0 in /Users/vmarquez/opt/anaconda3/lib/python3.9/site-packages (from spacy<3.6.0,>=3.5.0->en-core-web-sm==3.5.0) (2.26.0)\n",
      "Requirement already satisfied: tqdm<5.0.0,>=4.38.0 in /Users/vmarquez/opt/anaconda3/lib/python3.9/site-packages (from spacy<3.6.0,>=3.5.0->en-core-web-sm==3.5.0) (4.62.3)\n",
      "Requirement already satisfied: pyparsing!=3.0.5,>=2.0.2 in /Users/vmarquez/opt/anaconda3/lib/python3.9/site-packages (from packaging>=20.0->spacy<3.6.0,>=3.5.0->en-core-web-sm==3.5.0) (3.0.4)\n",
      "Requirement already satisfied: typing-extensions>=3.7.4.3 in /Users/vmarquez/opt/anaconda3/lib/python3.9/site-packages (from pydantic!=1.8,!=1.8.1,<1.11.0,>=1.7.4->spacy<3.6.0,>=3.5.0->en-core-web-sm==3.5.0) (4.4.0)\n",
      "Requirement already satisfied: charset-normalizer~=2.0.0 in /Users/vmarquez/opt/anaconda3/lib/python3.9/site-packages (from requests<3.0.0,>=2.13.0->spacy<3.6.0,>=3.5.0->en-core-web-sm==3.5.0) (2.0.4)\n",
      "Requirement already satisfied: urllib3<1.27,>=1.21.1 in /Users/vmarquez/opt/anaconda3/lib/python3.9/site-packages (from requests<3.0.0,>=2.13.0->spacy<3.6.0,>=3.5.0->en-core-web-sm==3.5.0) (1.26.14)\n",
      "Requirement already satisfied: certifi>=2017.4.17 in /Users/vmarquez/opt/anaconda3/lib/python3.9/site-packages (from requests<3.0.0,>=2.13.0->spacy<3.6.0,>=3.5.0->en-core-web-sm==3.5.0) (2021.10.8)\n",
      "Requirement already satisfied: idna<4,>=2.5 in /Users/vmarquez/opt/anaconda3/lib/python3.9/site-packages (from requests<3.0.0,>=2.13.0->spacy<3.6.0,>=3.5.0->en-core-web-sm==3.5.0) (3.2)\n",
      "Requirement already satisfied: blis<0.8.0,>=0.7.8 in /Users/vmarquez/opt/anaconda3/lib/python3.9/site-packages (from thinc<8.2.0,>=8.1.8->spacy<3.6.0,>=3.5.0->en-core-web-sm==3.5.0) (0.7.9)\n",
      "Requirement already satisfied: confection<1.0.0,>=0.0.1 in /Users/vmarquez/opt/anaconda3/lib/python3.9/site-packages (from thinc<8.2.0,>=8.1.8->spacy<3.6.0,>=3.5.0->en-core-web-sm==3.5.0) (0.0.4)\n",
      "Requirement already satisfied: click<9.0.0,>=7.1.1 in /Users/vmarquez/opt/anaconda3/lib/python3.9/site-packages (from typer<0.8.0,>=0.3.0->spacy<3.6.0,>=3.5.0->en-core-web-sm==3.5.0) (7.1.2)\n",
      "Requirement already satisfied: MarkupSafe>=0.23 in /Users/vmarquez/opt/anaconda3/lib/python3.9/site-packages (from jinja2->spacy<3.6.0,>=3.5.0->en-core-web-sm==3.5.0) (1.1.1)\n",
      "Installing collected packages: en-core-web-sm\n",
      "  Attempting uninstall: en-core-web-sm\n",
      "    Found existing installation: en-core-web-sm 3.3.0\n",
      "    Uninstalling en-core-web-sm-3.3.0:\n",
      "      Successfully uninstalled en-core-web-sm-3.3.0\n",
      "Successfully installed en-core-web-sm-3.5.0\n",
      "\u001b[33mWARNING: You are using pip version 22.0.4; however, version 23.1.2 is available.\n",
      "You should consider upgrading via the '/Users/vmarquez/opt/anaconda3/bin/python -m pip install --upgrade pip' command.\u001b[0m\u001b[33m\n",
      "\u001b[0m\u001b[38;5;2m‚úî Download and installation successful\u001b[0m\n",
      "You can now load the package via spacy.load('en_core_web_sm')\n"
     ]
    }
   ],
   "source": [
    "! python -m spacy download en_core_web_sm"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "id": "786ffd49",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[nltk_data] Downloading package stopwords to\n",
      "[nltk_data]     /Users/vmarquez/nltk_data...\n",
      "[nltk_data]   Package stopwords is already up-to-date!\n",
      "/Users/vmarquez/opt/anaconda3/lib/python3.9/site-packages/spacy/util.py:837: UserWarning: [W095] Model 'en_core_web_sm' (3.5.0) was trained with spaCy v3.5 and may not be 100% compatible with the current version (3.3.0). If you see errors or degraded performance, download a newer compatible model or retrain your custom model with the current spaCy version. For more details and available updates, run: python -m spacy validate\n",
      "  except (TypeError, InvalidVersion):\n"
     ]
    }
   ],
   "source": [
    "import nltk\n",
    "nltk.download('stopwords')\n",
    "\n",
    "import json\n",
    "import pandas as pd\n",
    "import re\n",
    "from nltk.corpus import stopwords\n",
    "import spacy\n",
    "\n",
    "sw = stopwords.words(\"english\")\n",
    "lemma_obj = spacy.load('en_core_web_sm')\n",
    "\n",
    "def text_pre_processing(text):\n",
    "    text = text.lower() # Make it lowercase\n",
    "    text = re.sub(r\"[\\W\\d]\", \" \", text) # Remove punctuation (regex)\n",
    "    text = text.split() # Tokenize\n",
    "    text = [word for word in text if word not in sw] # Remove stop words\n",
    "    text = \" \".join(text) # Make it a string\n",
    "\n",
    "    # Lemmatization\n",
    "    doc = lemma_obj(text)\n",
    "    text = [token.lemma_ for token in doc]\n",
    "\n",
    "    text = \" \".join(text) # Make it a string\n",
    "    return text\n",
    "\n",
    "from sklearn.feature_extraction.text import TfidfVectorizer\n",
    "\n",
    "tfidf_vect = TfidfVectorizer(preprocessor=text_pre_processing)\n",
    "tfidf_vect.fit(X_train.note.values);"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "id": "3ab553fc",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>cancer_type</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>Thyroid_Cancer</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>535</th>\n",
       "      <td>Thyroid_Cancer</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>695</th>\n",
       "      <td>Thyroid_Cancer</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>557</th>\n",
       "      <td>Thyroid_Cancer</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>836</th>\n",
       "      <td>Lung_Cancer</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>106</th>\n",
       "      <td>Thyroid_Cancer</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>270</th>\n",
       "      <td>Thyroid_Cancer</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>860</th>\n",
       "      <td>Lung_Cancer</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>435</th>\n",
       "      <td>Thyroid_Cancer</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>102</th>\n",
       "      <td>Lung_Cancer</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>800 rows √ó 1 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        cancer_type\n",
       "29   Thyroid_Cancer\n",
       "535  Thyroid_Cancer\n",
       "695  Thyroid_Cancer\n",
       "557  Thyroid_Cancer\n",
       "836     Lung_Cancer\n",
       "..              ...\n",
       "106  Thyroid_Cancer\n",
       "270  Thyroid_Cancer\n",
       "860     Lung_Cancer\n",
       "435  Thyroid_Cancer\n",
       "102     Lung_Cancer\n",
       "\n",
       "[800 rows x 1 columns]"
      ]
     },
     "execution_count": 34,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "y_train"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "id": "7f5037db",
   "metadata": {},
   "outputs": [],
   "source": [
    "X_train = tfidf_vect.transform(X_train.note.values)\n",
    "y_train = y_train.cancer_type.values\n",
    "\n",
    "X_test = tfidf_vect.transform(X_test.note.values)\n",
    "y_test = y_test.cancer_type.values\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "id": "8ec60d15",
   "metadata": {},
   "outputs": [],
   "source": [
    "from sklearn.naive_bayes import MultinomialNB \n",
    "from sklearn.linear_model import LogisticRegression\n",
    "from sklearn.svm import LinearSVC\n",
    "from sklearn.tree import DecisionTreeClassifier\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "id": "9cd03dfb",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Initiliaze classifiers\n",
    "nb = MultinomialNB()\n",
    "logreg = LogisticRegression(class_weight='balanced')\n",
    "svm = LinearSVC(class_weight='balanced')\n",
    "dt = DecisionTreeClassifier(class_weight='balanced')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "id": "18b40505",
   "metadata": {
    "slideshow": {
     "slide_type": "slide"
    }
   },
   "outputs": [],
   "source": [
    "# Train the models\n",
    "nb.fit(X_train, y_train)\n",
    "logreg.fit(X_train, y_train)\n",
    "svm.fit(X_train, y_train)\n",
    "dt.fit(X_train, y_train);"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "id": "6ad8bfab",
   "metadata": {
    "slideshow": {
     "slide_type": "slide"
    }
   },
   "outputs": [],
   "source": [
    "y_pred_nb = nb.predict(X_test)\n",
    "y_pred_logreg = logreg.predict(X_test)\n",
    "y_pred_svm = svm.predict(X_test)\n",
    "y_pred_dt = dt.predict(X_test)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "7da83498",
   "metadata": {},
   "outputs": [],
   "source": [
    "pip install --upgrade scikit-learn"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "id": "dfc790e8",
   "metadata": {},
   "outputs": [],
   "source": [
    "from sklearn.metrics import accuracy_score, f1_score\n",
    "from sklearn.metrics import precision_score, recall_score\n",
    "from sklearn.metrics import confusion_matrix, classification_report\n",
    "\n",
    "import matplotlib.pyplot as plt\n",
    "from sklearn.metrics import plot_confusion_matrix"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "id": "28697b55",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Naive Bayes:            0.97\n",
      "Regresi√≥n Log√≠stica:    0.99\n",
      "SVM:                    1.00\n",
      "√Årbol de decisi√≥n:      1.00\n"
     ]
    }
   ],
   "source": [
    "print(f\"Naive Bayes: {accuracy_score(y_test, y_pred_nb):>15.2f}\")\n",
    "print(f\"Regresi√≥n Log√≠stica: {accuracy_score(y_test, y_pred_logreg):>7.2f}\")\n",
    "print(f\"SVM: {accuracy_score(y_test, y_pred_svm):>23.2f}\")\n",
    "print(f\"√Årbol de decisi√≥n: {accuracy_score(y_test, y_pred_dt):>9.2f}\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "id": "fc6b66e8",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "precision_score\n",
      "Naive Bayes: 0.95\n",
      "Regresi√≥n Log√≠stica: 0.98\n",
      "SVM: 1.00\n",
      "√Årbol de decisi√≥n: 1.00\n",
      "\n",
      "recall_score\n",
      "Naive Bayes: 1.00\n",
      "Regresi√≥n Log√≠stica: 1.00\n",
      "SVM: 1.00\n",
      "√Årbol de decisi√≥n: 1.00\n",
      "\n",
      "f1_score\n",
      "Naive Bayes: 0.97\n",
      "Regresi√≥n Log√≠stica: 0.99\n",
      "SVM: 1.00\n",
      "√Årbol de decisi√≥n: 1.00\n",
      "\n"
     ]
    }
   ],
   "source": [
    "metricas = [precision_score, recall_score, f1_score]\n",
    "\n",
    "for metrica in metricas:\n",
    "    print(metrica.__name__)\n",
    "    print(f\"Naive Bayes: {metrica(y_test, y_pred_nb, pos_label='Thyroid_Cancer'):.2f}\")\n",
    "    print(f\"Regresi√≥n Log√≠stica: {metrica(y_test, y_pred_logreg, pos_label='Thyroid_Cancer'):.2f}\")\n",
    "    print(f\"SVM: {metrica(y_test, y_pred_svm, pos_label='Thyroid_Cancer'):.2f}\")\n",
    "    print(f\"√Årbol de decisi√≥n: {metrica(y_test, y_pred_dt, pos_label='Thyroid_Cancer'):.2f}\")\n",
    "    print()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "id": "b0c48e6c",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Naive Bayes\n",
      "                precision    recall  f1-score   support\n",
      "\n",
      "   Lung_Cancer       1.00      0.93      0.97        90\n",
      "Thyroid_Cancer       0.95      1.00      0.97       110\n",
      "\n",
      "      accuracy                           0.97       200\n",
      "     macro avg       0.97      0.97      0.97       200\n",
      "  weighted avg       0.97      0.97      0.97       200\n",
      "\n",
      "\n",
      "Regresi√≥n Log√≠stica\n",
      "                precision    recall  f1-score   support\n",
      "\n",
      "   Lung_Cancer       1.00      0.98      0.99        90\n",
      "Thyroid_Cancer       0.98      1.00      0.99       110\n",
      "\n",
      "      accuracy                           0.99       200\n",
      "     macro avg       0.99      0.99      0.99       200\n",
      "  weighted avg       0.99      0.99      0.99       200\n",
      "\n",
      "\n",
      "SVM\n",
      "                precision    recall  f1-score   support\n",
      "\n",
      "   Lung_Cancer       1.00      1.00      1.00        90\n",
      "Thyroid_Cancer       1.00      1.00      1.00       110\n",
      "\n",
      "      accuracy                           1.00       200\n",
      "     macro avg       1.00      1.00      1.00       200\n",
      "  weighted avg       1.00      1.00      1.00       200\n",
      "\n",
      "\n",
      "√Årbol de decisi√≥n\n",
      "                precision    recall  f1-score   support\n",
      "\n",
      "   Lung_Cancer       1.00      1.00      1.00        90\n",
      "Thyroid_Cancer       1.00      1.00      1.00       110\n",
      "\n",
      "      accuracy                           1.00       200\n",
      "     macro avg       1.00      1.00      1.00       200\n",
      "  weighted avg       1.00      1.00      1.00       200\n",
      "\n",
      "\n"
     ]
    }
   ],
   "source": [
    "print(\"Naive Bayes\")\n",
    "print(classification_report(y_test, y_pred_nb))\n",
    "print()\n",
    "\n",
    "print(\"Regresi√≥n Log√≠stica\")\n",
    "print(classification_report(y_test, y_pred_logreg))\n",
    "print()\n",
    "\n",
    "print(\"SVM\")\n",
    "print(classification_report(y_test, y_pred_svm))\n",
    "print()\n",
    "\n",
    "print(\"√Årbol de decisi√≥n\")\n",
    "print(classification_report(y_test, y_pred_dt))\n",
    "print()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "id": "17a63ef3",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array([[ 84,   6],\n",
       "       [  0, 110]])"
      ]
     },
     "execution_count": 46,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "confusion_matrix(y_test, y_pred_nb)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "id": "044622cf",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "image/svg+xml": [
       "<?xml version=\"1.0\" encoding=\"utf-8\" standalone=\"no\"?>\n",
       "<!DOCTYPE svg PUBLIC \"-//W3C//DTD SVG 1.1//EN\"\n",
       "  \"http://www.w3.org/Graphics/SVG/1.1/DTD/svg11.dtd\">\n",
       "<!-- Created with matplotlib (https://matplotlib.org/) -->\n",
       "<svg height=\"560.050682pt\" version=\"1.1\" viewBox=\"0 0 620.057102 560.050682\" width=\"620.057102pt\" xmlns=\"http://www.w3.org/2000/svg\" xmlns:xlink=\"http://www.w3.org/1999/xlink\">\n",
       " <metadata>\n",
       "  <rdf:RDF xmlns:cc=\"http://creativecommons.org/ns#\" xmlns:dc=\"http://purl.org/dc/elements/1.1/\" xmlns:rdf=\"http://www.w3.org/1999/02/22-rdf-syntax-ns#\">\n",
       "   <cc:Work>\n",
       "    <dc:type rdf:resource=\"http://purl.org/dc/dcmitype/StillImage\"/>\n",
       "    <dc:date>2023-05-15T13:24:44.011875</dc:date>\n",
       "    <dc:format>image/svg+xml</dc:format>\n",
       "    <dc:creator>\n",
       "     <cc:Agent>\n",
       "      <dc:title>Matplotlib v3.3.1, https://matplotlib.org/</dc:title>\n",
       "     </cc:Agent>\n",
       "    </dc:creator>\n",
       "   </cc:Work>\n",
       "  </rdf:RDF>\n",
       " </metadata>\n",
       " <defs>\n",
       "  <style type=\"text/css\">*{stroke-linecap:butt;stroke-linejoin:round;}</style>\n",
       " </defs>\n",
       " <g id=\"figure_1\">\n",
       "  <g id=\"patch_1\">\n",
       "   <path d=\"M 0 560.050682 \n",
       "L 620.057102 560.050682 \n",
       "L 620.057102 0 \n",
       "L 0 0 \n",
       "z\n",
       "\" style=\"fill:none;\"/>\n",
       "  </g>\n",
       "  <g id=\"axes_1\">\n",
       "   <g id=\"patch_2\">\n",
       "    <path d=\"M 105.584375 225.707216 \n",
       "L 308.493466 225.707216 \n",
       "L 308.493466 22.798125 \n",
       "L 105.584375 22.798125 \n",
       "z\n",
       "\" style=\"fill:#ffffff;\"/>\n",
       "   </g>\n",
       "   <g clip-path=\"url(#p227bc72e00)\">\n",
       "    <image height=\"203\" id=\"image40009a8c3e\" transform=\"scale(1 -1)translate(0 -203)\" width=\"203\" x=\"105.584375\" xlink:href=\"data:image/png;base64,\n",
       "iVBORw0KGgoAAAANSUhEUgAAAMsAAADLCAYAAADA+2czAAACWElEQVR4nO3VMQ3CUABFUSAYqBgsoKY6WPGEk6rowNDuTYe7fUjOUfCWm3ddv9t24adMj3n0BE7cRg+AfyEWiMQCkVggEgtEYoFILBCJBSKxQCQWiMQCkVggEgtEYoFILBCJBSKxQCQWiMQCkVggEgtEYoFILBCJBSKxQCQWiMQCkVggEgtEYoFILBCJBSKxQCQWiMQCkVggEgtEYoFILBCJBSKxQCQWiMQCkVggEgtEYoFILBCJBSKxQCQWiMQCkVggEgtEYoFILBCJBSKxQCQWiMQCkVggEgtEYoFILBCJBSKxQCQWiMQCkVggEgtEYoFILBCJBSKxQCQWiMQCkVggEgtEYoFILBCJBSKxQCQWiMQCkVggEgtEYoFILBCJBSKxQCQWiMQCkVggEgtEYoFILBCJBSKxQCQWiMQCkVggEgtE9+n5Gr2Bg+XzHj2BE54FIrFAJBaIxAKRWCASC0RigUgsEIkFIrFAJBaIxAKRWCASC0RigUgsEIkFIrFAJBaIxAKRWCASC0RigUgsEIkFIrFAJBaIxAKRWCASC0RigUgsEIkFIrFAJBaIxAKRWCASC0RigUgsEIkFIrFAJBaIxAKRWCASC0RigUgsEIkFIrFAJBaIxAKRWCASC0RigUgsEIkFIrFAJBaIxAKRWCASC0RigUgsEIkFIrFAJBaIxAKRWCASC0RigUgsEIkFIrFAJBaIxAKRWCASC0RigUgsEIkFIrFAJBaIxAKRWCASC0RigUgsEIkFIrFAJBaIxAKRWCASC0RigUgsEIkFIrFAJBaIxALRDmLNCUkNxQdjAAAAAElFTkSuQmCC\" y=\"-22.707216\"/>\n",
       "   </g>\n",
       "   <g id=\"matplotlib.axis_1\">\n",
       "    <g id=\"xtick_1\">\n",
       "     <g id=\"line2d_1\">\n",
       "      <defs>\n",
       "       <path d=\"M 0 0 \n",
       "L 0 3.5 \n",
       "\" id=\"mb703f7e117\" style=\"stroke:#000000;stroke-width:0.8;\"/>\n",
       "      </defs>\n",
       "      <g>\n",
       "       <use style=\"stroke:#000000;stroke-width:0.8;\" x=\"156.311648\" xlink:href=\"#mb703f7e117\" y=\"225.707216\"/>\n",
       "      </g>\n",
       "     </g>\n",
       "     <g id=\"text_1\">\n",
       "      <!-- Lung_Cancer -->\n",
       "      <g transform=\"translate(123.996023 240.129091)scale(0.1 -0.1)\">\n",
       "       <defs>\n",
       "        <path d=\"M 9.8125 72.90625 \n",
       "L 19.671875 72.90625 \n",
       "L 19.671875 8.296875 \n",
       "L 55.171875 8.296875 \n",
       "L 55.171875 0 \n",
       "L 9.8125 0 \n",
       "z\n",
       "\" id=\"DejaVuSans-76\"/>\n",
       "        <path d=\"M 8.5 21.578125 \n",
       "L 8.5 54.6875 \n",
       "L 17.484375 54.6875 \n",
       "L 17.484375 21.921875 \n",
       "Q 17.484375 14.15625 20.5 10.265625 \n",
       "Q 23.53125 6.390625 29.59375 6.390625 \n",
       "Q 36.859375 6.390625 41.078125 11.03125 \n",
       "Q 45.3125 15.671875 45.3125 23.6875 \n",
       "L 45.3125 54.6875 \n",
       "L 54.296875 54.6875 \n",
       "L 54.296875 0 \n",
       "L 45.3125 0 \n",
       "L 45.3125 8.40625 \n",
       "Q 42.046875 3.421875 37.71875 1 \n",
       "Q 33.40625 -1.421875 27.6875 -1.421875 \n",
       "Q 18.265625 -1.421875 13.375 4.4375 \n",
       "Q 8.5 10.296875 8.5 21.578125 \n",
       "z\n",
       "M 31.109375 56 \n",
       "z\n",
       "\" id=\"DejaVuSans-117\"/>\n",
       "        <path d=\"M 54.890625 33.015625 \n",
       "L 54.890625 0 \n",
       "L 45.90625 0 \n",
       "L 45.90625 32.71875 \n",
       "Q 45.90625 40.484375 42.875 44.328125 \n",
       "Q 39.84375 48.1875 33.796875 48.1875 \n",
       "Q 26.515625 48.1875 22.3125 43.546875 \n",
       "Q 18.109375 38.921875 18.109375 30.90625 \n",
       "L 18.109375 0 \n",
       "L 9.078125 0 \n",
       "L 9.078125 54.6875 \n",
       "L 18.109375 54.6875 \n",
       "L 18.109375 46.1875 \n",
       "Q 21.34375 51.125 25.703125 53.5625 \n",
       "Q 30.078125 56 35.796875 56 \n",
       "Q 45.21875 56 50.046875 50.171875 \n",
       "Q 54.890625 44.34375 54.890625 33.015625 \n",
       "z\n",
       "\" id=\"DejaVuSans-110\"/>\n",
       "        <path d=\"M 45.40625 27.984375 \n",
       "Q 45.40625 37.75 41.375 43.109375 \n",
       "Q 37.359375 48.484375 30.078125 48.484375 \n",
       "Q 22.859375 48.484375 18.828125 43.109375 \n",
       "Q 14.796875 37.75 14.796875 27.984375 \n",
       "Q 14.796875 18.265625 18.828125 12.890625 \n",
       "Q 22.859375 7.515625 30.078125 7.515625 \n",
       "Q 37.359375 7.515625 41.375 12.890625 \n",
       "Q 45.40625 18.265625 45.40625 27.984375 \n",
       "z\n",
       "M 54.390625 6.78125 \n",
       "Q 54.390625 -7.171875 48.1875 -13.984375 \n",
       "Q 42 -20.796875 29.203125 -20.796875 \n",
       "Q 24.46875 -20.796875 20.265625 -20.09375 \n",
       "Q 16.0625 -19.390625 12.109375 -17.921875 \n",
       "L 12.109375 -9.1875 \n",
       "Q 16.0625 -11.328125 19.921875 -12.34375 \n",
       "Q 23.78125 -13.375 27.78125 -13.375 \n",
       "Q 36.625 -13.375 41.015625 -8.765625 \n",
       "Q 45.40625 -4.15625 45.40625 5.171875 \n",
       "L 45.40625 9.625 \n",
       "Q 42.625 4.78125 38.28125 2.390625 \n",
       "Q 33.9375 0 27.875 0 \n",
       "Q 17.828125 0 11.671875 7.65625 \n",
       "Q 5.515625 15.328125 5.515625 27.984375 \n",
       "Q 5.515625 40.671875 11.671875 48.328125 \n",
       "Q 17.828125 56 27.875 56 \n",
       "Q 33.9375 56 38.28125 53.609375 \n",
       "Q 42.625 51.21875 45.40625 46.390625 \n",
       "L 45.40625 54.6875 \n",
       "L 54.390625 54.6875 \n",
       "z\n",
       "\" id=\"DejaVuSans-103\"/>\n",
       "        <path d=\"M 50.984375 -16.609375 \n",
       "L 50.984375 -23.578125 \n",
       "L -0.984375 -23.578125 \n",
       "L -0.984375 -16.609375 \n",
       "z\n",
       "\" id=\"DejaVuSans-95\"/>\n",
       "        <path d=\"M 64.40625 67.28125 \n",
       "L 64.40625 56.890625 \n",
       "Q 59.421875 61.53125 53.78125 63.8125 \n",
       "Q 48.140625 66.109375 41.796875 66.109375 \n",
       "Q 29.296875 66.109375 22.65625 58.46875 \n",
       "Q 16.015625 50.828125 16.015625 36.375 \n",
       "Q 16.015625 21.96875 22.65625 14.328125 \n",
       "Q 29.296875 6.6875 41.796875 6.6875 \n",
       "Q 48.140625 6.6875 53.78125 8.984375 \n",
       "Q 59.421875 11.28125 64.40625 15.921875 \n",
       "L 64.40625 5.609375 \n",
       "Q 59.234375 2.09375 53.4375 0.328125 \n",
       "Q 47.65625 -1.421875 41.21875 -1.421875 \n",
       "Q 24.65625 -1.421875 15.125 8.703125 \n",
       "Q 5.609375 18.84375 5.609375 36.375 \n",
       "Q 5.609375 53.953125 15.125 64.078125 \n",
       "Q 24.65625 74.21875 41.21875 74.21875 \n",
       "Q 47.75 74.21875 53.53125 72.484375 \n",
       "Q 59.328125 70.75 64.40625 67.28125 \n",
       "z\n",
       "\" id=\"DejaVuSans-67\"/>\n",
       "        <path d=\"M 34.28125 27.484375 \n",
       "Q 23.390625 27.484375 19.1875 25 \n",
       "Q 14.984375 22.515625 14.984375 16.5 \n",
       "Q 14.984375 11.71875 18.140625 8.90625 \n",
       "Q 21.296875 6.109375 26.703125 6.109375 \n",
       "Q 34.1875 6.109375 38.703125 11.40625 \n",
       "Q 43.21875 16.703125 43.21875 25.484375 \n",
       "L 43.21875 27.484375 \n",
       "z\n",
       "M 52.203125 31.203125 \n",
       "L 52.203125 0 \n",
       "L 43.21875 0 \n",
       "L 43.21875 8.296875 \n",
       "Q 40.140625 3.328125 35.546875 0.953125 \n",
       "Q 30.953125 -1.421875 24.3125 -1.421875 \n",
       "Q 15.921875 -1.421875 10.953125 3.296875 \n",
       "Q 6 8.015625 6 15.921875 \n",
       "Q 6 25.140625 12.171875 29.828125 \n",
       "Q 18.359375 34.515625 30.609375 34.515625 \n",
       "L 43.21875 34.515625 \n",
       "L 43.21875 35.40625 \n",
       "Q 43.21875 41.609375 39.140625 45 \n",
       "Q 35.0625 48.390625 27.6875 48.390625 \n",
       "Q 23 48.390625 18.546875 47.265625 \n",
       "Q 14.109375 46.140625 10.015625 43.890625 \n",
       "L 10.015625 52.203125 \n",
       "Q 14.9375 54.109375 19.578125 55.046875 \n",
       "Q 24.21875 56 28.609375 56 \n",
       "Q 40.484375 56 46.34375 49.84375 \n",
       "Q 52.203125 43.703125 52.203125 31.203125 \n",
       "z\n",
       "\" id=\"DejaVuSans-97\"/>\n",
       "        <path d=\"M 48.78125 52.59375 \n",
       "L 48.78125 44.1875 \n",
       "Q 44.96875 46.296875 41.140625 47.34375 \n",
       "Q 37.3125 48.390625 33.40625 48.390625 \n",
       "Q 24.65625 48.390625 19.8125 42.84375 \n",
       "Q 14.984375 37.3125 14.984375 27.296875 \n",
       "Q 14.984375 17.28125 19.8125 11.734375 \n",
       "Q 24.65625 6.203125 33.40625 6.203125 \n",
       "Q 37.3125 6.203125 41.140625 7.25 \n",
       "Q 44.96875 8.296875 48.78125 10.40625 \n",
       "L 48.78125 2.09375 \n",
       "Q 45.015625 0.34375 40.984375 -0.53125 \n",
       "Q 36.96875 -1.421875 32.421875 -1.421875 \n",
       "Q 20.0625 -1.421875 12.78125 6.34375 \n",
       "Q 5.515625 14.109375 5.515625 27.296875 \n",
       "Q 5.515625 40.671875 12.859375 48.328125 \n",
       "Q 20.21875 56 33.015625 56 \n",
       "Q 37.15625 56 41.109375 55.140625 \n",
       "Q 45.0625 54.296875 48.78125 52.59375 \n",
       "z\n",
       "\" id=\"DejaVuSans-99\"/>\n",
       "        <path d=\"M 56.203125 29.59375 \n",
       "L 56.203125 25.203125 \n",
       "L 14.890625 25.203125 \n",
       "Q 15.484375 15.921875 20.484375 11.0625 \n",
       "Q 25.484375 6.203125 34.421875 6.203125 \n",
       "Q 39.59375 6.203125 44.453125 7.46875 \n",
       "Q 49.3125 8.734375 54.109375 11.28125 \n",
       "L 54.109375 2.78125 \n",
       "Q 49.265625 0.734375 44.1875 -0.34375 \n",
       "Q 39.109375 -1.421875 33.890625 -1.421875 \n",
       "Q 20.796875 -1.421875 13.15625 6.1875 \n",
       "Q 5.515625 13.8125 5.515625 26.8125 \n",
       "Q 5.515625 40.234375 12.765625 48.109375 \n",
       "Q 20.015625 56 32.328125 56 \n",
       "Q 43.359375 56 49.78125 48.890625 \n",
       "Q 56.203125 41.796875 56.203125 29.59375 \n",
       "z\n",
       "M 47.21875 32.234375 \n",
       "Q 47.125 39.59375 43.09375 43.984375 \n",
       "Q 39.0625 48.390625 32.421875 48.390625 \n",
       "Q 24.90625 48.390625 20.390625 44.140625 \n",
       "Q 15.875 39.890625 15.1875 32.171875 \n",
       "z\n",
       "\" id=\"DejaVuSans-101\"/>\n",
       "        <path d=\"M 41.109375 46.296875 \n",
       "Q 39.59375 47.171875 37.8125 47.578125 \n",
       "Q 36.03125 48 33.890625 48 \n",
       "Q 26.265625 48 22.1875 43.046875 \n",
       "Q 18.109375 38.09375 18.109375 28.8125 \n",
       "L 18.109375 0 \n",
       "L 9.078125 0 \n",
       "L 9.078125 54.6875 \n",
       "L 18.109375 54.6875 \n",
       "L 18.109375 46.1875 \n",
       "Q 20.953125 51.171875 25.484375 53.578125 \n",
       "Q 30.03125 56 36.53125 56 \n",
       "Q 37.453125 56 38.578125 55.875 \n",
       "Q 39.703125 55.765625 41.0625 55.515625 \n",
       "z\n",
       "\" id=\"DejaVuSans-114\"/>\n",
       "       </defs>\n",
       "       <use xlink:href=\"#DejaVuSans-76\"/>\n",
       "       <use x=\"53.962891\" xlink:href=\"#DejaVuSans-117\"/>\n",
       "       <use x=\"117.341797\" xlink:href=\"#DejaVuSans-110\"/>\n",
       "       <use x=\"180.720703\" xlink:href=\"#DejaVuSans-103\"/>\n",
       "       <use x=\"244.197266\" xlink:href=\"#DejaVuSans-95\"/>\n",
       "       <use x=\"294.197266\" xlink:href=\"#DejaVuSans-67\"/>\n",
       "       <use x=\"364.021484\" xlink:href=\"#DejaVuSans-97\"/>\n",
       "       <use x=\"425.300781\" xlink:href=\"#DejaVuSans-110\"/>\n",
       "       <use x=\"488.679688\" xlink:href=\"#DejaVuSans-99\"/>\n",
       "       <use x=\"543.660156\" xlink:href=\"#DejaVuSans-101\"/>\n",
       "       <use x=\"605.183594\" xlink:href=\"#DejaVuSans-114\"/>\n",
       "      </g>\n",
       "     </g>\n",
       "    </g>\n",
       "    <g id=\"xtick_2\">\n",
       "     <g id=\"line2d_2\">\n",
       "      <g>\n",
       "       <use style=\"stroke:#000000;stroke-width:0.8;\" x=\"257.766193\" xlink:href=\"#mb703f7e117\" y=\"225.707216\"/>\n",
       "      </g>\n",
       "     </g>\n",
       "     <g id=\"text_2\">\n",
       "      <!-- Thyroid_Cancer -->\n",
       "      <g transform=\"translate(218.913068 240.305653)scale(0.1 -0.1)\">\n",
       "       <defs>\n",
       "        <path d=\"M -0.296875 72.90625 \n",
       "L 61.375 72.90625 \n",
       "L 61.375 64.59375 \n",
       "L 35.5 64.59375 \n",
       "L 35.5 0 \n",
       "L 25.59375 0 \n",
       "L 25.59375 64.59375 \n",
       "L -0.296875 64.59375 \n",
       "z\n",
       "\" id=\"DejaVuSans-84\"/>\n",
       "        <path d=\"M 54.890625 33.015625 \n",
       "L 54.890625 0 \n",
       "L 45.90625 0 \n",
       "L 45.90625 32.71875 \n",
       "Q 45.90625 40.484375 42.875 44.328125 \n",
       "Q 39.84375 48.1875 33.796875 48.1875 \n",
       "Q 26.515625 48.1875 22.3125 43.546875 \n",
       "Q 18.109375 38.921875 18.109375 30.90625 \n",
       "L 18.109375 0 \n",
       "L 9.078125 0 \n",
       "L 9.078125 75.984375 \n",
       "L 18.109375 75.984375 \n",
       "L 18.109375 46.1875 \n",
       "Q 21.34375 51.125 25.703125 53.5625 \n",
       "Q 30.078125 56 35.796875 56 \n",
       "Q 45.21875 56 50.046875 50.171875 \n",
       "Q 54.890625 44.34375 54.890625 33.015625 \n",
       "z\n",
       "\" id=\"DejaVuSans-104\"/>\n",
       "        <path d=\"M 32.171875 -5.078125 \n",
       "Q 28.375 -14.84375 24.75 -17.8125 \n",
       "Q 21.140625 -20.796875 15.09375 -20.796875 \n",
       "L 7.90625 -20.796875 \n",
       "L 7.90625 -13.28125 \n",
       "L 13.1875 -13.28125 \n",
       "Q 16.890625 -13.28125 18.9375 -11.515625 \n",
       "Q 21 -9.765625 23.484375 -3.21875 \n",
       "L 25.09375 0.875 \n",
       "L 2.984375 54.6875 \n",
       "L 12.5 54.6875 \n",
       "L 29.59375 11.921875 \n",
       "L 46.6875 54.6875 \n",
       "L 56.203125 54.6875 \n",
       "z\n",
       "\" id=\"DejaVuSans-121\"/>\n",
       "        <path d=\"M 30.609375 48.390625 \n",
       "Q 23.390625 48.390625 19.1875 42.75 \n",
       "Q 14.984375 37.109375 14.984375 27.296875 \n",
       "Q 14.984375 17.484375 19.15625 11.84375 \n",
       "Q 23.34375 6.203125 30.609375 6.203125 \n",
       "Q 37.796875 6.203125 41.984375 11.859375 \n",
       "Q 46.1875 17.53125 46.1875 27.296875 \n",
       "Q 46.1875 37.015625 41.984375 42.703125 \n",
       "Q 37.796875 48.390625 30.609375 48.390625 \n",
       "z\n",
       "M 30.609375 56 \n",
       "Q 42.328125 56 49.015625 48.375 \n",
       "Q 55.71875 40.765625 55.71875 27.296875 \n",
       "Q 55.71875 13.875 49.015625 6.21875 \n",
       "Q 42.328125 -1.421875 30.609375 -1.421875 \n",
       "Q 18.84375 -1.421875 12.171875 6.21875 \n",
       "Q 5.515625 13.875 5.515625 27.296875 \n",
       "Q 5.515625 40.765625 12.171875 48.375 \n",
       "Q 18.84375 56 30.609375 56 \n",
       "z\n",
       "\" id=\"DejaVuSans-111\"/>\n",
       "        <path d=\"M 9.421875 54.6875 \n",
       "L 18.40625 54.6875 \n",
       "L 18.40625 0 \n",
       "L 9.421875 0 \n",
       "z\n",
       "M 9.421875 75.984375 \n",
       "L 18.40625 75.984375 \n",
       "L 18.40625 64.59375 \n",
       "L 9.421875 64.59375 \n",
       "z\n",
       "\" id=\"DejaVuSans-105\"/>\n",
       "        <path d=\"M 45.40625 46.390625 \n",
       "L 45.40625 75.984375 \n",
       "L 54.390625 75.984375 \n",
       "L 54.390625 0 \n",
       "L 45.40625 0 \n",
       "L 45.40625 8.203125 \n",
       "Q 42.578125 3.328125 38.25 0.953125 \n",
       "Q 33.9375 -1.421875 27.875 -1.421875 \n",
       "Q 17.96875 -1.421875 11.734375 6.484375 \n",
       "Q 5.515625 14.40625 5.515625 27.296875 \n",
       "Q 5.515625 40.1875 11.734375 48.09375 \n",
       "Q 17.96875 56 27.875 56 \n",
       "Q 33.9375 56 38.25 53.625 \n",
       "Q 42.578125 51.265625 45.40625 46.390625 \n",
       "z\n",
       "M 14.796875 27.296875 \n",
       "Q 14.796875 17.390625 18.875 11.75 \n",
       "Q 22.953125 6.109375 30.078125 6.109375 \n",
       "Q 37.203125 6.109375 41.296875 11.75 \n",
       "Q 45.40625 17.390625 45.40625 27.296875 \n",
       "Q 45.40625 37.203125 41.296875 42.84375 \n",
       "Q 37.203125 48.484375 30.078125 48.484375 \n",
       "Q 22.953125 48.484375 18.875 42.84375 \n",
       "Q 14.796875 37.203125 14.796875 27.296875 \n",
       "z\n",
       "\" id=\"DejaVuSans-100\"/>\n",
       "       </defs>\n",
       "       <use xlink:href=\"#DejaVuSans-84\"/>\n",
       "       <use x=\"61.083984\" xlink:href=\"#DejaVuSans-104\"/>\n",
       "       <use x=\"124.462891\" xlink:href=\"#DejaVuSans-121\"/>\n",
       "       <use x=\"183.642578\" xlink:href=\"#DejaVuSans-114\"/>\n",
       "       <use x=\"222.505859\" xlink:href=\"#DejaVuSans-111\"/>\n",
       "       <use x=\"283.6875\" xlink:href=\"#DejaVuSans-105\"/>\n",
       "       <use x=\"311.470703\" xlink:href=\"#DejaVuSans-100\"/>\n",
       "       <use x=\"374.947266\" xlink:href=\"#DejaVuSans-95\"/>\n",
       "       <use x=\"424.947266\" xlink:href=\"#DejaVuSans-67\"/>\n",
       "       <use x=\"494.771484\" xlink:href=\"#DejaVuSans-97\"/>\n",
       "       <use x=\"556.050781\" xlink:href=\"#DejaVuSans-110\"/>\n",
       "       <use x=\"619.429688\" xlink:href=\"#DejaVuSans-99\"/>\n",
       "       <use x=\"674.410156\" xlink:href=\"#DejaVuSans-101\"/>\n",
       "       <use x=\"735.933594\" xlink:href=\"#DejaVuSans-114\"/>\n",
       "      </g>\n",
       "     </g>\n",
       "    </g>\n",
       "    <g id=\"text_3\">\n",
       "     <!-- Predicted label -->\n",
       "     <g transform=\"translate(169.886577 254.261903)scale(0.1 -0.1)\">\n",
       "      <defs>\n",
       "       <path d=\"M 19.671875 64.796875 \n",
       "L 19.671875 37.40625 \n",
       "L 32.078125 37.40625 \n",
       "Q 38.96875 37.40625 42.71875 40.96875 \n",
       "Q 46.484375 44.53125 46.484375 51.125 \n",
       "Q 46.484375 57.671875 42.71875 61.234375 \n",
       "Q 38.96875 64.796875 32.078125 64.796875 \n",
       "z\n",
       "M 9.8125 72.90625 \n",
       "L 32.078125 72.90625 \n",
       "Q 44.34375 72.90625 50.609375 67.359375 \n",
       "Q 56.890625 61.8125 56.890625 51.125 \n",
       "Q 56.890625 40.328125 50.609375 34.8125 \n",
       "Q 44.34375 29.296875 32.078125 29.296875 \n",
       "L 19.671875 29.296875 \n",
       "L 19.671875 0 \n",
       "L 9.8125 0 \n",
       "z\n",
       "\" id=\"DejaVuSans-80\"/>\n",
       "       <path d=\"M 18.3125 70.21875 \n",
       "L 18.3125 54.6875 \n",
       "L 36.8125 54.6875 \n",
       "L 36.8125 47.703125 \n",
       "L 18.3125 47.703125 \n",
       "L 18.3125 18.015625 \n",
       "Q 18.3125 11.328125 20.140625 9.421875 \n",
       "Q 21.96875 7.515625 27.59375 7.515625 \n",
       "L 36.8125 7.515625 \n",
       "L 36.8125 0 \n",
       "L 27.59375 0 \n",
       "Q 17.1875 0 13.234375 3.875 \n",
       "Q 9.28125 7.765625 9.28125 18.015625 \n",
       "L 9.28125 47.703125 \n",
       "L 2.6875 47.703125 \n",
       "L 2.6875 54.6875 \n",
       "L 9.28125 54.6875 \n",
       "L 9.28125 70.21875 \n",
       "z\n",
       "\" id=\"DejaVuSans-116\"/>\n",
       "       <path id=\"DejaVuSans-32\"/>\n",
       "       <path d=\"M 9.421875 75.984375 \n",
       "L 18.40625 75.984375 \n",
       "L 18.40625 0 \n",
       "L 9.421875 0 \n",
       "z\n",
       "\" id=\"DejaVuSans-108\"/>\n",
       "       <path d=\"M 48.6875 27.296875 \n",
       "Q 48.6875 37.203125 44.609375 42.84375 \n",
       "Q 40.53125 48.484375 33.40625 48.484375 \n",
       "Q 26.265625 48.484375 22.1875 42.84375 \n",
       "Q 18.109375 37.203125 18.109375 27.296875 \n",
       "Q 18.109375 17.390625 22.1875 11.75 \n",
       "Q 26.265625 6.109375 33.40625 6.109375 \n",
       "Q 40.53125 6.109375 44.609375 11.75 \n",
       "Q 48.6875 17.390625 48.6875 27.296875 \n",
       "z\n",
       "M 18.109375 46.390625 \n",
       "Q 20.953125 51.265625 25.265625 53.625 \n",
       "Q 29.59375 56 35.59375 56 \n",
       "Q 45.5625 56 51.78125 48.09375 \n",
       "Q 58.015625 40.1875 58.015625 27.296875 \n",
       "Q 58.015625 14.40625 51.78125 6.484375 \n",
       "Q 45.5625 -1.421875 35.59375 -1.421875 \n",
       "Q 29.59375 -1.421875 25.265625 0.953125 \n",
       "Q 20.953125 3.328125 18.109375 8.203125 \n",
       "L 18.109375 0 \n",
       "L 9.078125 0 \n",
       "L 9.078125 75.984375 \n",
       "L 18.109375 75.984375 \n",
       "z\n",
       "\" id=\"DejaVuSans-98\"/>\n",
       "      </defs>\n",
       "      <use xlink:href=\"#DejaVuSans-80\"/>\n",
       "      <use x=\"58.552734\" xlink:href=\"#DejaVuSans-114\"/>\n",
       "      <use x=\"97.416016\" xlink:href=\"#DejaVuSans-101\"/>\n",
       "      <use x=\"158.939453\" xlink:href=\"#DejaVuSans-100\"/>\n",
       "      <use x=\"222.416016\" xlink:href=\"#DejaVuSans-105\"/>\n",
       "      <use x=\"250.199219\" xlink:href=\"#DejaVuSans-99\"/>\n",
       "      <use x=\"305.179688\" xlink:href=\"#DejaVuSans-116\"/>\n",
       "      <use x=\"344.388672\" xlink:href=\"#DejaVuSans-101\"/>\n",
       "      <use x=\"405.912109\" xlink:href=\"#DejaVuSans-100\"/>\n",
       "      <use x=\"469.388672\" xlink:href=\"#DejaVuSans-32\"/>\n",
       "      <use x=\"501.175781\" xlink:href=\"#DejaVuSans-108\"/>\n",
       "      <use x=\"528.958984\" xlink:href=\"#DejaVuSans-97\"/>\n",
       "      <use x=\"590.238281\" xlink:href=\"#DejaVuSans-98\"/>\n",
       "      <use x=\"653.714844\" xlink:href=\"#DejaVuSans-101\"/>\n",
       "      <use x=\"715.238281\" xlink:href=\"#DejaVuSans-108\"/>\n",
       "     </g>\n",
       "    </g>\n",
       "   </g>\n",
       "   <g id=\"matplotlib.axis_2\">\n",
       "    <g id=\"ytick_1\">\n",
       "     <g id=\"line2d_3\">\n",
       "      <defs>\n",
       "       <path d=\"M 0 0 \n",
       "L -3.5 0 \n",
       "\" id=\"m60f16bc87b\" style=\"stroke:#000000;stroke-width:0.8;\"/>\n",
       "      </defs>\n",
       "      <g>\n",
       "       <use style=\"stroke:#000000;stroke-width:0.8;\" x=\"105.584375\" xlink:href=\"#m60f16bc87b\" y=\"73.525398\"/>\n",
       "      </g>\n",
       "     </g>\n",
       "     <g id=\"text_4\">\n",
       "      <!-- Lung_Cancer -->\n",
       "      <g transform=\"translate(33.953125 77.236335)scale(0.1 -0.1)\">\n",
       "       <use xlink:href=\"#DejaVuSans-76\"/>\n",
       "       <use x=\"53.962891\" xlink:href=\"#DejaVuSans-117\"/>\n",
       "       <use x=\"117.341797\" xlink:href=\"#DejaVuSans-110\"/>\n",
       "       <use x=\"180.720703\" xlink:href=\"#DejaVuSans-103\"/>\n",
       "       <use x=\"244.197266\" xlink:href=\"#DejaVuSans-95\"/>\n",
       "       <use x=\"294.197266\" xlink:href=\"#DejaVuSans-67\"/>\n",
       "       <use x=\"364.021484\" xlink:href=\"#DejaVuSans-97\"/>\n",
       "       <use x=\"425.300781\" xlink:href=\"#DejaVuSans-110\"/>\n",
       "       <use x=\"488.679688\" xlink:href=\"#DejaVuSans-99\"/>\n",
       "       <use x=\"543.660156\" xlink:href=\"#DejaVuSans-101\"/>\n",
       "       <use x=\"605.183594\" xlink:href=\"#DejaVuSans-114\"/>\n",
       "      </g>\n",
       "     </g>\n",
       "    </g>\n",
       "    <g id=\"ytick_2\">\n",
       "     <g id=\"line2d_4\">\n",
       "      <g>\n",
       "       <use style=\"stroke:#000000;stroke-width:0.8;\" x=\"105.584375\" xlink:href=\"#m60f16bc87b\" y=\"174.979943\"/>\n",
       "      </g>\n",
       "     </g>\n",
       "     <g id=\"text_5\">\n",
       "      <!-- Thyroid_Cancer -->\n",
       "      <g transform=\"translate(20.878125 178.779162)scale(0.1 -0.1)\">\n",
       "       <use xlink:href=\"#DejaVuSans-84\"/>\n",
       "       <use x=\"61.083984\" xlink:href=\"#DejaVuSans-104\"/>\n",
       "       <use x=\"124.462891\" xlink:href=\"#DejaVuSans-121\"/>\n",
       "       <use x=\"183.642578\" xlink:href=\"#DejaVuSans-114\"/>\n",
       "       <use x=\"222.505859\" xlink:href=\"#DejaVuSans-111\"/>\n",
       "       <use x=\"283.6875\" xlink:href=\"#DejaVuSans-105\"/>\n",
       "       <use x=\"311.470703\" xlink:href=\"#DejaVuSans-100\"/>\n",
       "       <use x=\"374.947266\" xlink:href=\"#DejaVuSans-95\"/>\n",
       "       <use x=\"424.947266\" xlink:href=\"#DejaVuSans-67\"/>\n",
       "       <use x=\"494.771484\" xlink:href=\"#DejaVuSans-97\"/>\n",
       "       <use x=\"556.050781\" xlink:href=\"#DejaVuSans-110\"/>\n",
       "       <use x=\"619.429688\" xlink:href=\"#DejaVuSans-99\"/>\n",
       "       <use x=\"674.410156\" xlink:href=\"#DejaVuSans-101\"/>\n",
       "       <use x=\"735.933594\" xlink:href=\"#DejaVuSans-114\"/>\n",
       "      </g>\n",
       "     </g>\n",
       "    </g>\n",
       "    <g id=\"text_6\">\n",
       "     <!-- True label -->\n",
       "     <g transform=\"translate(14.798437 148.551889)rotate(-90)scale(0.1 -0.1)\">\n",
       "      <use xlink:href=\"#DejaVuSans-84\"/>\n",
       "      <use x=\"46.333984\" xlink:href=\"#DejaVuSans-114\"/>\n",
       "      <use x=\"87.447266\" xlink:href=\"#DejaVuSans-117\"/>\n",
       "      <use x=\"150.826172\" xlink:href=\"#DejaVuSans-101\"/>\n",
       "      <use x=\"212.349609\" xlink:href=\"#DejaVuSans-32\"/>\n",
       "      <use x=\"244.136719\" xlink:href=\"#DejaVuSans-108\"/>\n",
       "      <use x=\"271.919922\" xlink:href=\"#DejaVuSans-97\"/>\n",
       "      <use x=\"333.199219\" xlink:href=\"#DejaVuSans-98\"/>\n",
       "      <use x=\"396.675781\" xlink:href=\"#DejaVuSans-101\"/>\n",
       "      <use x=\"458.199219\" xlink:href=\"#DejaVuSans-108\"/>\n",
       "     </g>\n",
       "    </g>\n",
       "   </g>\n",
       "   <g id=\"patch_3\">\n",
       "    <path d=\"M 105.584375 225.707216 \n",
       "L 105.584375 22.798125 \n",
       "\" style=\"fill:none;stroke:#000000;stroke-linecap:square;stroke-linejoin:miter;stroke-width:0.8;\"/>\n",
       "   </g>\n",
       "   <g id=\"patch_4\">\n",
       "    <path d=\"M 308.493466 225.707216 \n",
       "L 308.493466 22.798125 \n",
       "\" style=\"fill:none;stroke:#000000;stroke-linecap:square;stroke-linejoin:miter;stroke-width:0.8;\"/>\n",
       "   </g>\n",
       "   <g id=\"patch_5\">\n",
       "    <path d=\"M 105.584375 225.707216 \n",
       "L 308.493466 225.707216 \n",
       "\" style=\"fill:none;stroke:#000000;stroke-linecap:square;stroke-linejoin:miter;stroke-width:0.8;\"/>\n",
       "   </g>\n",
       "   <g id=\"patch_6\">\n",
       "    <path d=\"M 105.584375 22.798125 \n",
       "L 308.493466 22.798125 \n",
       "\" style=\"fill:none;stroke:#000000;stroke-linecap:square;stroke-linejoin:miter;stroke-width:0.8;\"/>\n",
       "   </g>\n",
       "   <g id=\"text_7\">\n",
       "    <!-- 0.93 -->\n",
       "    <g style=\"fill:#f7fbff;\" transform=\"translate(145.178835 76.284773)scale(0.1 -0.1)\">\n",
       "     <defs>\n",
       "      <path d=\"M 31.78125 66.40625 \n",
       "Q 24.171875 66.40625 20.328125 58.90625 \n",
       "Q 16.5 51.421875 16.5 36.375 \n",
       "Q 16.5 21.390625 20.328125 13.890625 \n",
       "Q 24.171875 6.390625 31.78125 6.390625 \n",
       "Q 39.453125 6.390625 43.28125 13.890625 \n",
       "Q 47.125 21.390625 47.125 36.375 \n",
       "Q 47.125 51.421875 43.28125 58.90625 \n",
       "Q 39.453125 66.40625 31.78125 66.40625 \n",
       "z\n",
       "M 31.78125 74.21875 \n",
       "Q 44.046875 74.21875 50.515625 64.515625 \n",
       "Q 56.984375 54.828125 56.984375 36.375 \n",
       "Q 56.984375 17.96875 50.515625 8.265625 \n",
       "Q 44.046875 -1.421875 31.78125 -1.421875 \n",
       "Q 19.53125 -1.421875 13.0625 8.265625 \n",
       "Q 6.59375 17.96875 6.59375 36.375 \n",
       "Q 6.59375 54.828125 13.0625 64.515625 \n",
       "Q 19.53125 74.21875 31.78125 74.21875 \n",
       "z\n",
       "\" id=\"DejaVuSans-48\"/>\n",
       "      <path d=\"M 10.6875 12.40625 \n",
       "L 21 12.40625 \n",
       "L 21 0 \n",
       "L 10.6875 0 \n",
       "z\n",
       "\" id=\"DejaVuSans-46\"/>\n",
       "      <path d=\"M 10.984375 1.515625 \n",
       "L 10.984375 10.5 \n",
       "Q 14.703125 8.734375 18.5 7.8125 \n",
       "Q 22.3125 6.890625 25.984375 6.890625 \n",
       "Q 35.75 6.890625 40.890625 13.453125 \n",
       "Q 46.046875 20.015625 46.78125 33.40625 \n",
       "Q 43.953125 29.203125 39.59375 26.953125 \n",
       "Q 35.25 24.703125 29.984375 24.703125 \n",
       "Q 19.046875 24.703125 12.671875 31.3125 \n",
       "Q 6.296875 37.9375 6.296875 49.421875 \n",
       "Q 6.296875 60.640625 12.9375 67.421875 \n",
       "Q 19.578125 74.21875 30.609375 74.21875 \n",
       "Q 43.265625 74.21875 49.921875 64.515625 \n",
       "Q 56.59375 54.828125 56.59375 36.375 \n",
       "Q 56.59375 19.140625 48.40625 8.859375 \n",
       "Q 40.234375 -1.421875 26.421875 -1.421875 \n",
       "Q 22.703125 -1.421875 18.890625 -0.6875 \n",
       "Q 15.09375 0.046875 10.984375 1.515625 \n",
       "z\n",
       "M 30.609375 32.421875 \n",
       "Q 37.25 32.421875 41.125 36.953125 \n",
       "Q 45.015625 41.5 45.015625 49.421875 \n",
       "Q 45.015625 57.28125 41.125 61.84375 \n",
       "Q 37.25 66.40625 30.609375 66.40625 \n",
       "Q 23.96875 66.40625 20.09375 61.84375 \n",
       "Q 16.21875 57.28125 16.21875 49.421875 \n",
       "Q 16.21875 41.5 20.09375 36.953125 \n",
       "Q 23.96875 32.421875 30.609375 32.421875 \n",
       "z\n",
       "\" id=\"DejaVuSans-57\"/>\n",
       "      <path d=\"M 40.578125 39.3125 \n",
       "Q 47.65625 37.796875 51.625 33 \n",
       "Q 55.609375 28.21875 55.609375 21.1875 \n",
       "Q 55.609375 10.40625 48.1875 4.484375 \n",
       "Q 40.765625 -1.421875 27.09375 -1.421875 \n",
       "Q 22.515625 -1.421875 17.65625 -0.515625 \n",
       "Q 12.796875 0.390625 7.625 2.203125 \n",
       "L 7.625 11.71875 \n",
       "Q 11.71875 9.328125 16.59375 8.109375 \n",
       "Q 21.484375 6.890625 26.8125 6.890625 \n",
       "Q 36.078125 6.890625 40.9375 10.546875 \n",
       "Q 45.796875 14.203125 45.796875 21.1875 \n",
       "Q 45.796875 27.640625 41.28125 31.265625 \n",
       "Q 36.765625 34.90625 28.71875 34.90625 \n",
       "L 20.21875 34.90625 \n",
       "L 20.21875 43.015625 \n",
       "L 29.109375 43.015625 \n",
       "Q 36.375 43.015625 40.234375 45.921875 \n",
       "Q 44.09375 48.828125 44.09375 54.296875 \n",
       "Q 44.09375 59.90625 40.109375 62.90625 \n",
       "Q 36.140625 65.921875 28.71875 65.921875 \n",
       "Q 24.65625 65.921875 20.015625 65.03125 \n",
       "Q 15.375 64.15625 9.8125 62.3125 \n",
       "L 9.8125 71.09375 \n",
       "Q 15.4375 72.65625 20.34375 73.4375 \n",
       "Q 25.25 74.21875 29.59375 74.21875 \n",
       "Q 40.828125 74.21875 47.359375 69.109375 \n",
       "Q 53.90625 64.015625 53.90625 55.328125 \n",
       "Q 53.90625 49.265625 50.4375 45.09375 \n",
       "Q 46.96875 40.921875 40.578125 39.3125 \n",
       "z\n",
       "\" id=\"DejaVuSans-51\"/>\n",
       "     </defs>\n",
       "     <use xlink:href=\"#DejaVuSans-48\"/>\n",
       "     <use x=\"63.623047\" xlink:href=\"#DejaVuSans-46\"/>\n",
       "     <use x=\"95.410156\" xlink:href=\"#DejaVuSans-57\"/>\n",
       "     <use x=\"159.033203\" xlink:href=\"#DejaVuSans-51\"/>\n",
       "    </g>\n",
       "   </g>\n",
       "   <g id=\"text_8\">\n",
       "    <!-- 0.067 -->\n",
       "    <g style=\"fill:#08306b;\" transform=\"translate(243.452131 76.284773)scale(0.1 -0.1)\">\n",
       "     <defs>\n",
       "      <path d=\"M 33.015625 40.375 \n",
       "Q 26.375 40.375 22.484375 35.828125 \n",
       "Q 18.609375 31.296875 18.609375 23.390625 \n",
       "Q 18.609375 15.53125 22.484375 10.953125 \n",
       "Q 26.375 6.390625 33.015625 6.390625 \n",
       "Q 39.65625 6.390625 43.53125 10.953125 \n",
       "Q 47.40625 15.53125 47.40625 23.390625 \n",
       "Q 47.40625 31.296875 43.53125 35.828125 \n",
       "Q 39.65625 40.375 33.015625 40.375 \n",
       "z\n",
       "M 52.59375 71.296875 \n",
       "L 52.59375 62.3125 \n",
       "Q 48.875 64.0625 45.09375 64.984375 \n",
       "Q 41.3125 65.921875 37.59375 65.921875 \n",
       "Q 27.828125 65.921875 22.671875 59.328125 \n",
       "Q 17.53125 52.734375 16.796875 39.40625 \n",
       "Q 19.671875 43.65625 24.015625 45.921875 \n",
       "Q 28.375 48.1875 33.59375 48.1875 \n",
       "Q 44.578125 48.1875 50.953125 41.515625 \n",
       "Q 57.328125 34.859375 57.328125 23.390625 \n",
       "Q 57.328125 12.15625 50.6875 5.359375 \n",
       "Q 44.046875 -1.421875 33.015625 -1.421875 \n",
       "Q 20.359375 -1.421875 13.671875 8.265625 \n",
       "Q 6.984375 17.96875 6.984375 36.375 \n",
       "Q 6.984375 53.65625 15.1875 63.9375 \n",
       "Q 23.390625 74.21875 37.203125 74.21875 \n",
       "Q 40.921875 74.21875 44.703125 73.484375 \n",
       "Q 48.484375 72.75 52.59375 71.296875 \n",
       "z\n",
       "\" id=\"DejaVuSans-54\"/>\n",
       "      <path d=\"M 8.203125 72.90625 \n",
       "L 55.078125 72.90625 \n",
       "L 55.078125 68.703125 \n",
       "L 28.609375 0 \n",
       "L 18.3125 0 \n",
       "L 43.21875 64.59375 \n",
       "L 8.203125 64.59375 \n",
       "z\n",
       "\" id=\"DejaVuSans-55\"/>\n",
       "     </defs>\n",
       "     <use xlink:href=\"#DejaVuSans-48\"/>\n",
       "     <use x=\"63.623047\" xlink:href=\"#DejaVuSans-46\"/>\n",
       "     <use x=\"95.410156\" xlink:href=\"#DejaVuSans-48\"/>\n",
       "     <use x=\"159.033203\" xlink:href=\"#DejaVuSans-54\"/>\n",
       "     <use x=\"222.65625\" xlink:href=\"#DejaVuSans-55\"/>\n",
       "    </g>\n",
       "   </g>\n",
       "   <g id=\"text_9\">\n",
       "    <!-- 0 -->\n",
       "    <g style=\"fill:#08306b;\" transform=\"translate(153.130398 177.739318)scale(0.1 -0.1)\">\n",
       "     <use xlink:href=\"#DejaVuSans-48\"/>\n",
       "    </g>\n",
       "   </g>\n",
       "   <g id=\"text_10\">\n",
       "    <!-- 1 -->\n",
       "    <g style=\"fill:#f7fbff;\" transform=\"translate(254.584943 177.739318)scale(0.1 -0.1)\">\n",
       "     <defs>\n",
       "      <path d=\"M 12.40625 8.296875 \n",
       "L 28.515625 8.296875 \n",
       "L 28.515625 63.921875 \n",
       "L 10.984375 60.40625 \n",
       "L 10.984375 69.390625 \n",
       "L 28.421875 72.90625 \n",
       "L 38.28125 72.90625 \n",
       "L 38.28125 8.296875 \n",
       "L 54.390625 8.296875 \n",
       "L 54.390625 0 \n",
       "L 12.40625 0 \n",
       "z\n",
       "\" id=\"DejaVuSans-49\"/>\n",
       "     </defs>\n",
       "     <use xlink:href=\"#DejaVuSans-49\"/>\n",
       "    </g>\n",
       "   </g>\n",
       "   <g id=\"text_11\">\n",
       "    <!-- Naive Bayes -->\n",
       "    <g transform=\"translate(169.894233 16.798125)scale(0.12 -0.12)\">\n",
       "     <defs>\n",
       "      <path d=\"M 9.8125 72.90625 \n",
       "L 23.09375 72.90625 \n",
       "L 55.421875 11.921875 \n",
       "L 55.421875 72.90625 \n",
       "L 64.984375 72.90625 \n",
       "L 64.984375 0 \n",
       "L 51.703125 0 \n",
       "L 19.390625 60.984375 \n",
       "L 19.390625 0 \n",
       "L 9.8125 0 \n",
       "z\n",
       "\" id=\"DejaVuSans-78\"/>\n",
       "      <path d=\"M 2.984375 54.6875 \n",
       "L 12.5 54.6875 \n",
       "L 29.59375 8.796875 \n",
       "L 46.6875 54.6875 \n",
       "L 56.203125 54.6875 \n",
       "L 35.6875 0 \n",
       "L 23.484375 0 \n",
       "z\n",
       "\" id=\"DejaVuSans-118\"/>\n",
       "      <path d=\"M 19.671875 34.8125 \n",
       "L 19.671875 8.109375 \n",
       "L 35.5 8.109375 \n",
       "Q 43.453125 8.109375 47.28125 11.40625 \n",
       "Q 51.125 14.703125 51.125 21.484375 \n",
       "Q 51.125 28.328125 47.28125 31.5625 \n",
       "Q 43.453125 34.8125 35.5 34.8125 \n",
       "z\n",
       "M 19.671875 64.796875 \n",
       "L 19.671875 42.828125 \n",
       "L 34.28125 42.828125 \n",
       "Q 41.5 42.828125 45.03125 45.53125 \n",
       "Q 48.578125 48.25 48.578125 53.8125 \n",
       "Q 48.578125 59.328125 45.03125 62.0625 \n",
       "Q 41.5 64.796875 34.28125 64.796875 \n",
       "z\n",
       "M 9.8125 72.90625 \n",
       "L 35.015625 72.90625 \n",
       "Q 46.296875 72.90625 52.390625 68.21875 \n",
       "Q 58.5 63.53125 58.5 54.890625 \n",
       "Q 58.5 48.1875 55.375 44.234375 \n",
       "Q 52.25 40.28125 46.1875 39.3125 \n",
       "Q 53.46875 37.75 57.5 32.78125 \n",
       "Q 61.53125 27.828125 61.53125 20.40625 \n",
       "Q 61.53125 10.640625 54.890625 5.3125 \n",
       "Q 48.25 0 35.984375 0 \n",
       "L 9.8125 0 \n",
       "z\n",
       "\" id=\"DejaVuSans-66\"/>\n",
       "      <path d=\"M 44.28125 53.078125 \n",
       "L 44.28125 44.578125 \n",
       "Q 40.484375 46.53125 36.375 47.5 \n",
       "Q 32.28125 48.484375 27.875 48.484375 \n",
       "Q 21.1875 48.484375 17.84375 46.4375 \n",
       "Q 14.5 44.390625 14.5 40.28125 \n",
       "Q 14.5 37.15625 16.890625 35.375 \n",
       "Q 19.28125 33.59375 26.515625 31.984375 \n",
       "L 29.59375 31.296875 \n",
       "Q 39.15625 29.25 43.1875 25.515625 \n",
       "Q 47.21875 21.78125 47.21875 15.09375 \n",
       "Q 47.21875 7.46875 41.1875 3.015625 \n",
       "Q 35.15625 -1.421875 24.609375 -1.421875 \n",
       "Q 20.21875 -1.421875 15.453125 -0.5625 \n",
       "Q 10.6875 0.296875 5.421875 2 \n",
       "L 5.421875 11.28125 \n",
       "Q 10.40625 8.6875 15.234375 7.390625 \n",
       "Q 20.0625 6.109375 24.8125 6.109375 \n",
       "Q 31.15625 6.109375 34.5625 8.28125 \n",
       "Q 37.984375 10.453125 37.984375 14.40625 \n",
       "Q 37.984375 18.0625 35.515625 20.015625 \n",
       "Q 33.0625 21.96875 24.703125 23.78125 \n",
       "L 21.578125 24.515625 \n",
       "Q 13.234375 26.265625 9.515625 29.90625 \n",
       "Q 5.8125 33.546875 5.8125 39.890625 \n",
       "Q 5.8125 47.609375 11.28125 51.796875 \n",
       "Q 16.75 56 26.8125 56 \n",
       "Q 31.78125 56 36.171875 55.265625 \n",
       "Q 40.578125 54.546875 44.28125 53.078125 \n",
       "z\n",
       "\" id=\"DejaVuSans-115\"/>\n",
       "     </defs>\n",
       "     <use xlink:href=\"#DejaVuSans-78\"/>\n",
       "     <use x=\"74.804688\" xlink:href=\"#DejaVuSans-97\"/>\n",
       "     <use x=\"136.083984\" xlink:href=\"#DejaVuSans-105\"/>\n",
       "     <use x=\"163.867188\" xlink:href=\"#DejaVuSans-118\"/>\n",
       "     <use x=\"223.046875\" xlink:href=\"#DejaVuSans-101\"/>\n",
       "     <use x=\"284.570312\" xlink:href=\"#DejaVuSans-32\"/>\n",
       "     <use x=\"316.357422\" xlink:href=\"#DejaVuSans-66\"/>\n",
       "     <use x=\"384.960938\" xlink:href=\"#DejaVuSans-97\"/>\n",
       "     <use x=\"446.240234\" xlink:href=\"#DejaVuSans-121\"/>\n",
       "     <use x=\"505.419922\" xlink:href=\"#DejaVuSans-101\"/>\n",
       "     <use x=\"566.943359\" xlink:href=\"#DejaVuSans-115\"/>\n",
       "    </g>\n",
       "   </g>\n",
       "  </g>\n",
       "  <g id=\"axes_2\">\n",
       "   <g id=\"patch_7\">\n",
       "    <path d=\"M 409.948011 225.707216 \n",
       "L 612.857102 225.707216 \n",
       "L 612.857102 22.798125 \n",
       "L 409.948011 22.798125 \n",
       "z\n",
       "\" style=\"fill:#ffffff;\"/>\n",
       "   </g>\n",
       "   <g clip-path=\"url(#p1df93f9291)\">\n",
       "    <image height=\"203\" id=\"imagea2bec7c919\" transform=\"scale(1 -1)translate(0 -203)\" width=\"203\" x=\"409.948011\" xlink:href=\"data:image/png;base64,\n",
       "iVBORw0KGgoAAAANSUhEUgAAAMsAAADLCAYAAADA+2czAAACV0lEQVR4nO3VoQ3CUABFUYqqZZhuhWINxuo4DIBAfHSTius+JOdM8MzNW96fMS78lNt2nz2BE9fZA+BfiAUisUAkFojEApFYIBILRGKBSCwQiQUisUAkFojEApFYIBILRGKBSCwQiQUisUAkFojEApFYIBILRGKBSCwQiQUisUAkFojEApFYIBILRGKBSCwQiQUisUAkFojEApFYIBILRGKBSCwQiQUisUAkFojEApFYIBILRGKBSCwQiQUisUAkFojEApFYIBILRGKBSCwQiQUisUAkFojEApFYIBILRGKBSCwQiQUisUAkFojEApFYIBILRGKBSCwQiQUisUAkFojEApFYIBILRGKBSCwQiQUisUAkFojEApFYIBILRGKBSCwQiQUisUAkFojEApFYIBILRGKBSCwQiQUisUAkFojEAtGybo8xewRHr/05ewInPAtEYoFILBCJBSKxQCQWiMQCkVggEgtEYoFILBCJBSKxQCQWiMQCkVggEgtEYoFILBCJBSKxQCQWiMQCkVggEgtEYoFILBCJBSKxQCQWiMQCkVggEgtEYoFILBCJBSKxQCQWiMQCkVggEgtEYoFILBCJBSKxQCQWiMQCkVggEgtEYoFILBCJBSKxQCQWiMQCkVggEgtEYoFILBCJBSKxQCQWiMQCkVggEgtEYoFILBCJBSKxQCQWiMQCkVggEgtEYoFILBCJBSKxQCQWiMQCkVggEgtEYoFILBCJBSKxQCQWiMQCkVggEgtEYoFILBCJBSKxQCQWiMQCkVggEgtEYoFILBCJBaIvRGYKHmYw/mAAAAAASUVORK5CYII=\" y=\"-22.707216\"/>\n",
       "   </g>\n",
       "   <g id=\"matplotlib.axis_3\">\n",
       "    <g id=\"xtick_3\">\n",
       "     <g id=\"line2d_5\">\n",
       "      <g>\n",
       "       <use style=\"stroke:#000000;stroke-width:0.8;\" x=\"460.675284\" xlink:href=\"#mb703f7e117\" y=\"225.707216\"/>\n",
       "      </g>\n",
       "     </g>\n",
       "     <g id=\"text_12\">\n",
       "      <!-- Lung_Cancer -->\n",
       "      <g transform=\"translate(428.359659 240.129091)scale(0.1 -0.1)\">\n",
       "       <use xlink:href=\"#DejaVuSans-76\"/>\n",
       "       <use x=\"53.962891\" xlink:href=\"#DejaVuSans-117\"/>\n",
       "       <use x=\"117.341797\" xlink:href=\"#DejaVuSans-110\"/>\n",
       "       <use x=\"180.720703\" xlink:href=\"#DejaVuSans-103\"/>\n",
       "       <use x=\"244.197266\" xlink:href=\"#DejaVuSans-95\"/>\n",
       "       <use x=\"294.197266\" xlink:href=\"#DejaVuSans-67\"/>\n",
       "       <use x=\"364.021484\" xlink:href=\"#DejaVuSans-97\"/>\n",
       "       <use x=\"425.300781\" xlink:href=\"#DejaVuSans-110\"/>\n",
       "       <use x=\"488.679688\" xlink:href=\"#DejaVuSans-99\"/>\n",
       "       <use x=\"543.660156\" xlink:href=\"#DejaVuSans-101\"/>\n",
       "       <use x=\"605.183594\" xlink:href=\"#DejaVuSans-114\"/>\n",
       "      </g>\n",
       "     </g>\n",
       "    </g>\n",
       "    <g id=\"xtick_4\">\n",
       "     <g id=\"line2d_6\">\n",
       "      <g>\n",
       "       <use style=\"stroke:#000000;stroke-width:0.8;\" x=\"562.12983\" xlink:href=\"#mb703f7e117\" y=\"225.707216\"/>\n",
       "      </g>\n",
       "     </g>\n",
       "     <g id=\"text_13\">\n",
       "      <!-- Thyroid_Cancer -->\n",
       "      <g transform=\"translate(523.276705 240.305653)scale(0.1 -0.1)\">\n",
       "       <use xlink:href=\"#DejaVuSans-84\"/>\n",
       "       <use x=\"61.083984\" xlink:href=\"#DejaVuSans-104\"/>\n",
       "       <use x=\"124.462891\" xlink:href=\"#DejaVuSans-121\"/>\n",
       "       <use x=\"183.642578\" xlink:href=\"#DejaVuSans-114\"/>\n",
       "       <use x=\"222.505859\" xlink:href=\"#DejaVuSans-111\"/>\n",
       "       <use x=\"283.6875\" xlink:href=\"#DejaVuSans-105\"/>\n",
       "       <use x=\"311.470703\" xlink:href=\"#DejaVuSans-100\"/>\n",
       "       <use x=\"374.947266\" xlink:href=\"#DejaVuSans-95\"/>\n",
       "       <use x=\"424.947266\" xlink:href=\"#DejaVuSans-67\"/>\n",
       "       <use x=\"494.771484\" xlink:href=\"#DejaVuSans-97\"/>\n",
       "       <use x=\"556.050781\" xlink:href=\"#DejaVuSans-110\"/>\n",
       "       <use x=\"619.429688\" xlink:href=\"#DejaVuSans-99\"/>\n",
       "       <use x=\"674.410156\" xlink:href=\"#DejaVuSans-101\"/>\n",
       "       <use x=\"735.933594\" xlink:href=\"#DejaVuSans-114\"/>\n",
       "      </g>\n",
       "     </g>\n",
       "    </g>\n",
       "    <g id=\"text_14\">\n",
       "     <!-- Predicted label -->\n",
       "     <g transform=\"translate(474.250213 254.261903)scale(0.1 -0.1)\">\n",
       "      <use xlink:href=\"#DejaVuSans-80\"/>\n",
       "      <use x=\"58.552734\" xlink:href=\"#DejaVuSans-114\"/>\n",
       "      <use x=\"97.416016\" xlink:href=\"#DejaVuSans-101\"/>\n",
       "      <use x=\"158.939453\" xlink:href=\"#DejaVuSans-100\"/>\n",
       "      <use x=\"222.416016\" xlink:href=\"#DejaVuSans-105\"/>\n",
       "      <use x=\"250.199219\" xlink:href=\"#DejaVuSans-99\"/>\n",
       "      <use x=\"305.179688\" xlink:href=\"#DejaVuSans-116\"/>\n",
       "      <use x=\"344.388672\" xlink:href=\"#DejaVuSans-101\"/>\n",
       "      <use x=\"405.912109\" xlink:href=\"#DejaVuSans-100\"/>\n",
       "      <use x=\"469.388672\" xlink:href=\"#DejaVuSans-32\"/>\n",
       "      <use x=\"501.175781\" xlink:href=\"#DejaVuSans-108\"/>\n",
       "      <use x=\"528.958984\" xlink:href=\"#DejaVuSans-97\"/>\n",
       "      <use x=\"590.238281\" xlink:href=\"#DejaVuSans-98\"/>\n",
       "      <use x=\"653.714844\" xlink:href=\"#DejaVuSans-101\"/>\n",
       "      <use x=\"715.238281\" xlink:href=\"#DejaVuSans-108\"/>\n",
       "     </g>\n",
       "    </g>\n",
       "   </g>\n",
       "   <g id=\"matplotlib.axis_4\">\n",
       "    <g id=\"ytick_3\">\n",
       "     <g id=\"line2d_7\">\n",
       "      <g>\n",
       "       <use style=\"stroke:#000000;stroke-width:0.8;\" x=\"409.948011\" xlink:href=\"#m60f16bc87b\" y=\"73.525398\"/>\n",
       "      </g>\n",
       "     </g>\n",
       "     <g id=\"text_15\">\n",
       "      <!-- Lung_Cancer -->\n",
       "      <g transform=\"translate(338.316761 77.236335)scale(0.1 -0.1)\">\n",
       "       <use xlink:href=\"#DejaVuSans-76\"/>\n",
       "       <use x=\"53.962891\" xlink:href=\"#DejaVuSans-117\"/>\n",
       "       <use x=\"117.341797\" xlink:href=\"#DejaVuSans-110\"/>\n",
       "       <use x=\"180.720703\" xlink:href=\"#DejaVuSans-103\"/>\n",
       "       <use x=\"244.197266\" xlink:href=\"#DejaVuSans-95\"/>\n",
       "       <use x=\"294.197266\" xlink:href=\"#DejaVuSans-67\"/>\n",
       "       <use x=\"364.021484\" xlink:href=\"#DejaVuSans-97\"/>\n",
       "       <use x=\"425.300781\" xlink:href=\"#DejaVuSans-110\"/>\n",
       "       <use x=\"488.679688\" xlink:href=\"#DejaVuSans-99\"/>\n",
       "       <use x=\"543.660156\" xlink:href=\"#DejaVuSans-101\"/>\n",
       "       <use x=\"605.183594\" xlink:href=\"#DejaVuSans-114\"/>\n",
       "      </g>\n",
       "     </g>\n",
       "    </g>\n",
       "    <g id=\"ytick_4\">\n",
       "     <g id=\"line2d_8\">\n",
       "      <g>\n",
       "       <use style=\"stroke:#000000;stroke-width:0.8;\" x=\"409.948011\" xlink:href=\"#m60f16bc87b\" y=\"174.979943\"/>\n",
       "      </g>\n",
       "     </g>\n",
       "     <g id=\"text_16\">\n",
       "      <!-- Thyroid_Cancer -->\n",
       "      <g transform=\"translate(325.241761 178.779162)scale(0.1 -0.1)\">\n",
       "       <use xlink:href=\"#DejaVuSans-84\"/>\n",
       "       <use x=\"61.083984\" xlink:href=\"#DejaVuSans-104\"/>\n",
       "       <use x=\"124.462891\" xlink:href=\"#DejaVuSans-121\"/>\n",
       "       <use x=\"183.642578\" xlink:href=\"#DejaVuSans-114\"/>\n",
       "       <use x=\"222.505859\" xlink:href=\"#DejaVuSans-111\"/>\n",
       "       <use x=\"283.6875\" xlink:href=\"#DejaVuSans-105\"/>\n",
       "       <use x=\"311.470703\" xlink:href=\"#DejaVuSans-100\"/>\n",
       "       <use x=\"374.947266\" xlink:href=\"#DejaVuSans-95\"/>\n",
       "       <use x=\"424.947266\" xlink:href=\"#DejaVuSans-67\"/>\n",
       "       <use x=\"494.771484\" xlink:href=\"#DejaVuSans-97\"/>\n",
       "       <use x=\"556.050781\" xlink:href=\"#DejaVuSans-110\"/>\n",
       "       <use x=\"619.429688\" xlink:href=\"#DejaVuSans-99\"/>\n",
       "       <use x=\"674.410156\" xlink:href=\"#DejaVuSans-101\"/>\n",
       "       <use x=\"735.933594\" xlink:href=\"#DejaVuSans-114\"/>\n",
       "      </g>\n",
       "     </g>\n",
       "    </g>\n",
       "    <g id=\"text_17\">\n",
       "     <!-- True label -->\n",
       "     <g transform=\"translate(319.162074 148.551889)rotate(-90)scale(0.1 -0.1)\">\n",
       "      <use xlink:href=\"#DejaVuSans-84\"/>\n",
       "      <use x=\"46.333984\" xlink:href=\"#DejaVuSans-114\"/>\n",
       "      <use x=\"87.447266\" xlink:href=\"#DejaVuSans-117\"/>\n",
       "      <use x=\"150.826172\" xlink:href=\"#DejaVuSans-101\"/>\n",
       "      <use x=\"212.349609\" xlink:href=\"#DejaVuSans-32\"/>\n",
       "      <use x=\"244.136719\" xlink:href=\"#DejaVuSans-108\"/>\n",
       "      <use x=\"271.919922\" xlink:href=\"#DejaVuSans-97\"/>\n",
       "      <use x=\"333.199219\" xlink:href=\"#DejaVuSans-98\"/>\n",
       "      <use x=\"396.675781\" xlink:href=\"#DejaVuSans-101\"/>\n",
       "      <use x=\"458.199219\" xlink:href=\"#DejaVuSans-108\"/>\n",
       "     </g>\n",
       "    </g>\n",
       "   </g>\n",
       "   <g id=\"patch_8\">\n",
       "    <path d=\"M 409.948011 225.707216 \n",
       "L 409.948011 22.798125 \n",
       "\" style=\"fill:none;stroke:#000000;stroke-linecap:square;stroke-linejoin:miter;stroke-width:0.8;\"/>\n",
       "   </g>\n",
       "   <g id=\"patch_9\">\n",
       "    <path d=\"M 612.857102 225.707216 \n",
       "L 612.857102 22.798125 \n",
       "\" style=\"fill:none;stroke:#000000;stroke-linecap:square;stroke-linejoin:miter;stroke-width:0.8;\"/>\n",
       "   </g>\n",
       "   <g id=\"patch_10\">\n",
       "    <path d=\"M 409.948011 225.707216 \n",
       "L 612.857102 225.707216 \n",
       "\" style=\"fill:none;stroke:#000000;stroke-linecap:square;stroke-linejoin:miter;stroke-width:0.8;\"/>\n",
       "   </g>\n",
       "   <g id=\"patch_11\">\n",
       "    <path d=\"M 409.948011 22.798125 \n",
       "L 612.857102 22.798125 \n",
       "\" style=\"fill:none;stroke:#000000;stroke-linecap:square;stroke-linejoin:miter;stroke-width:0.8;\"/>\n",
       "   </g>\n",
       "   <g id=\"text_18\">\n",
       "    <!-- 0.98 -->\n",
       "    <g style=\"fill:#f7fbff;\" transform=\"translate(449.542472 76.284773)scale(0.1 -0.1)\">\n",
       "     <defs>\n",
       "      <path d=\"M 31.78125 34.625 \n",
       "Q 24.75 34.625 20.71875 30.859375 \n",
       "Q 16.703125 27.09375 16.703125 20.515625 \n",
       "Q 16.703125 13.921875 20.71875 10.15625 \n",
       "Q 24.75 6.390625 31.78125 6.390625 \n",
       "Q 38.8125 6.390625 42.859375 10.171875 \n",
       "Q 46.921875 13.96875 46.921875 20.515625 \n",
       "Q 46.921875 27.09375 42.890625 30.859375 \n",
       "Q 38.875 34.625 31.78125 34.625 \n",
       "z\n",
       "M 21.921875 38.8125 \n",
       "Q 15.578125 40.375 12.03125 44.71875 \n",
       "Q 8.5 49.078125 8.5 55.328125 \n",
       "Q 8.5 64.0625 14.71875 69.140625 \n",
       "Q 20.953125 74.21875 31.78125 74.21875 \n",
       "Q 42.671875 74.21875 48.875 69.140625 \n",
       "Q 55.078125 64.0625 55.078125 55.328125 \n",
       "Q 55.078125 49.078125 51.53125 44.71875 \n",
       "Q 48 40.375 41.703125 38.8125 \n",
       "Q 48.828125 37.15625 52.796875 32.3125 \n",
       "Q 56.78125 27.484375 56.78125 20.515625 \n",
       "Q 56.78125 9.90625 50.3125 4.234375 \n",
       "Q 43.84375 -1.421875 31.78125 -1.421875 \n",
       "Q 19.734375 -1.421875 13.25 4.234375 \n",
       "Q 6.78125 9.90625 6.78125 20.515625 \n",
       "Q 6.78125 27.484375 10.78125 32.3125 \n",
       "Q 14.796875 37.15625 21.921875 38.8125 \n",
       "z\n",
       "M 18.3125 54.390625 \n",
       "Q 18.3125 48.734375 21.84375 45.5625 \n",
       "Q 25.390625 42.390625 31.78125 42.390625 \n",
       "Q 38.140625 42.390625 41.71875 45.5625 \n",
       "Q 45.3125 48.734375 45.3125 54.390625 \n",
       "Q 45.3125 60.0625 41.71875 63.234375 \n",
       "Q 38.140625 66.40625 31.78125 66.40625 \n",
       "Q 25.390625 66.40625 21.84375 63.234375 \n",
       "Q 18.3125 60.0625 18.3125 54.390625 \n",
       "z\n",
       "\" id=\"DejaVuSans-56\"/>\n",
       "     </defs>\n",
       "     <use xlink:href=\"#DejaVuSans-48\"/>\n",
       "     <use x=\"63.623047\" xlink:href=\"#DejaVuSans-46\"/>\n",
       "     <use x=\"95.410156\" xlink:href=\"#DejaVuSans-57\"/>\n",
       "     <use x=\"159.033203\" xlink:href=\"#DejaVuSans-56\"/>\n",
       "    </g>\n",
       "   </g>\n",
       "   <g id=\"text_19\">\n",
       "    <!-- 0.022 -->\n",
       "    <g style=\"fill:#08306b;\" transform=\"translate(547.815767 76.284773)scale(0.1 -0.1)\">\n",
       "     <defs>\n",
       "      <path d=\"M 19.1875 8.296875 \n",
       "L 53.609375 8.296875 \n",
       "L 53.609375 0 \n",
       "L 7.328125 0 \n",
       "L 7.328125 8.296875 \n",
       "Q 12.9375 14.109375 22.625 23.890625 \n",
       "Q 32.328125 33.6875 34.8125 36.53125 \n",
       "Q 39.546875 41.84375 41.421875 45.53125 \n",
       "Q 43.3125 49.21875 43.3125 52.78125 \n",
       "Q 43.3125 58.59375 39.234375 62.25 \n",
       "Q 35.15625 65.921875 28.609375 65.921875 \n",
       "Q 23.96875 65.921875 18.8125 64.3125 \n",
       "Q 13.671875 62.703125 7.8125 59.421875 \n",
       "L 7.8125 69.390625 \n",
       "Q 13.765625 71.78125 18.9375 73 \n",
       "Q 24.125 74.21875 28.421875 74.21875 \n",
       "Q 39.75 74.21875 46.484375 68.546875 \n",
       "Q 53.21875 62.890625 53.21875 53.421875 \n",
       "Q 53.21875 48.921875 51.53125 44.890625 \n",
       "Q 49.859375 40.875 45.40625 35.40625 \n",
       "Q 44.1875 33.984375 37.640625 27.21875 \n",
       "Q 31.109375 20.453125 19.1875 8.296875 \n",
       "z\n",
       "\" id=\"DejaVuSans-50\"/>\n",
       "     </defs>\n",
       "     <use xlink:href=\"#DejaVuSans-48\"/>\n",
       "     <use x=\"63.623047\" xlink:href=\"#DejaVuSans-46\"/>\n",
       "     <use x=\"95.410156\" xlink:href=\"#DejaVuSans-48\"/>\n",
       "     <use x=\"159.033203\" xlink:href=\"#DejaVuSans-50\"/>\n",
       "     <use x=\"222.65625\" xlink:href=\"#DejaVuSans-50\"/>\n",
       "    </g>\n",
       "   </g>\n",
       "   <g id=\"text_20\">\n",
       "    <!-- 0 -->\n",
       "    <g style=\"fill:#08306b;\" transform=\"translate(457.494034 177.739318)scale(0.1 -0.1)\">\n",
       "     <use xlink:href=\"#DejaVuSans-48\"/>\n",
       "    </g>\n",
       "   </g>\n",
       "   <g id=\"text_21\">\n",
       "    <!-- 1 -->\n",
       "    <g style=\"fill:#f7fbff;\" transform=\"translate(558.94858 177.739318)scale(0.1 -0.1)\">\n",
       "     <use xlink:href=\"#DejaVuSans-49\"/>\n",
       "    </g>\n",
       "   </g>\n",
       "   <g id=\"text_22\">\n",
       "    <!-- Regresi√≥n Log√≠stica -->\n",
       "    <g transform=\"translate(453.300057 16.798125)scale(0.12 -0.12)\">\n",
       "     <defs>\n",
       "      <path d=\"M 44.390625 34.1875 \n",
       "Q 47.5625 33.109375 50.5625 29.59375 \n",
       "Q 53.5625 26.078125 56.59375 19.921875 \n",
       "L 66.609375 0 \n",
       "L 56 0 \n",
       "L 46.6875 18.703125 \n",
       "Q 43.0625 26.03125 39.671875 28.421875 \n",
       "Q 36.28125 30.8125 30.421875 30.8125 \n",
       "L 19.671875 30.8125 \n",
       "L 19.671875 0 \n",
       "L 9.8125 0 \n",
       "L 9.8125 72.90625 \n",
       "L 32.078125 72.90625 \n",
       "Q 44.578125 72.90625 50.734375 67.671875 \n",
       "Q 56.890625 62.453125 56.890625 51.90625 \n",
       "Q 56.890625 45.015625 53.6875 40.46875 \n",
       "Q 50.484375 35.9375 44.390625 34.1875 \n",
       "z\n",
       "M 19.671875 64.796875 \n",
       "L 19.671875 38.921875 \n",
       "L 32.078125 38.921875 \n",
       "Q 39.203125 38.921875 42.84375 42.21875 \n",
       "Q 46.484375 45.515625 46.484375 51.90625 \n",
       "Q 46.484375 58.296875 42.84375 61.546875 \n",
       "Q 39.203125 64.796875 32.078125 64.796875 \n",
       "z\n",
       "\" id=\"DejaVuSans-82\"/>\n",
       "      <path d=\"M 30.609375 48.390625 \n",
       "Q 23.390625 48.390625 19.1875 42.75 \n",
       "Q 14.984375 37.109375 14.984375 27.296875 \n",
       "Q 14.984375 17.484375 19.15625 11.84375 \n",
       "Q 23.34375 6.203125 30.609375 6.203125 \n",
       "Q 37.796875 6.203125 41.984375 11.859375 \n",
       "Q 46.1875 17.53125 46.1875 27.296875 \n",
       "Q 46.1875 37.015625 41.984375 42.703125 \n",
       "Q 37.796875 48.390625 30.609375 48.390625 \n",
       "z\n",
       "M 30.609375 56 \n",
       "Q 42.328125 56 49.015625 48.375 \n",
       "Q 55.71875 40.765625 55.71875 27.296875 \n",
       "Q 55.71875 13.875 49.015625 6.21875 \n",
       "Q 42.328125 -1.421875 30.609375 -1.421875 \n",
       "Q 18.84375 -1.421875 12.171875 6.21875 \n",
       "Q 5.515625 13.875 5.515625 27.296875 \n",
       "Q 5.515625 40.765625 12.171875 48.375 \n",
       "Q 18.84375 56 30.609375 56 \n",
       "z\n",
       "M 37.40625 79.984375 \n",
       "L 47.125 79.984375 \n",
       "L 31.21875 61.625 \n",
       "L 23.734375 61.625 \n",
       "z\n",
       "\" id=\"DejaVuSans-243\"/>\n",
       "      <path d=\"M 20.65625 79.984375 \n",
       "L 30.375 79.984375 \n",
       "L 14.46875 61.625 \n",
       "L 6.984375 61.625 \n",
       "z\n",
       "M 9.421875 54.6875 \n",
       "L 18.40625 54.6875 \n",
       "L 18.40625 0 \n",
       "L 9.421875 0 \n",
       "z\n",
       "M 13.921875 56 \n",
       "z\n",
       "\" id=\"DejaVuSans-237\"/>\n",
       "     </defs>\n",
       "     <use xlink:href=\"#DejaVuSans-82\"/>\n",
       "     <use x=\"64.982422\" xlink:href=\"#DejaVuSans-101\"/>\n",
       "     <use x=\"126.505859\" xlink:href=\"#DejaVuSans-103\"/>\n",
       "     <use x=\"189.982422\" xlink:href=\"#DejaVuSans-114\"/>\n",
       "     <use x=\"228.845703\" xlink:href=\"#DejaVuSans-101\"/>\n",
       "     <use x=\"290.369141\" xlink:href=\"#DejaVuSans-115\"/>\n",
       "     <use x=\"342.46875\" xlink:href=\"#DejaVuSans-105\"/>\n",
       "     <use x=\"370.251953\" xlink:href=\"#DejaVuSans-243\"/>\n",
       "     <use x=\"431.433594\" xlink:href=\"#DejaVuSans-110\"/>\n",
       "     <use x=\"494.8125\" xlink:href=\"#DejaVuSans-32\"/>\n",
       "     <use x=\"526.599609\" xlink:href=\"#DejaVuSans-76\"/>\n",
       "     <use x=\"580.5625\" xlink:href=\"#DejaVuSans-111\"/>\n",
       "     <use x=\"641.744141\" xlink:href=\"#DejaVuSans-103\"/>\n",
       "     <use x=\"705.220703\" xlink:href=\"#DejaVuSans-237\"/>\n",
       "     <use x=\"733.003906\" xlink:href=\"#DejaVuSans-115\"/>\n",
       "     <use x=\"785.103516\" xlink:href=\"#DejaVuSans-116\"/>\n",
       "     <use x=\"824.3125\" xlink:href=\"#DejaVuSans-105\"/>\n",
       "     <use x=\"852.095703\" xlink:href=\"#DejaVuSans-99\"/>\n",
       "     <use x=\"907.076172\" xlink:href=\"#DejaVuSans-97\"/>\n",
       "    </g>\n",
       "   </g>\n",
       "  </g>\n",
       "  <g id=\"axes_3\">\n",
       "   <g id=\"patch_12\">\n",
       "    <path d=\"M 105.584375 522.216307 \n",
       "L 308.493466 522.216307 \n",
       "L 308.493466 319.307216 \n",
       "L 105.584375 319.307216 \n",
       "z\n",
       "\" style=\"fill:#ffffff;\"/>\n",
       "   </g>\n",
       "   <g clip-path=\"url(#p199fdd1e77)\">\n",
       "    <image height=\"203\" id=\"image28e57cef71\" transform=\"scale(1 -1)translate(0 -203)\" width=\"203\" x=\"105.584375\" xlink:href=\"data:image/png;base64,\n",
       "iVBORw0KGgoAAAANSUhEUgAAAMsAAADLCAYAAADA+2czAAACV0lEQVR4nO3VoQ3CUABFUYqqZRMMU9Fx2IphkAjEryapuO5Dcs4Ez9y85f0Z48RPudy22RM4cJ49AP6FWCASC0RigUgsEIkFIrFAJBaIxAKRWCASC0RigUgsEIkFIrFAJBaIxAKRWCASC0RigUgsEIkFIrFAJBaIxAKRWCASC0RigUgsEIkFIrFAJBaIxAKRWCASC0RigUgsEIkFIrFAJBaIxAKRWCASC0RigUgsEIkFIrFAJBaIxAKRWCASC0RigUgsEIkFIrFAJBaIxAKRWCASC0RigUgsEIkFIrFAJBaIxAKRWCASC0RigUgsEIkFIrFAJBaIxAKRWCASC0RigUgsEIkFIrFAJBaIxAKRWCASC0RigUgsEIkFIrFAJBaIxAKRWCASC0RigUgsEIkFIrFAJBaIxAKRWCASC0RigUgsEC3r9T5mj+Db6/mYPYEDngUisUAkFojEApFYIBILRGKBSCwQiQUisUAkFojEApFYIBILRGKBSCwQiQUisUAkFojEApFYIBILRGKBSCwQiQUisUAkFojEApFYIBILRGKBSCwQiQUisUAkFojEApFYIBILRGKBSCwQiQUisUAkFojEApFYIBILRGKBSCwQiQUisUAkFojEApFYIBILRGKBSCwQiQUisUAkFojEApFYIBILRGKBSCwQiQUisUAkFojEApFYIBILRGKBSCwQiQUisUAkFojEApFYIBILRGKBSCwQiQUisUAkFojEApFYIBILRGKBSCwQiQUisUAkFojEApFYIBILRGKBSCwQiQUisUAkFojEAtEOaqAKJ9BluMgAAAAASUVORK5CYII=\" y=\"-319.216307\"/>\n",
       "   </g>\n",
       "   <g id=\"matplotlib.axis_5\">\n",
       "    <g id=\"xtick_5\">\n",
       "     <g id=\"line2d_9\">\n",
       "      <g>\n",
       "       <use style=\"stroke:#000000;stroke-width:0.8;\" x=\"156.311648\" xlink:href=\"#mb703f7e117\" y=\"522.216307\"/>\n",
       "      </g>\n",
       "     </g>\n",
       "     <g id=\"text_23\">\n",
       "      <!-- Lung_Cancer -->\n",
       "      <g transform=\"translate(123.996023 536.638182)scale(0.1 -0.1)\">\n",
       "       <use xlink:href=\"#DejaVuSans-76\"/>\n",
       "       <use x=\"53.962891\" xlink:href=\"#DejaVuSans-117\"/>\n",
       "       <use x=\"117.341797\" xlink:href=\"#DejaVuSans-110\"/>\n",
       "       <use x=\"180.720703\" xlink:href=\"#DejaVuSans-103\"/>\n",
       "       <use x=\"244.197266\" xlink:href=\"#DejaVuSans-95\"/>\n",
       "       <use x=\"294.197266\" xlink:href=\"#DejaVuSans-67\"/>\n",
       "       <use x=\"364.021484\" xlink:href=\"#DejaVuSans-97\"/>\n",
       "       <use x=\"425.300781\" xlink:href=\"#DejaVuSans-110\"/>\n",
       "       <use x=\"488.679688\" xlink:href=\"#DejaVuSans-99\"/>\n",
       "       <use x=\"543.660156\" xlink:href=\"#DejaVuSans-101\"/>\n",
       "       <use x=\"605.183594\" xlink:href=\"#DejaVuSans-114\"/>\n",
       "      </g>\n",
       "     </g>\n",
       "    </g>\n",
       "    <g id=\"xtick_6\">\n",
       "     <g id=\"line2d_10\">\n",
       "      <g>\n",
       "       <use style=\"stroke:#000000;stroke-width:0.8;\" x=\"257.766193\" xlink:href=\"#mb703f7e117\" y=\"522.216307\"/>\n",
       "      </g>\n",
       "     </g>\n",
       "     <g id=\"text_24\">\n",
       "      <!-- Thyroid_Cancer -->\n",
       "      <g transform=\"translate(218.913068 536.814744)scale(0.1 -0.1)\">\n",
       "       <use xlink:href=\"#DejaVuSans-84\"/>\n",
       "       <use x=\"61.083984\" xlink:href=\"#DejaVuSans-104\"/>\n",
       "       <use x=\"124.462891\" xlink:href=\"#DejaVuSans-121\"/>\n",
       "       <use x=\"183.642578\" xlink:href=\"#DejaVuSans-114\"/>\n",
       "       <use x=\"222.505859\" xlink:href=\"#DejaVuSans-111\"/>\n",
       "       <use x=\"283.6875\" xlink:href=\"#DejaVuSans-105\"/>\n",
       "       <use x=\"311.470703\" xlink:href=\"#DejaVuSans-100\"/>\n",
       "       <use x=\"374.947266\" xlink:href=\"#DejaVuSans-95\"/>\n",
       "       <use x=\"424.947266\" xlink:href=\"#DejaVuSans-67\"/>\n",
       "       <use x=\"494.771484\" xlink:href=\"#DejaVuSans-97\"/>\n",
       "       <use x=\"556.050781\" xlink:href=\"#DejaVuSans-110\"/>\n",
       "       <use x=\"619.429688\" xlink:href=\"#DejaVuSans-99\"/>\n",
       "       <use x=\"674.410156\" xlink:href=\"#DejaVuSans-101\"/>\n",
       "       <use x=\"735.933594\" xlink:href=\"#DejaVuSans-114\"/>\n",
       "      </g>\n",
       "     </g>\n",
       "    </g>\n",
       "    <g id=\"text_25\">\n",
       "     <!-- Predicted label -->\n",
       "     <g transform=\"translate(169.886577 550.770994)scale(0.1 -0.1)\">\n",
       "      <use xlink:href=\"#DejaVuSans-80\"/>\n",
       "      <use x=\"58.552734\" xlink:href=\"#DejaVuSans-114\"/>\n",
       "      <use x=\"97.416016\" xlink:href=\"#DejaVuSans-101\"/>\n",
       "      <use x=\"158.939453\" xlink:href=\"#DejaVuSans-100\"/>\n",
       "      <use x=\"222.416016\" xlink:href=\"#DejaVuSans-105\"/>\n",
       "      <use x=\"250.199219\" xlink:href=\"#DejaVuSans-99\"/>\n",
       "      <use x=\"305.179688\" xlink:href=\"#DejaVuSans-116\"/>\n",
       "      <use x=\"344.388672\" xlink:href=\"#DejaVuSans-101\"/>\n",
       "      <use x=\"405.912109\" xlink:href=\"#DejaVuSans-100\"/>\n",
       "      <use x=\"469.388672\" xlink:href=\"#DejaVuSans-32\"/>\n",
       "      <use x=\"501.175781\" xlink:href=\"#DejaVuSans-108\"/>\n",
       "      <use x=\"528.958984\" xlink:href=\"#DejaVuSans-97\"/>\n",
       "      <use x=\"590.238281\" xlink:href=\"#DejaVuSans-98\"/>\n",
       "      <use x=\"653.714844\" xlink:href=\"#DejaVuSans-101\"/>\n",
       "      <use x=\"715.238281\" xlink:href=\"#DejaVuSans-108\"/>\n",
       "     </g>\n",
       "    </g>\n",
       "   </g>\n",
       "   <g id=\"matplotlib.axis_6\">\n",
       "    <g id=\"ytick_5\">\n",
       "     <g id=\"line2d_11\">\n",
       "      <g>\n",
       "       <use style=\"stroke:#000000;stroke-width:0.8;\" x=\"105.584375\" xlink:href=\"#m60f16bc87b\" y=\"370.034489\"/>\n",
       "      </g>\n",
       "     </g>\n",
       "     <g id=\"text_26\">\n",
       "      <!-- Lung_Cancer -->\n",
       "      <g transform=\"translate(33.953125 373.745426)scale(0.1 -0.1)\">\n",
       "       <use xlink:href=\"#DejaVuSans-76\"/>\n",
       "       <use x=\"53.962891\" xlink:href=\"#DejaVuSans-117\"/>\n",
       "       <use x=\"117.341797\" xlink:href=\"#DejaVuSans-110\"/>\n",
       "       <use x=\"180.720703\" xlink:href=\"#DejaVuSans-103\"/>\n",
       "       <use x=\"244.197266\" xlink:href=\"#DejaVuSans-95\"/>\n",
       "       <use x=\"294.197266\" xlink:href=\"#DejaVuSans-67\"/>\n",
       "       <use x=\"364.021484\" xlink:href=\"#DejaVuSans-97\"/>\n",
       "       <use x=\"425.300781\" xlink:href=\"#DejaVuSans-110\"/>\n",
       "       <use x=\"488.679688\" xlink:href=\"#DejaVuSans-99\"/>\n",
       "       <use x=\"543.660156\" xlink:href=\"#DejaVuSans-101\"/>\n",
       "       <use x=\"605.183594\" xlink:href=\"#DejaVuSans-114\"/>\n",
       "      </g>\n",
       "     </g>\n",
       "    </g>\n",
       "    <g id=\"ytick_6\">\n",
       "     <g id=\"line2d_12\">\n",
       "      <g>\n",
       "       <use style=\"stroke:#000000;stroke-width:0.8;\" x=\"105.584375\" xlink:href=\"#m60f16bc87b\" y=\"471.489034\"/>\n",
       "      </g>\n",
       "     </g>\n",
       "     <g id=\"text_27\">\n",
       "      <!-- Thyroid_Cancer -->\n",
       "      <g transform=\"translate(20.878125 475.288253)scale(0.1 -0.1)\">\n",
       "       <use xlink:href=\"#DejaVuSans-84\"/>\n",
       "       <use x=\"61.083984\" xlink:href=\"#DejaVuSans-104\"/>\n",
       "       <use x=\"124.462891\" xlink:href=\"#DejaVuSans-121\"/>\n",
       "       <use x=\"183.642578\" xlink:href=\"#DejaVuSans-114\"/>\n",
       "       <use x=\"222.505859\" xlink:href=\"#DejaVuSans-111\"/>\n",
       "       <use x=\"283.6875\" xlink:href=\"#DejaVuSans-105\"/>\n",
       "       <use x=\"311.470703\" xlink:href=\"#DejaVuSans-100\"/>\n",
       "       <use x=\"374.947266\" xlink:href=\"#DejaVuSans-95\"/>\n",
       "       <use x=\"424.947266\" xlink:href=\"#DejaVuSans-67\"/>\n",
       "       <use x=\"494.771484\" xlink:href=\"#DejaVuSans-97\"/>\n",
       "       <use x=\"556.050781\" xlink:href=\"#DejaVuSans-110\"/>\n",
       "       <use x=\"619.429688\" xlink:href=\"#DejaVuSans-99\"/>\n",
       "       <use x=\"674.410156\" xlink:href=\"#DejaVuSans-101\"/>\n",
       "       <use x=\"735.933594\" xlink:href=\"#DejaVuSans-114\"/>\n",
       "      </g>\n",
       "     </g>\n",
       "    </g>\n",
       "    <g id=\"text_28\">\n",
       "     <!-- True label -->\n",
       "     <g transform=\"translate(14.798437 445.06098)rotate(-90)scale(0.1 -0.1)\">\n",
       "      <use xlink:href=\"#DejaVuSans-84\"/>\n",
       "      <use x=\"46.333984\" xlink:href=\"#DejaVuSans-114\"/>\n",
       "      <use x=\"87.447266\" xlink:href=\"#DejaVuSans-117\"/>\n",
       "      <use x=\"150.826172\" xlink:href=\"#DejaVuSans-101\"/>\n",
       "      <use x=\"212.349609\" xlink:href=\"#DejaVuSans-32\"/>\n",
       "      <use x=\"244.136719\" xlink:href=\"#DejaVuSans-108\"/>\n",
       "      <use x=\"271.919922\" xlink:href=\"#DejaVuSans-97\"/>\n",
       "      <use x=\"333.199219\" xlink:href=\"#DejaVuSans-98\"/>\n",
       "      <use x=\"396.675781\" xlink:href=\"#DejaVuSans-101\"/>\n",
       "      <use x=\"458.199219\" xlink:href=\"#DejaVuSans-108\"/>\n",
       "     </g>\n",
       "    </g>\n",
       "   </g>\n",
       "   <g id=\"patch_13\">\n",
       "    <path d=\"M 105.584375 522.216307 \n",
       "L 105.584375 319.307216 \n",
       "\" style=\"fill:none;stroke:#000000;stroke-linecap:square;stroke-linejoin:miter;stroke-width:0.8;\"/>\n",
       "   </g>\n",
       "   <g id=\"patch_14\">\n",
       "    <path d=\"M 308.493466 522.216307 \n",
       "L 308.493466 319.307216 \n",
       "\" style=\"fill:none;stroke:#000000;stroke-linecap:square;stroke-linejoin:miter;stroke-width:0.8;\"/>\n",
       "   </g>\n",
       "   <g id=\"patch_15\">\n",
       "    <path d=\"M 105.584375 522.216307 \n",
       "L 308.493466 522.216307 \n",
       "\" style=\"fill:none;stroke:#000000;stroke-linecap:square;stroke-linejoin:miter;stroke-width:0.8;\"/>\n",
       "   </g>\n",
       "   <g id=\"patch_16\">\n",
       "    <path d=\"M 105.584375 319.307216 \n",
       "L 308.493466 319.307216 \n",
       "\" style=\"fill:none;stroke:#000000;stroke-linecap:square;stroke-linejoin:miter;stroke-width:0.8;\"/>\n",
       "   </g>\n",
       "   <g id=\"text_29\">\n",
       "    <!-- 1 -->\n",
       "    <g style=\"fill:#f7fbff;\" transform=\"translate(153.130398 372.793864)scale(0.1 -0.1)\">\n",
       "     <use xlink:href=\"#DejaVuSans-49\"/>\n",
       "    </g>\n",
       "   </g>\n",
       "   <g id=\"text_30\">\n",
       "    <!-- 0 -->\n",
       "    <g style=\"fill:#08306b;\" transform=\"translate(254.584943 372.793864)scale(0.1 -0.1)\">\n",
       "     <use xlink:href=\"#DejaVuSans-48\"/>\n",
       "    </g>\n",
       "   </g>\n",
       "   <g id=\"text_31\">\n",
       "    <!-- 0 -->\n",
       "    <g style=\"fill:#08306b;\" transform=\"translate(153.130398 474.248409)scale(0.1 -0.1)\">\n",
       "     <use xlink:href=\"#DejaVuSans-48\"/>\n",
       "    </g>\n",
       "   </g>\n",
       "   <g id=\"text_32\">\n",
       "    <!-- 1 -->\n",
       "    <g style=\"fill:#f7fbff;\" transform=\"translate(254.584943 474.248409)scale(0.1 -0.1)\">\n",
       "     <use xlink:href=\"#DejaVuSans-49\"/>\n",
       "    </g>\n",
       "   </g>\n",
       "   <g id=\"text_33\">\n",
       "    <!-- SVM -->\n",
       "    <g transform=\"translate(193.948608 313.307216)scale(0.12 -0.12)\">\n",
       "     <defs>\n",
       "      <path d=\"M 53.515625 70.515625 \n",
       "L 53.515625 60.890625 \n",
       "Q 47.90625 63.578125 42.921875 64.890625 \n",
       "Q 37.9375 66.21875 33.296875 66.21875 \n",
       "Q 25.25 66.21875 20.875 63.09375 \n",
       "Q 16.5 59.96875 16.5 54.203125 \n",
       "Q 16.5 49.359375 19.40625 46.890625 \n",
       "Q 22.3125 44.4375 30.421875 42.921875 \n",
       "L 36.375 41.703125 \n",
       "Q 47.40625 39.59375 52.65625 34.296875 \n",
       "Q 57.90625 29 57.90625 20.125 \n",
       "Q 57.90625 9.515625 50.796875 4.046875 \n",
       "Q 43.703125 -1.421875 29.984375 -1.421875 \n",
       "Q 24.8125 -1.421875 18.96875 -0.25 \n",
       "Q 13.140625 0.921875 6.890625 3.21875 \n",
       "L 6.890625 13.375 \n",
       "Q 12.890625 10.015625 18.65625 8.296875 \n",
       "Q 24.421875 6.59375 29.984375 6.59375 \n",
       "Q 38.421875 6.59375 43.015625 9.90625 \n",
       "Q 47.609375 13.234375 47.609375 19.390625 \n",
       "Q 47.609375 24.75 44.3125 27.78125 \n",
       "Q 41.015625 30.8125 33.5 32.328125 \n",
       "L 27.484375 33.5 \n",
       "Q 16.453125 35.6875 11.515625 40.375 \n",
       "Q 6.59375 45.0625 6.59375 53.421875 \n",
       "Q 6.59375 63.09375 13.40625 68.65625 \n",
       "Q 20.21875 74.21875 32.171875 74.21875 \n",
       "Q 37.3125 74.21875 42.625 73.28125 \n",
       "Q 47.953125 72.359375 53.515625 70.515625 \n",
       "z\n",
       "\" id=\"DejaVuSans-83\"/>\n",
       "      <path d=\"M 28.609375 0 \n",
       "L 0.78125 72.90625 \n",
       "L 11.078125 72.90625 \n",
       "L 34.1875 11.53125 \n",
       "L 57.328125 72.90625 \n",
       "L 67.578125 72.90625 \n",
       "L 39.796875 0 \n",
       "z\n",
       "\" id=\"DejaVuSans-86\"/>\n",
       "      <path d=\"M 9.8125 72.90625 \n",
       "L 24.515625 72.90625 \n",
       "L 43.109375 23.296875 \n",
       "L 61.8125 72.90625 \n",
       "L 76.515625 72.90625 \n",
       "L 76.515625 0 \n",
       "L 66.890625 0 \n",
       "L 66.890625 64.015625 \n",
       "L 48.09375 14.015625 \n",
       "L 38.1875 14.015625 \n",
       "L 19.390625 64.015625 \n",
       "L 19.390625 0 \n",
       "L 9.8125 0 \n",
       "z\n",
       "\" id=\"DejaVuSans-77\"/>\n",
       "     </defs>\n",
       "     <use xlink:href=\"#DejaVuSans-83\"/>\n",
       "     <use x=\"63.476562\" xlink:href=\"#DejaVuSans-86\"/>\n",
       "     <use x=\"131.884766\" xlink:href=\"#DejaVuSans-77\"/>\n",
       "    </g>\n",
       "   </g>\n",
       "  </g>\n",
       "  <g id=\"axes_4\">\n",
       "   <g id=\"patch_17\">\n",
       "    <path d=\"M 409.948011 522.216307 \n",
       "L 612.857102 522.216307 \n",
       "L 612.857102 319.307216 \n",
       "L 409.948011 319.307216 \n",
       "z\n",
       "\" style=\"fill:#ffffff;\"/>\n",
       "   </g>\n",
       "   <g clip-path=\"url(#pe1968005ec)\">\n",
       "    <image height=\"203\" id=\"image6cabeb6b5f\" transform=\"scale(1 -1)translate(0 -203)\" width=\"203\" x=\"409.948011\" xlink:href=\"data:image/png;base64,\n",
       "iVBORw0KGgoAAAANSUhEUgAAAMsAAADLCAYAAADA+2czAAACV0lEQVR4nO3VoQ3CUABFUYqqZRMMU9Fx2IphkAjEryapuO5Dcs4Ez9y85f0Z48RPudy22RM4cJ49AP6FWCASC0RigUgsEIkFIrFAJBaIxAKRWCASC0RigUgsEIkFIrFAJBaIxAKRWCASC0RigUgsEIkFIrFAJBaIxAKRWCASC0RigUgsEIkFIrFAJBaIxAKRWCASC0RigUgsEIkFIrFAJBaIxAKRWCASC0RigUgsEIkFIrFAJBaIxAKRWCASC0RigUgsEIkFIrFAJBaIxAKRWCASC0RigUgsEIkFIrFAJBaIxAKRWCASC0RigUgsEIkFIrFAJBaIxAKRWCASC0RigUgsEIkFIrFAJBaIxAKRWCASC0RigUgsEIkFIrFAJBaIxAKRWCASC0RigUgsEIkFIrFAJBaIxAKRWCASC0RigUgsEC3r9T5mj+Db6/mYPYEDngUisUAkFojEApFYIBILRGKBSCwQiQUisUAkFojEApFYIBILRGKBSCwQiQUisUAkFojEApFYIBILRGKBSCwQiQUisUAkFojEApFYIBILRGKBSCwQiQUisUAkFojEApFYIBILRGKBSCwQiQUisUAkFojEApFYIBILRGKBSCwQiQUisUAkFojEApFYIBILRGKBSCwQiQUisUAkFojEApFYIBILRGKBSCwQiQUisUAkFojEApFYIBILRGKBSCwQiQUisUAkFojEApFYIBILRGKBSCwQiQUisUAkFojEApFYIBILRGKBSCwQiQUisUAkFojEApFYIBILRGKBSCwQiQUisUAkFojEAtEOaqAKJ9BluMgAAAAASUVORK5CYII=\" y=\"-319.216307\"/>\n",
       "   </g>\n",
       "   <g id=\"matplotlib.axis_7\">\n",
       "    <g id=\"xtick_7\">\n",
       "     <g id=\"line2d_13\">\n",
       "      <g>\n",
       "       <use style=\"stroke:#000000;stroke-width:0.8;\" x=\"460.675284\" xlink:href=\"#mb703f7e117\" y=\"522.216307\"/>\n",
       "      </g>\n",
       "     </g>\n",
       "     <g id=\"text_34\">\n",
       "      <!-- Lung_Cancer -->\n",
       "      <g transform=\"translate(428.359659 536.638182)scale(0.1 -0.1)\">\n",
       "       <use xlink:href=\"#DejaVuSans-76\"/>\n",
       "       <use x=\"53.962891\" xlink:href=\"#DejaVuSans-117\"/>\n",
       "       <use x=\"117.341797\" xlink:href=\"#DejaVuSans-110\"/>\n",
       "       <use x=\"180.720703\" xlink:href=\"#DejaVuSans-103\"/>\n",
       "       <use x=\"244.197266\" xlink:href=\"#DejaVuSans-95\"/>\n",
       "       <use x=\"294.197266\" xlink:href=\"#DejaVuSans-67\"/>\n",
       "       <use x=\"364.021484\" xlink:href=\"#DejaVuSans-97\"/>\n",
       "       <use x=\"425.300781\" xlink:href=\"#DejaVuSans-110\"/>\n",
       "       <use x=\"488.679688\" xlink:href=\"#DejaVuSans-99\"/>\n",
       "       <use x=\"543.660156\" xlink:href=\"#DejaVuSans-101\"/>\n",
       "       <use x=\"605.183594\" xlink:href=\"#DejaVuSans-114\"/>\n",
       "      </g>\n",
       "     </g>\n",
       "    </g>\n",
       "    <g id=\"xtick_8\">\n",
       "     <g id=\"line2d_14\">\n",
       "      <g>\n",
       "       <use style=\"stroke:#000000;stroke-width:0.8;\" x=\"562.12983\" xlink:href=\"#mb703f7e117\" y=\"522.216307\"/>\n",
       "      </g>\n",
       "     </g>\n",
       "     <g id=\"text_35\">\n",
       "      <!-- Thyroid_Cancer -->\n",
       "      <g transform=\"translate(523.276705 536.814744)scale(0.1 -0.1)\">\n",
       "       <use xlink:href=\"#DejaVuSans-84\"/>\n",
       "       <use x=\"61.083984\" xlink:href=\"#DejaVuSans-104\"/>\n",
       "       <use x=\"124.462891\" xlink:href=\"#DejaVuSans-121\"/>\n",
       "       <use x=\"183.642578\" xlink:href=\"#DejaVuSans-114\"/>\n",
       "       <use x=\"222.505859\" xlink:href=\"#DejaVuSans-111\"/>\n",
       "       <use x=\"283.6875\" xlink:href=\"#DejaVuSans-105\"/>\n",
       "       <use x=\"311.470703\" xlink:href=\"#DejaVuSans-100\"/>\n",
       "       <use x=\"374.947266\" xlink:href=\"#DejaVuSans-95\"/>\n",
       "       <use x=\"424.947266\" xlink:href=\"#DejaVuSans-67\"/>\n",
       "       <use x=\"494.771484\" xlink:href=\"#DejaVuSans-97\"/>\n",
       "       <use x=\"556.050781\" xlink:href=\"#DejaVuSans-110\"/>\n",
       "       <use x=\"619.429688\" xlink:href=\"#DejaVuSans-99\"/>\n",
       "       <use x=\"674.410156\" xlink:href=\"#DejaVuSans-101\"/>\n",
       "       <use x=\"735.933594\" xlink:href=\"#DejaVuSans-114\"/>\n",
       "      </g>\n",
       "     </g>\n",
       "    </g>\n",
       "    <g id=\"text_36\">\n",
       "     <!-- Predicted label -->\n",
       "     <g transform=\"translate(474.250213 550.770994)scale(0.1 -0.1)\">\n",
       "      <use xlink:href=\"#DejaVuSans-80\"/>\n",
       "      <use x=\"58.552734\" xlink:href=\"#DejaVuSans-114\"/>\n",
       "      <use x=\"97.416016\" xlink:href=\"#DejaVuSans-101\"/>\n",
       "      <use x=\"158.939453\" xlink:href=\"#DejaVuSans-100\"/>\n",
       "      <use x=\"222.416016\" xlink:href=\"#DejaVuSans-105\"/>\n",
       "      <use x=\"250.199219\" xlink:href=\"#DejaVuSans-99\"/>\n",
       "      <use x=\"305.179688\" xlink:href=\"#DejaVuSans-116\"/>\n",
       "      <use x=\"344.388672\" xlink:href=\"#DejaVuSans-101\"/>\n",
       "      <use x=\"405.912109\" xlink:href=\"#DejaVuSans-100\"/>\n",
       "      <use x=\"469.388672\" xlink:href=\"#DejaVuSans-32\"/>\n",
       "      <use x=\"501.175781\" xlink:href=\"#DejaVuSans-108\"/>\n",
       "      <use x=\"528.958984\" xlink:href=\"#DejaVuSans-97\"/>\n",
       "      <use x=\"590.238281\" xlink:href=\"#DejaVuSans-98\"/>\n",
       "      <use x=\"653.714844\" xlink:href=\"#DejaVuSans-101\"/>\n",
       "      <use x=\"715.238281\" xlink:href=\"#DejaVuSans-108\"/>\n",
       "     </g>\n",
       "    </g>\n",
       "   </g>\n",
       "   <g id=\"matplotlib.axis_8\">\n",
       "    <g id=\"ytick_7\">\n",
       "     <g id=\"line2d_15\">\n",
       "      <g>\n",
       "       <use style=\"stroke:#000000;stroke-width:0.8;\" x=\"409.948011\" xlink:href=\"#m60f16bc87b\" y=\"370.034489\"/>\n",
       "      </g>\n",
       "     </g>\n",
       "     <g id=\"text_37\">\n",
       "      <!-- Lung_Cancer -->\n",
       "      <g transform=\"translate(338.316761 373.745426)scale(0.1 -0.1)\">\n",
       "       <use xlink:href=\"#DejaVuSans-76\"/>\n",
       "       <use x=\"53.962891\" xlink:href=\"#DejaVuSans-117\"/>\n",
       "       <use x=\"117.341797\" xlink:href=\"#DejaVuSans-110\"/>\n",
       "       <use x=\"180.720703\" xlink:href=\"#DejaVuSans-103\"/>\n",
       "       <use x=\"244.197266\" xlink:href=\"#DejaVuSans-95\"/>\n",
       "       <use x=\"294.197266\" xlink:href=\"#DejaVuSans-67\"/>\n",
       "       <use x=\"364.021484\" xlink:href=\"#DejaVuSans-97\"/>\n",
       "       <use x=\"425.300781\" xlink:href=\"#DejaVuSans-110\"/>\n",
       "       <use x=\"488.679688\" xlink:href=\"#DejaVuSans-99\"/>\n",
       "       <use x=\"543.660156\" xlink:href=\"#DejaVuSans-101\"/>\n",
       "       <use x=\"605.183594\" xlink:href=\"#DejaVuSans-114\"/>\n",
       "      </g>\n",
       "     </g>\n",
       "    </g>\n",
       "    <g id=\"ytick_8\">\n",
       "     <g id=\"line2d_16\">\n",
       "      <g>\n",
       "       <use style=\"stroke:#000000;stroke-width:0.8;\" x=\"409.948011\" xlink:href=\"#m60f16bc87b\" y=\"471.489034\"/>\n",
       "      </g>\n",
       "     </g>\n",
       "     <g id=\"text_38\">\n",
       "      <!-- Thyroid_Cancer -->\n",
       "      <g transform=\"translate(325.241761 475.288253)scale(0.1 -0.1)\">\n",
       "       <use xlink:href=\"#DejaVuSans-84\"/>\n",
       "       <use x=\"61.083984\" xlink:href=\"#DejaVuSans-104\"/>\n",
       "       <use x=\"124.462891\" xlink:href=\"#DejaVuSans-121\"/>\n",
       "       <use x=\"183.642578\" xlink:href=\"#DejaVuSans-114\"/>\n",
       "       <use x=\"222.505859\" xlink:href=\"#DejaVuSans-111\"/>\n",
       "       <use x=\"283.6875\" xlink:href=\"#DejaVuSans-105\"/>\n",
       "       <use x=\"311.470703\" xlink:href=\"#DejaVuSans-100\"/>\n",
       "       <use x=\"374.947266\" xlink:href=\"#DejaVuSans-95\"/>\n",
       "       <use x=\"424.947266\" xlink:href=\"#DejaVuSans-67\"/>\n",
       "       <use x=\"494.771484\" xlink:href=\"#DejaVuSans-97\"/>\n",
       "       <use x=\"556.050781\" xlink:href=\"#DejaVuSans-110\"/>\n",
       "       <use x=\"619.429688\" xlink:href=\"#DejaVuSans-99\"/>\n",
       "       <use x=\"674.410156\" xlink:href=\"#DejaVuSans-101\"/>\n",
       "       <use x=\"735.933594\" xlink:href=\"#DejaVuSans-114\"/>\n",
       "      </g>\n",
       "     </g>\n",
       "    </g>\n",
       "    <g id=\"text_39\">\n",
       "     <!-- True label -->\n",
       "     <g transform=\"translate(319.162074 445.06098)rotate(-90)scale(0.1 -0.1)\">\n",
       "      <use xlink:href=\"#DejaVuSans-84\"/>\n",
       "      <use x=\"46.333984\" xlink:href=\"#DejaVuSans-114\"/>\n",
       "      <use x=\"87.447266\" xlink:href=\"#DejaVuSans-117\"/>\n",
       "      <use x=\"150.826172\" xlink:href=\"#DejaVuSans-101\"/>\n",
       "      <use x=\"212.349609\" xlink:href=\"#DejaVuSans-32\"/>\n",
       "      <use x=\"244.136719\" xlink:href=\"#DejaVuSans-108\"/>\n",
       "      <use x=\"271.919922\" xlink:href=\"#DejaVuSans-97\"/>\n",
       "      <use x=\"333.199219\" xlink:href=\"#DejaVuSans-98\"/>\n",
       "      <use x=\"396.675781\" xlink:href=\"#DejaVuSans-101\"/>\n",
       "      <use x=\"458.199219\" xlink:href=\"#DejaVuSans-108\"/>\n",
       "     </g>\n",
       "    </g>\n",
       "   </g>\n",
       "   <g id=\"patch_18\">\n",
       "    <path d=\"M 409.948011 522.216307 \n",
       "L 409.948011 319.307216 \n",
       "\" style=\"fill:none;stroke:#000000;stroke-linecap:square;stroke-linejoin:miter;stroke-width:0.8;\"/>\n",
       "   </g>\n",
       "   <g id=\"patch_19\">\n",
       "    <path d=\"M 612.857102 522.216307 \n",
       "L 612.857102 319.307216 \n",
       "\" style=\"fill:none;stroke:#000000;stroke-linecap:square;stroke-linejoin:miter;stroke-width:0.8;\"/>\n",
       "   </g>\n",
       "   <g id=\"patch_20\">\n",
       "    <path d=\"M 409.948011 522.216307 \n",
       "L 612.857102 522.216307 \n",
       "\" style=\"fill:none;stroke:#000000;stroke-linecap:square;stroke-linejoin:miter;stroke-width:0.8;\"/>\n",
       "   </g>\n",
       "   <g id=\"patch_21\">\n",
       "    <path d=\"M 409.948011 319.307216 \n",
       "L 612.857102 319.307216 \n",
       "\" style=\"fill:none;stroke:#000000;stroke-linecap:square;stroke-linejoin:miter;stroke-width:0.8;\"/>\n",
       "   </g>\n",
       "   <g id=\"text_40\">\n",
       "    <!-- 1 -->\n",
       "    <g style=\"fill:#f7fbff;\" transform=\"translate(457.494034 372.793864)scale(0.1 -0.1)\">\n",
       "     <use xlink:href=\"#DejaVuSans-49\"/>\n",
       "    </g>\n",
       "   </g>\n",
       "   <g id=\"text_41\">\n",
       "    <!-- 0 -->\n",
       "    <g style=\"fill:#08306b;\" transform=\"translate(558.94858 372.793864)scale(0.1 -0.1)\">\n",
       "     <use xlink:href=\"#DejaVuSans-48\"/>\n",
       "    </g>\n",
       "   </g>\n",
       "   <g id=\"text_42\">\n",
       "    <!-- 0 -->\n",
       "    <g style=\"fill:#08306b;\" transform=\"translate(457.494034 474.248409)scale(0.1 -0.1)\">\n",
       "     <use xlink:href=\"#DejaVuSans-48\"/>\n",
       "    </g>\n",
       "   </g>\n",
       "   <g id=\"text_43\">\n",
       "    <!-- 1 -->\n",
       "    <g style=\"fill:#f7fbff;\" transform=\"translate(558.94858 474.248409)scale(0.1 -0.1)\">\n",
       "     <use xlink:href=\"#DejaVuSans-49\"/>\n",
       "    </g>\n",
       "   </g>\n",
       "   <g id=\"text_44\">\n",
       "    <!-- √Årbol de decisi√≥n -->\n",
       "    <g transform=\"translate(459.636619 313.307216)scale(0.12 -0.12)\">\n",
       "     <defs>\n",
       "      <path d=\"M 34.1875 63.1875 \n",
       "L 20.796875 26.90625 \n",
       "L 47.609375 26.90625 \n",
       "z\n",
       "M 28.609375 72.90625 \n",
       "L 39.796875 72.90625 \n",
       "L 67.578125 0 \n",
       "L 57.328125 0 \n",
       "L 50.6875 18.703125 \n",
       "L 17.828125 18.703125 \n",
       "L 11.1875 0 \n",
       "L 0.78125 0 \n",
       "z\n",
       "M 36.8125 92.515625 \n",
       "L 45.84375 92.515625 \n",
       "L 34.71875 79.625 \n",
       "L 27.234375 79.625 \n",
       "z\n",
       "\" id=\"DejaVuSans-193\"/>\n",
       "     </defs>\n",
       "     <use xlink:href=\"#DejaVuSans-193\"/>\n",
       "     <use x=\"68.408203\" xlink:href=\"#DejaVuSans-114\"/>\n",
       "     <use x=\"109.521484\" xlink:href=\"#DejaVuSans-98\"/>\n",
       "     <use x=\"172.998047\" xlink:href=\"#DejaVuSans-111\"/>\n",
       "     <use x=\"234.179688\" xlink:href=\"#DejaVuSans-108\"/>\n",
       "     <use x=\"261.962891\" xlink:href=\"#DejaVuSans-32\"/>\n",
       "     <use x=\"293.75\" xlink:href=\"#DejaVuSans-100\"/>\n",
       "     <use x=\"357.226562\" xlink:href=\"#DejaVuSans-101\"/>\n",
       "     <use x=\"418.75\" xlink:href=\"#DejaVuSans-32\"/>\n",
       "     <use x=\"450.537109\" xlink:href=\"#DejaVuSans-100\"/>\n",
       "     <use x=\"514.013672\" xlink:href=\"#DejaVuSans-101\"/>\n",
       "     <use x=\"575.537109\" xlink:href=\"#DejaVuSans-99\"/>\n",
       "     <use x=\"630.517578\" xlink:href=\"#DejaVuSans-105\"/>\n",
       "     <use x=\"658.300781\" xlink:href=\"#DejaVuSans-115\"/>\n",
       "     <use x=\"710.400391\" xlink:href=\"#DejaVuSans-105\"/>\n",
       "     <use x=\"738.183594\" xlink:href=\"#DejaVuSans-243\"/>\n",
       "     <use x=\"799.365234\" xlink:href=\"#DejaVuSans-110\"/>\n",
       "    </g>\n",
       "   </g>\n",
       "  </g>\n",
       " </g>\n",
       " <defs>\n",
       "  <clipPath id=\"p227bc72e00\">\n",
       "   <rect height=\"202.909091\" width=\"202.909091\" x=\"105.584375\" y=\"22.798125\"/>\n",
       "  </clipPath>\n",
       "  <clipPath id=\"p1df93f9291\">\n",
       "   <rect height=\"202.909091\" width=\"202.909091\" x=\"409.948011\" y=\"22.798125\"/>\n",
       "  </clipPath>\n",
       "  <clipPath id=\"p199fdd1e77\">\n",
       "   <rect height=\"202.909091\" width=\"202.909091\" x=\"105.584375\" y=\"319.307216\"/>\n",
       "  </clipPath>\n",
       "  <clipPath id=\"pe1968005ec\">\n",
       "   <rect height=\"202.909091\" width=\"202.909091\" x=\"409.948011\" y=\"319.307216\"/>\n",
       "  </clipPath>\n",
       " </defs>\n",
       "</svg>\n"
      ],
      "text/plain": [
       "<Figure size 720x720 with 4 Axes>"
      ]
     },
     "metadata": {
      "needs_background": "light"
     },
     "output_type": "display_data"
    }
   ],
   "source": [
    "%config InlineBackend.figure_format = 'svg'\n",
    "\n",
    "fig = plt.figure(figsize=(10,10))\n",
    "\n",
    "ax1 = fig.add_subplot(2,2,1)\n",
    "plot_confusion_matrix(nb, X_test, y_test, cmap=\"Blues\", ax=ax1, normalize='true').im_.colorbar.remove()\n",
    "plt.title(\"Naive Bayes\")\n",
    "\n",
    "ax2 = fig.add_subplot(2,2,2)\n",
    "plot_confusion_matrix(logreg, X_test, y_test, cmap = 'Blues', ax=ax2, normalize='true').im_.colorbar.remove()\n",
    "plt.title(\"Regresi√≥n Log√≠stica\")\n",
    "\n",
    "ax3 = fig.add_subplot(2,2,3)\n",
    "plot_confusion_matrix(svm, X_test, y_test, cmap = 'Blues', ax=ax3, normalize='true').im_.colorbar.remove() \n",
    "plt.title(\"SVM\");\n",
    "\n",
    "ax4 = fig.add_subplot(2,2,4)\n",
    "plot_confusion_matrix(dt, X_test, y_test, cmap = 'Blues', ax=ax4, normalize='true').im_.colorbar.remove() \n",
    "plt.title(\"√Årbol de decisi√≥n\");"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "id": "7e9ef9bc",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{32543: 'gastroenterol',\n",
       " 69720: 'res',\n",
       " 65481: 'primary',\n",
       " 47453: 'localize',\n",
       " 3725: 'amyloidosis',\n",
       " 41752: 'intestine',\n",
       " 61494: 'pathologist',\n",
       " 88227: 'viewpointsaeed',\n",
       " 2707: 'ali',\n",
       " 3117: 'alshehria',\n",
       " 48547: 'mahmoud',\n",
       " 70435: 'rezk',\n",
       " 152: 'abdelwahe',\n",
       " 38001: 'husseinbabstractbackground',\n",
       " 67903: 'rare',\n",
       " 27167: 'entity',\n",
       " 17116: 'clinically',\n",
       " 49332: 'masquerade',\n",
       " 73778: 'several',\n",
       " 18550: 'condition',\n",
       " 17697: 'colitis',\n",
       " 64177: 'polyp',\n",
       " 48773: 'malignant',\n",
       " 85410: 'tumor',\n",
       " 77735: 'study',\n",
       " 2291: 'aim',\n",
       " 28142: 'evaluate',\n",
       " 17172: 'clinicopathological',\n",
       " 29650: 'feature',\n",
       " 27169: 'entitymethod',\n",
       " 18344: 'comprehensive',\n",
       " 72942: 'search',\n",
       " 47185: 'literature',\n",
       " 24442: 'do',\n",
       " 87114: 'use',\n",
       " 30676: 'follow',\n",
       " 44430: 'keyword',\n",
       " 75397: 'small',\n",
       " 25233: 'duodenum',\n",
       " 39032: 'ileum',\n",
       " 43267: 'jejunum',\n",
       " 17817: 'colon',\n",
       " 38658: 'identify',\n",
       " 32559: 'gastrointestinal',\n",
       " 21420: 'datum',\n",
       " 28378: 'examine',\n",
       " 41738: 'intestinal',\n",
       " 22477: 'describedresult',\n",
       " 1950: 'age',\n",
       " 65263: 'presentation',\n",
       " 67784: 'range',\n",
       " 91603: 'year',\n",
       " 48703: 'male',\n",
       " 29777: 'female',\n",
       " 68017: 'ratio',\n",
       " 74515: 'sigmoid',\n",
       " 18019: 'commonly',\n",
       " 42154: 'involve',\n",
       " 75085: 'site',\n",
       " 159: 'abdominal',\n",
       " 60750: 'pain',\n",
       " 57357: 'obstruction',\n",
       " 68672: 'rectal',\n",
       " 10906: 'bleeding',\n",
       " 68968: 'region',\n",
       " 18005: 'common',\n",
       " 17083: 'clinical',\n",
       " 17840: 'colonoscopic',\n",
       " 30168: 'finding',\n",
       " 39792: 'include',\n",
       " 88758: 'wall',\n",
       " 82400: 'thicken',\n",
       " 53512: 'mucosal',\n",
       " 86143: 'ulceration',\n",
       " 85485: 'tumorlike',\n",
       " 49346: 'masse',\n",
       " 17823: 'colonconclusion',\n",
       " 22310: 'depend',\n",
       " 22352: 'deposition',\n",
       " 3720: 'amyloid',\n",
       " 14090: 'case',\n",
       " 22351: 'deposit',\n",
       " 18845: 'consist',\n",
       " 46826: 'light',\n",
       " 15502: 'chain',\n",
       " 66332: 'proteinkeyword',\n",
       " 68677: 'rectum',\n",
       " 84313: 'tractintroductionthe',\n",
       " 80055: 'term',\n",
       " 40798: 'initially',\n",
       " 17655: 'coin',\n",
       " 22465: 'describe',\n",
       " 49406: 'material',\n",
       " 48976: 'manuscript',\n",
       " 78037: 'submit',\n",
       " 43793: 'june',\n",
       " 448: 'accept',\n",
       " 43772: 'july',\n",
       " 66850: 'publish',\n",
       " 59247: 'online',\n",
       " 8178: 'august',\n",
       " 1306: 'adepartment',\n",
       " 61496: 'pathology',\n",
       " 6965: 'armed',\n",
       " 30841: 'force',\n",
       " 37346: 'hospitals',\n",
       " 76053: 'southern',\n",
       " 44631: 'king',\n",
       " 29294: 'fahd',\n",
       " 37300: 'hospital',\n",
       " 72205: 'saudi',\n",
       " 6634: 'arabiabdepartment',\n",
       " 29268: 'faculty',\n",
       " 50155: 'medicine',\n",
       " 7733: 'assuit',\n",
       " 86656: 'university',\n",
       " 26166: 'egyptccorresponde',\n",
       " 8229: 'author',\n",
       " 71713: 'saeed',\n",
       " 3116: 'alshehri',\n",
       " 22300: 'department',\n",
       " 15823: 'chemistry',\n",
       " 73043: 'section',\n",
       " 44466: 'khamis',\n",
       " 53814: 'mushyte',\n",
       " 6632: 'arabia',\n",
       " 26559: 'email',\n",
       " 78959: 'syd',\n",
       " 33903: 'gmailcom',\n",
       " 34224: 'gr',\n",
       " 30166: 'find',\n",
       " 63639: 'plant',\n",
       " 82419: 'think',\n",
       " 69605: 'represent',\n",
       " 13675: 'carbohydrate',\n",
       " 34413: 'greek',\n",
       " 3729: 'amylon',\n",
       " 76868: 'starch',\n",
       " 50095: 'medical',\n",
       " 44840: 'know',\n",
       " 54246: 'name',\n",
       " 35186: 'hallmark',\n",
       " 66295: 'protein',\n",
       " 12858: 'call',\n",
       " 20191: 'crosssheet',\n",
       " 30006: 'fibril',\n",
       " 26239: 'either',\n",
       " 51929: 'misfolde',\n",
       " 51910: 'misassemble',\n",
       " 56059: 'nomenclature',\n",
       " 85741: 'two',\n",
       " 46433: 'letter',\n",
       " 30240: 'first',\n",
       " 68826: 'refer',\n",
       " 89874: 'whereas',\n",
       " 72976: 'second',\n",
       " 64983: 'precursor',\n",
       " 85823: 'type',\n",
       " 39302: 'immunoglobulin',\n",
       " 2469: 'al',\n",
       " 84639: 'transthyretin',\n",
       " 85343: 'ttr',\n",
       " 9966: 'beta',\n",
       " 51393: 'microglobulin',\n",
       " 36110: 'hemodialysisassociate',\n",
       " 73619: 'serum',\n",
       " 71650: 'saa',\n",
       " 72978: 'secondary',\n",
       " 1: 'aa',\n",
       " 7672: 'association',\n",
       " 36074: 'hematological',\n",
       " 53998: 'myeloma',\n",
       " 48265: 'lymphoplasmacytic',\n",
       " 24013: 'disorder',\n",
       " 63654: 'plasma',\n",
       " 14788: 'cell',\n",
       " 25391: 'dyscrasia',\n",
       " 52924: 'moreover',\n",
       " 49595: 'may',\n",
       " 74302: 'show',\n",
       " 74819: 'similar',\n",
       " 33074: 'genetic',\n",
       " 271: 'abnormality',\n",
       " 84574: 'translocation',\n",
       " 22033: 'deletion',\n",
       " 3724: 'amyloidose',\n",
       " 21919: 'degenerative',\n",
       " 7654: 'associate',\n",
       " 74631: 'significant',\n",
       " 78217: 'suffering',\n",
       " 61650: 'patient',\n",
       " 3124: 'also',\n",
       " 29364: 'familie',\n",
       " 89429: 'well',\n",
       " 15712: 'characterize',\n",
       " 28989: 'extracellular',\n",
       " 5796: 'ans',\n",
       " 44503: 'kidney',\n",
       " 55001: 'nerve',\n",
       " 84303: 'tract',\n",
       " 35866: 'heart',\n",
       " 75263: 'skin',\n",
       " 52871: 'morbidity',\n",
       " 48683: 'malabsorption',\n",
       " 69414: 'renal',\n",
       " 41228: 'insufficiency',\n",
       " 28201: 'even',\n",
       " 46758: 'lifethreatene',\n",
       " 28215: 'event',\n",
       " 29303: 'failure',\n",
       " 29493: 'fatal',\n",
       " 7044: 'arrhythmia',\n",
       " 56426: 'nonspecific',\n",
       " 73823: 'severity',\n",
       " 23003: 'diagnosis',\n",
       " 28358: 'examination',\n",
       " 83166: 'tissue',\n",
       " 10675: 'biopsy',\n",
       " 46363: 'lesion',\n",
       " 27858: 'establish',\n",
       " 22640: 'detection',\n",
       " 15650: 'characteristic',\n",
       " 10778: 'birefringence',\n",
       " 64063: 'polarize',\n",
       " 76759: 'stain',\n",
       " 18700: 'congo',\n",
       " 68727: 'red',\n",
       " 82426: 'thioflavint',\n",
       " 39307: 'immunohistochemical',\n",
       " 48849: 'management',\n",
       " 68800: 'reduction',\n",
       " 78476: 'supply',\n",
       " 76664: 'stabilization',\n",
       " 30992: 'formation',\n",
       " 1469: 'administration',\n",
       " 6095: 'antiplasma',\n",
       " 81783: 'therapie',\n",
       " 32902: 'general',\n",
       " 13892: 'care',\n",
       " 79177: 'systemic',\n",
       " 78655: 'surgical',\n",
       " 69810: 'resection',\n",
       " 31021: 'former',\n",
       " 28265: 'evidence',\n",
       " 17560: 'coexist',\n",
       " 23771: 'disease',\n",
       " 48635: 'major',\n",
       " 18286: 'component',\n",
       " 3721: 'amyloidassociate',\n",
       " 70104: 'result',\n",
       " 65800: 'production',\n",
       " 40647: 'inherently',\n",
       " 3722: 'amyloidogenic',\n",
       " 60504: 'overproduction',\n",
       " 64700: 'potentially',\n",
       " 56531: 'normal',\n",
       " 3732: 'amys',\n",
       " 43651: 'journal',\n",
       " 18162: 'compilation',\n",
       " 26493: 'elmer',\n",
       " 65327: 'press',\n",
       " 39703: 'inc',\n",
       " 91135: 'wwwgastroresthis',\n",
       " 24186: 'distribute',\n",
       " 20002: 'creative',\n",
       " 8104: 'attribution',\n",
       " 56146: 'noncommercial',\n",
       " 41595: 'international',\n",
       " 46687: 'license',\n",
       " 62682: 'permit',\n",
       " 86793: 'unrestricted',\n",
       " 24190: 'distribution',\n",
       " 69643: 'reproduction',\n",
       " 50208: 'medium',\n",
       " 66504: 'provide',\n",
       " 59709: 'original',\n",
       " 90899: 'work',\n",
       " 66167: 'properly',\n",
       " 16782: 'cite',\n",
       " 17215: 'clocalize',\n",
       " 68178: 're',\n",
       " 47564: 'loid',\n",
       " 90158: 'widespread',\n",
       " 70094: 'restrict',\n",
       " 76206: 'specific',\n",
       " 79120: 'synthesize',\n",
       " 29413: 'far',\n",
       " 8507: 'away',\n",
       " 81906: 'therefore',\n",
       " 24064: 'disparate',\n",
       " 48902: 'manifestation',\n",
       " 55217: 'neuropathie',\n",
       " 66409: 'proteinuria',\n",
       " 25189: 'due',\n",
       " 13834: 'cardiac',\n",
       " 55104: 'neural',\n",
       " 42163: 'involvement',\n",
       " 69922: 'respectively',\n",
       " 20637: 'currently',\n",
       " 16927: 'classification',\n",
       " 9083: 'base',\n",
       " 40467: 'infiltration',\n",
       " 17249: 'close',\n",
       " 5814: 'anspecific',\n",
       " 65778: 'produce',\n",
       " 56100: 'nonan',\n",
       " 30007: 'fibrillar',\n",
       " 1710: 'affect',\n",
       " 6433: 'appendix',\n",
       " 17872: 'colorectum',\n",
       " 17731: 'collagenous',\n",
       " 42474: 'ischemic',\n",
       " 11519: 'bowel',\n",
       " 40391: 'infarction',\n",
       " 62488: 'perforation',\n",
       " 34310: 'granularity',\n",
       " 31567: 'friability',\n",
       " 27649: 'erosion',\n",
       " 62801: 'petechial',\n",
       " 36126: 'hemorrhage',\n",
       " 64193: 'polypoid',\n",
       " 67561: 'radiological',\n",
       " 30642: 'fold',\n",
       " 54397: 'narrow',\n",
       " 17825: 'colonic',\n",
       " 48026: 'luman',\n",
       " 25942: 'effacement',\n",
       " 35495: 'haustration',\n",
       " 55921: 'nodularity',\n",
       " 21410: 'date',\n",
       " 69534: 'report',\n",
       " 79317: 'tada',\n",
       " 49333: 'mass',\n",
       " 23107: 'diarrhea',\n",
       " 36067: 'hematochezia',\n",
       " 26880: 'endoscopy',\n",
       " 70318: 'reveal',\n",
       " 30197: 'fine',\n",
       " 34309: 'granular',\n",
       " 6419: 'appearance',\n",
       " 64930: 'prealbumin',\n",
       " 36829: 'histologically',\n",
       " 73083: 'see',\n",
       " 78040: 'submucosa',\n",
       " 53789: 'muscle',\n",
       " 45956: 'layer',\n",
       " 15917: 'chen',\n",
       " 84642: 'transverse',\n",
       " 91623: 'yearold',\n",
       " 11018: 'bloody',\n",
       " 77332: 'stool',\n",
       " 89347: 'weight',\n",
       " 47691: 'loss',\n",
       " 17841: 'colonoscopy',\n",
       " 49162: 'mark',\n",
       " 65256: 'presence',\n",
       " 53658: 'multiple',\n",
       " 34672: 'growth',\n",
       " 78481: 'support',\n",
       " 40793: 'initial',\n",
       " 39550: 'impression',\n",
       " 13729: 'carcinoma',\n",
       " 86466: 'underwent',\n",
       " 70625: 'right',\n",
       " 36094: 'hemicolectomy',\n",
       " 48193: 'lymph',\n",
       " 55879: 'node',\n",
       " 24115: 'dissection',\n",
       " 34512: 'grossly',\n",
       " 86148: 'ulcerative',\n",
       " 45582: 'lamina',\n",
       " 66208: 'propria',\n",
       " 44842: 'knowledge',\n",
       " 46930: 'limited',\n",
       " 32283: 'gain',\n",
       " 41120: 'insight',\n",
       " 42730: 'issue',\n",
       " 18590: 'conduct',\n",
       " 50653: 'metaanalysis',\n",
       " 69285: 'relevant',\n",
       " 65258: 'present',\n",
       " 42636: 'isolate',\n",
       " 3727: 'amyloidosismaterial',\n",
       " 50997: 'methodsthis',\n",
       " 88226: 'viewpoint',\n",
       " 69734: 'research',\n",
       " 42110: 'investigation',\n",
       " 41397: 'interaction',\n",
       " 41704: 'intervention',\n",
       " 37842: 'human',\n",
       " 78002: 'subject',\n",
       " 459: 'access',\n",
       " 38640: 'identifiable',\n",
       " 65579: 'private',\n",
       " 40538: 'information',\n",
       " 69690: 'require',\n",
       " 41186: 'institutional',\n",
       " 70367: 'review',\n",
       " 11160: 'board',\n",
       " 62491: 'perform',\n",
       " 508: 'accordance',\n",
       " 65537: 'principle',\n",
       " 21714: 'declaration',\n",
       " 36042: 'helsinki',\n",
       " 77408: 'strategy',\n",
       " 73189: 'selection',\n",
       " 20113: 'criteriathe',\n",
       " 1323: 'adhere',\n",
       " 65087: 'preferred',\n",
       " 69580: 'reporting',\n",
       " 42814: 'item',\n",
       " 79149: 'systematic',\n",
       " 50651: 'metaanalyse',\n",
       " 65571: 'prisma',\n",
       " 34843: 'guideline',\n",
       " 66863: 'pubmed',\n",
       " 26357: 'electronic',\n",
       " 21249: 'database',\n",
       " 44405: 'key',\n",
       " 90894: 'word',\n",
       " 4896: 'andor',\n",
       " 14755: 'cecum',\n",
       " 26437: 'eligible',\n",
       " 18810: 'consider',\n",
       " 83290: 'title',\n",
       " 383: 'abstract',\n",
       " 66823: 'publication',\n",
       " 31866: 'full',\n",
       " 80274: 'text',\n",
       " 18635: 'confirm',\n",
       " 26436: 'eligibility',\n",
       " 78324: 'summary',\n",
       " 20119: 'criterion',\n",
       " 30088: 'figure',\n",
       " 29010: 'extraction',\n",
       " 50916: 'methodological',\n",
       " 7592: 'assessment',\n",
       " 76959: 'statistical',\n",
       " 3973: 'analysisthe',\n",
       " 29003: 'extract',\n",
       " 29369: 'family',\n",
       " 32808: 'gender',\n",
       " 78999: 'symptom',\n",
       " 26872: 'endoscopic',\n",
       " 36838: 'histology',\n",
       " 76175: 'special',\n",
       " 350: 'absence',\n",
       " 40082: 'independently',\n",
       " 68221: 'read',\n",
       " 4003: 'analyze',\n",
       " 21881: 'define',\n",
       " 30135: 'final',\n",
       " 36852: 'histopathological',\n",
       " 27131: 'ensure',\n",
       " 37125: 'homogeneous',\n",
       " 28163: 'evaluation',\n",
       " 18373: 'computation',\n",
       " 38504: 'ibmspss',\n",
       " 15988: 'chicago',\n",
       " 39014: 'il',\n",
       " 87058: 'usa',\n",
       " 22486: 'descriptive',\n",
       " 76958: 'statistic',\n",
       " 49873: 'mean',\n",
       " 76806: 'standard',\n",
       " 27661: 'error',\n",
       " 12806: 'calculatedresultsflow',\n",
       " 84916: 'trialour',\n",
       " 91803: 'yield',\n",
       " 65425: 'previous',\n",
       " 28463: 'exclude',\n",
       " 69322: 'remain',\n",
       " 86398: 'undergone',\n",
       " 19808: 'cover',\n",
       " 62591: 'period',\n",
       " 69255: 'relatively',\n",
       " 47581: 'long',\n",
       " 2914: 'allow',\n",
       " 66166: 'proper',\n",
       " 34103: 'good',\n",
       " 57088: 'number',\n",
       " 85720: 'twentyseven',\n",
       " 90963: 'workup',\n",
       " 52690: 'monoclonal',\n",
       " 32383: 'gammopathy',\n",
       " 26366: 'electrophoresis',\n",
       " 61466: 'pathological',\n",
       " 76246: 'specimen',\n",
       " 78114: 'substantiate',\n",
       " 4103: 'ancillary',\n",
       " 86213: 'ultrasound',\n",
       " 9000: 'barium',\n",
       " 26931: 'enema',\n",
       " 90089: 'wholebody',\n",
       " 18378: 'compute',\n",
       " 83838: 'tomography',\n",
       " 48521: 'magnetic',\n",
       " 69901: 'resonance',\n",
       " 39089: 'image',\n",
       " 64491: 'positron',\n",
       " 26642: 'emission',\n",
       " 75242: 'skeletal',\n",
       " 78697: 'survey',\n",
       " 26340: 'electrocardiography',\n",
       " 25626: 'echocardiography',\n",
       " 8246: 'authors',\n",
       " 91134: 'wwwgastrore',\n",
       " 12905: 'calshehri',\n",
       " 30451: 'flow',\n",
       " 15743: 'chart',\n",
       " 39881: 'inclusion',\n",
       " 90571: 'without',\n",
       " 46063: 'lead',\n",
       " 28480: 'exclusion',\n",
       " 59150: 'one',\n",
       " 30682: 'following',\n",
       " 16364: 'chronic',\n",
       " 17724: 'collagen',\n",
       " 87751: 'vascular',\n",
       " 1139: 'additional',\n",
       " 78252: 'suggestive',\n",
       " 18642: 'confirmatory',\n",
       " 23050: 'diagnostic',\n",
       " 9231: 'bcell',\n",
       " 48766: 'malignancy',\n",
       " 88749: 'waldenstrom',\n",
       " 48396: 'macroglobulinemia',\n",
       " 45406: 'laboratory',\n",
       " 64433: 'positive',\n",
       " 87038: 'urine',\n",
       " 32916: 'generalize',\n",
       " 50677: 'metabolic',\n",
       " 18256: 'complication',\n",
       " 29029: 'extraintestinal',\n",
       " 535: 'accumulate',\n",
       " 84702: 'treatment',\n",
       " 52290: 'modality',\n",
       " 3728: 'amyloidosisthe',\n",
       " 48564: 'main',\n",
       " 88574: 'vomiting',\n",
       " 10905: 'bleed',\n",
       " 18885: 'constipation',\n",
       " 5428: 'anemia',\n",
       " 62646: 'periumbilical',\n",
       " 48033: 'luminal',\n",
       " 24975: 'drug',\n",
       " 81795: 'therapy',\n",
       " 65055: 'prednisolone',\n",
       " 8621: 'azathioprine',\n",
       " 50393: 'melphalan',\n",
       " 17674: 'colchicines',\n",
       " 61252: 'partial',\n",
       " 43266: 'jejunectomy',\n",
       " 78041: 'submucosal',\n",
       " 30713: 'followup',\n",
       " 79272: 'table',\n",
       " 40231: 'individual',\n",
       " 41759: 'intestinecase',\n",
       " 73836: 'sex',\n",
       " 75106: 'sitesileumclinical',\n",
       " 48904: 'manifestationscolonoscopic',\n",
       " 30186: 'findingsstudie',\n",
       " 30569: 'fmfmfmmmmmmfmiddle',\n",
       " 1953: 'ageagedacute',\n",
       " 88580: 'vomitingpseuntestinal',\n",
       " 57360: 'obstructionmelena',\n",
       " 5430: 'anemiajejunumjejunumsmall',\n",
       " 41761: 'intestineintestinal',\n",
       " 57359: 'obstructionjejunum',\n",
       " 39033: 'ileummalabsorptionsmall',\n",
       " 1049: 'acute',\n",
       " 158: 'abdomenduodenum',\n",
       " 43268: 'jejunumsmall',\n",
       " 41754: 'intestineabdominal',\n",
       " 54500: 'nausea',\n",
       " 18889: 'constipationsmall',\n",
       " 57362: 'obstructionsmall',\n",
       " 36066: 'hematemesisthickene',\n",
       " 7206: 'ascend',\n",
       " 17849: 'colonthickene',\n",
       " 66484: 'protrusionsdiffuse',\n",
       " 53510: 'mucosa',\n",
       " 41767: 'intestinethickening',\n",
       " 43265: 'jejunal',\n",
       " 66560: 'proximal',\n",
       " 39030: 'ileal',\n",
       " 30657: 'foldsperforation',\n",
       " 88773: 'wallmultiple',\n",
       " 73975: 'shallow',\n",
       " 86141: 'ulcer',\n",
       " 27650: 'erosionsband',\n",
       " 18732: 'connective',\n",
       " 78679: 'surround',\n",
       " 41764: 'intestinemucosal',\n",
       " 23713: 'discrete',\n",
       " 55954: 'nodulessmall',\n",
       " 23401: 'digestive',\n",
       " 64201: 'polypsjejunumsmall',\n",
       " 71258: 'rough',\n",
       " 66645: 'pseudotumoral',\n",
       " 31009: 'formationsheartburn',\n",
       " 18888: 'constipationpolypoid',\n",
       " 30656: 'foldsf',\n",
       " 53382: 'msmall',\n",
       " 82662: 'three',\n",
       " 14123: 'casesduodenumscreening',\n",
       " 13015: 'cancer',\n",
       " 77315: 'stomacha',\n",
       " 22360: 'depressed',\n",
       " 46371: 'lesionendoscopic',\n",
       " 36821: 'histological',\n",
       " 3726: 'amyloidosisin',\n",
       " 66483: 'protrusion',\n",
       " 75814: 'solitary',\n",
       " 60817: 'pale',\n",
       " 27252: 'eosinophilic',\n",
       " 3619: 'amorphous',\n",
       " 66072: 'prominent',\n",
       " 19918: 'crack',\n",
       " 7130: 'artifact',\n",
       " 10974: 'blood',\n",
       " 88101: 'vessel',\n",
       " 76766: 'staining',\n",
       " 71850: 'salmonpink',\n",
       " 17859: 'color',\n",
       " 6448: 'applegreen',\n",
       " 5534: 'angiopathy',\n",
       " 54757: 'necrotizing',\n",
       " 5504: 'angiitis',\n",
       " 30894: 'foreign',\n",
       " 11187: 'body',\n",
       " 33469: 'giant',\n",
       " 68196: 'reaction',\n",
       " 15558: 'change',\n",
       " 30599: 'focal',\n",
       " 935: 'active',\n",
       " 20272: 'cryptitis',\n",
       " 62731: 'personal',\n",
       " 57303: 'observation',\n",
       " 48643: 'majority',\n",
       " 10: 'aaamyloid',\n",
       " 36464: 'high',\n",
       " 69588: 'reportprimary',\n",
       " 41753: 'intestinea',\n",
       " 5747: 'anorexia',\n",
       " 63378: 'picture',\n",
       " 69374: 'reminiscent',\n",
       " 60913: 'pancolitis',\n",
       " 31569: 'friable',\n",
       " 75158: 'six',\n",
       " 22459: 'descend',\n",
       " 57367: 'obtain',\n",
       " 6730: 'areas',\n",
       " 17822: 'coloncase',\n",
       " 73883: 'sexsitesclinical',\n",
       " 65268: 'presentationcolonoscopic',\n",
       " 30187: 'findingsstudiesage',\n",
       " 91672: 'yearsmmmmmfmmmfmftransverse',\n",
       " 17821: 'colonbleeding',\n",
       " 47698: 'lossintestinal',\n",
       " 57363: 'obstructiontransverse',\n",
       " 5431: 'anemiasigmoid',\n",
       " 17851: 'colonulceration',\n",
       " 46383: 'lesionselderlya',\n",
       " 77120: 'stenose',\n",
       " 49358: 'massmiddle',\n",
       " 49364: 'massulcerative',\n",
       " 46387: 'lesionsmucosal',\n",
       " 82402: 'thickening',\n",
       " 88759: 'walla',\n",
       " 49359: 'massmucosal',\n",
       " 86142: 'ulcera',\n",
       " 74948: 'single',\n",
       " 71266: 'rounded',\n",
       " 46364: 'lesiona',\n",
       " 17309: 'cm',\n",
       " 46365: 'lesionacute',\n",
       " 29698: 'fecal',\n",
       " 62642: 'peritonitishematocheziahemepositive',\n",
       " 77336: 'stoolsrectal',\n",
       " 10911: 'bleedingroutine',\n",
       " 17843: 'colonoscopynonspecific',\n",
       " 42411: 'irregularlyshape',\n",
       " 86146: 'ulcerationssigmoid',\n",
       " 17846: 'colonsigmoid',\n",
       " 17847: 'colonsigmoiddescende',\n",
       " 10946: 'bloat',\n",
       " 30382: 'flatulence',\n",
       " 36068: 'hematocheziarectal',\n",
       " 10909: 'bleedinghemorrhagic',\n",
       " 3355: 'amass',\n",
       " 46370: 'lesiondiffuse',\n",
       " 55919: 'nodular',\n",
       " 42406: 'irregular',\n",
       " 45713: 'large',\n",
       " 86145: 'ulcerationsascende',\n",
       " 17845: 'colonscolon',\n",
       " 73763: 'seven',\n",
       " 14114: 'casesascende',\n",
       " 17824: 'colonf',\n",
       " 51500: 'middle',\n",
       " 14130: 'casesheartburn',\n",
       " 18887: 'constipationfacute',\n",
       " 88583: 'vomitingthickene',\n",
       " 41758: 'intestineaspectsage',\n",
       " 73349: 'semmale',\n",
       " 68046: 'ratiosite',\n",
       " 42164: 'involvementclinical',\n",
       " 65273: 'presentationscolonoscopic',\n",
       " 30176: 'findingshistological',\n",
       " 29682: 'featuressmall',\n",
       " 39043: 'ilium',\n",
       " 46118: 'leave',\n",
       " 39946: 'increase',\n",
       " 22242: 'density',\n",
       " 52049: 'mixed',\n",
       " 40493: 'inflammatory',\n",
       " 46830: 'lightchain',\n",
       " 76865: 'star',\n",
       " 398: 'abundant',\n",
       " 63468: 'pink',\n",
       " 78561: 'surface',\n",
       " 56679: 'note',\n",
       " 8687: 'background',\n",
       " 40485: 'inflammation',\n",
       " 7058: 'arrow',\n",
       " 61426: 'patchy',\n",
       " 20270: 'crypt',\n",
       " 24176: 'distortion',\n",
       " 34320: 'granuloma',\n",
       " 36084: 'hematoxylin',\n",
       " 27248: 'eosin',\n",
       " 28557: 'exhibit',\n",
       " 27159: 'entirely',\n",
       " 69508: 'replace',\n",
       " 57413: 'occasional',\n",
       " 344: 'abscess',\n",
       " 27651: 'erosionulceration',\n",
       " 30068: 'fig',\n",
       " 3835: 'analysis',\n",
       " 45563: 'lambda',\n",
       " 54809: 'negative',\n",
       " 18847: 'consistent',\n",
       " 71514: 'rule',\n",
       " 64497: 'possibility',\n",
       " 40205: 'indice',\n",
       " 68106: 'rbc',\n",
       " 19722: 'count',\n",
       " 36114: 'hemoglobin',\n",
       " 19515: 'corpuscular',\n",
       " 88550: 'volume',\n",
       " 49815: 'mcv',\n",
       " 49742: 'mch',\n",
       " 63684: 'platelet',\n",
       " 90047: 'white',\n",
       " 10436: 'biochemical',\n",
       " 80132: 'test',\n",
       " 84111: 'total',\n",
       " 2527: 'albumin',\n",
       " 33750: 'globulin',\n",
       " 2776: 'alkaline',\n",
       " 63179: 'phosphatase',\n",
       " 7439: 'aspartate',\n",
       " 3488: 'aminotransferase',\n",
       " 7747: 'ast',\n",
       " 10329: 'bilirubin',\n",
       " 29874: 'ferritin',\n",
       " 88432: 'vitamin',\n",
       " 30664: 'folic',\n",
       " 661: 'acid',\n",
       " 66253: 'prostatespecific',\n",
       " 5994: 'antigen',\n",
       " 66609: 'psa',\n",
       " 87013: 'urea',\n",
       " 19993: 'creatinine',\n",
       " 26352: 'electrolyte',\n",
       " 76459: 'spot',\n",
       " 66318: 'proteincreatinine',\n",
       " 39289: 'immunoelectrophoresis',\n",
       " 90485: 'within',\n",
       " 46921: 'limit',\n",
       " 82862: 'thyroid',\n",
       " 31900: 'function',\n",
       " 26338: 'electrocardiograms',\n",
       " 47239: 'liver',\n",
       " 76397: 'spleen',\n",
       " 86781: 'unremarkable',\n",
       " 11277: 'bone',\n",
       " 49237: 'marrow',\n",
       " 80062: 'terminal',\n",
       " 40501: 'inflammatoryregenerative',\n",
       " 27865: 'establishedthe',\n",
       " 69609: 'representative',\n",
       " 70286: 'retrospective',\n",
       " 6693: 'archival',\n",
       " 30981: 'formalinfixe',\n",
       " 61081: 'paraffinembedde',\n",
       " 18920: 'consultation',\n",
       " 30101: 'file',\n",
       " 10949: 'block',\n",
       " 75353: 'slide',\n",
       " 17521: 'code',\n",
       " 89145: 'way',\n",
       " 38685: 'identity',\n",
       " 12975: 'can',\n",
       " 56641: 'not',\n",
       " 364: 'absolutely',\n",
       " 7209: 'ascertain',\n",
       " 18770: 'consent',\n",
       " 57371: 'obtaineddiscussionlocalize',\n",
       " 41461: 'interest',\n",
       " 32547: 'gastroenterologist',\n",
       " 51687: 'mimic',\n",
       " 2183: 'agreement',\n",
       " 40158: 'indicate',\n",
       " 60281: 'outcome',\n",
       " 88060: 'versus',\n",
       " 19826: 'cowan',\n",
       " 73775: 'seventysix',\n",
       " 61247: 'part',\n",
       " 45899: 'latter',\n",
       " 53848: 'mutation',\n",
       " 32828: 'gene',\n",
       " 65295: 'presenting',\n",
       " 82695: 'threlkeld',\n",
       " 55481: 'nguyen',\n",
       " 17682: 'colectomy',\n",
       " 54754: 'necrotic',\n",
       " 15510: 'chainspositive',\n",
       " 68372: 'receive',\n",
       " 17673: 'colchicine',\n",
       " 39574: 'improvement',\n",
       " 74843: 'similarly',\n",
       " 23122: 'diaz',\n",
       " 22005: 'del',\n",
       " 6703: 'arco',\n",
       " 33748: 'globular',\n",
       " 46185: 'left',\n",
       " 39498: 'important',\n",
       " 25232: 'duodenal',\n",
       " 27328: 'epigastric',\n",
       " 77068: 'steatorrhea',\n",
       " 21965: 'deguchi',\n",
       " 66640: 'pseudoobstruction',\n",
       " 16191: 'choi',\n",
       " 25263: 'duration',\n",
       " 39325: 'immunohistological',\n",
       " 42637: 'isolated',\n",
       " 47435: 'local',\n",
       " 21133: 'damage',\n",
       " 79112: 'synthesis',\n",
       " 78100: 'substance',\n",
       " 60192: 'otherwise',\n",
       " 9774: 'benign',\n",
       " 64130: 'polyclonal',\n",
       " 77226: 'still',\n",
       " 64499: 'possible',\n",
       " 39612: 'inability',\n",
       " 17008: 'clear',\n",
       " 87275: 'usually',\n",
       " 42419: 'irreversible',\n",
       " 75903: 'sometimes',\n",
       " 70355: 'reversible',\n",
       " 15877: 'chemotherapy',\n",
       " 27819: 'especially',\n",
       " 77179: 'steroid',\n",
       " 87687: 'variation',\n",
       " 53803: 'muscularis',\n",
       " 53511: 'mucosae',\n",
       " 26385: 'element',\n",
       " 22582: 'destructive',\n",
       " 25944: 'effect',\n",
       " 69510: 'replacement',\n",
       " 18129: 'compartment',\n",
       " 41157: 'instance',\n",
       " 42471: 'ischemia',\n",
       " 3723: 'amyloidoma',\n",
       " 3216: 'alter',\n",
       " 53110: 'motility',\n",
       " 39034: 'ileus',\n",
       " 50354: 'melena',\n",
       " 62405: 'per',\n",
       " 68304: 'reason',\n",
       " 36129: 'hemorrhagic',\n",
       " 54748: 'necrosis',\n",
       " 256: 'abnormal',\n",
       " 2079: 'aggregation',\n",
       " 34509: 'gross',\n",
       " 68311: 'reasoned',\n",
       " 57442: 'occur',\n",
       " 65064: 'predominantly',\n",
       " 62647: 'perivascular',\n",
       " 41667: 'interstitial',\n",
       " 76084: 'space',\n",
       " 23368: 'diffuse',\n",
       " 68009: 'rather',\n",
       " 26414: 'elevation',\n",
       " 68887: 'reflect',\n",
       " 2077: 'aggregate',\n",
       " 47459: 'locally',\n",
       " 59717: 'originate',\n",
       " 18489: 'conclude',\n",
       " 40094: 'indepth',\n",
       " 15709: 'characterization',\n",
       " 32921: 'generally',\n",
       " 15517: 'challenge',\n",
       " 25537: 'easily',\n",
       " 51948: 'miss',\n",
       " 88862: 'warrant',\n",
       " 8479: 'avoid',\n",
       " 2084: 'aggressive',\n",
       " 2791: 'alkylate',\n",
       " 2002: 'agent',\n",
       " 40106: 'index',\n",
       " 78813: 'suspicion',\n",
       " 48597: 'maintain',\n",
       " 36859: 'histopathologistsacknowledgmentsnone',\n",
       " 21723: 'declarefinancial',\n",
       " 23669: 'disclosurenone',\n",
       " 21716: 'declare',\n",
       " 18650: 'conflict',\n",
       " 41487: 'interestnone',\n",
       " 21724: 'declareinformed',\n",
       " 18777: 'consentinformed',\n",
       " 57370: 'obtainedauthor',\n",
       " 19185: 'contributionssaeed',\n",
       " 18425: 'conceive',\n",
       " 22500: 'design',\n",
       " 17742: 'collect',\n",
       " 17171: 'clinicopathologic',\n",
       " 7451: 'aspect',\n",
       " 3798: 'analyses',\n",
       " 83905: 'tool',\n",
       " 91042: 'write',\n",
       " 61029: 'paper',\n",
       " 153: 'abdelwahed',\n",
       " 38000: 'hussein',\n",
       " 78430: 'supervise',\n",
       " 19124: 'contribute',\n",
       " 18466: 'conception',\n",
       " 38624: 'idea',\n",
       " 21241: 'data',\n",
       " 17756: 'collection',\n",
       " 70402: 'revise',\n",
       " 48985: 'manuscriptdata',\n",
       " 8395: 'availabilitythe',\n",
       " 8397: 'available',\n",
       " 68827: 'reference',\n",
       " 89774: 'westermark',\n",
       " 9802: 'benson',\n",
       " 49819: 'md',\n",
       " 12318: 'buxbaum',\n",
       " 43521: 'jn',\n",
       " 17597: 'cohen',\n",
       " 31466: 'frangione',\n",
       " 39004: 'ikeda',\n",
       " 49374: 'master',\n",
       " 16885: 'cl',\n",
       " 52652: 'monge',\n",
       " 15763: 'chauveau',\n",
       " 19452: 'cordonnier',\n",
       " 55964: 'noel',\n",
       " 46624: 'lh',\n",
       " 65321: 'presne',\n",
       " 48657: 'makdassi',\n",
       " 43164: 'jaureguy',\n",
       " 33138: 'genitourinary',\n",
       " 55331: 'new',\n",
       " 8889: 'baltimore',\n",
       " 49457: 'mathis',\n",
       " 71096: 'rokitansky',\n",
       " 88310: 'virchow',\n",
       " 36299: 'heschl',\n",
       " 65638: 'problem',\n",
       " 36866: 'historical',\n",
       " 92197: 'zentralbl',\n",
       " 2863: 'allg',\n",
       " 61463: 'pathol',\n",
       " 71686: 'sack',\n",
       " 33405: 'gh',\n",
       " 43710: 'jr',\n",
       " 52535: 'mol',\n",
       " 50019: 'med',\n",
       " 33505: 'gillmore',\n",
       " 43239: 'jd',\n",
       " 89243: 'wechalekar',\n",
       " 10772: 'bird',\n",
       " 14440: 'cavenagh',\n",
       " 35585: 'hawkins',\n",
       " 44209: 'kazmi',\n",
       " 45439: 'lachmann',\n",
       " 36905: 'hj',\n",
       " 11580: 'br',\n",
       " 35078: 'haematol',\n",
       " 2355: 'aj',\n",
       " 75269: 'skinner',\n",
       " 73165: 'seldin',\n",
       " 21529: 'dc',\n",
       " 9882: 'berk',\n",
       " 43489: 'jl',\n",
       " ...}"
      ]
     },
     "execution_count": 48,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "vocab = {value:key for key,value in tfidf_vect.vocabulary_.items()}\n",
    "vocab"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "id": "1a245f18",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[('lung', -3.43),\n",
       " ('nsclc', -2.48),\n",
       " ('alk', -1.76),\n",
       " ('non', -1.58),\n",
       " ('egfr', -1.51),\n",
       " ('tumor', -1.48),\n",
       " ('logic', -1.21),\n",
       " ('module', -1.2),\n",
       " ('anti', -1.16),\n",
       " ('asbestos', -1.14)]"
      ]
     },
     "execution_count": 49,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Las palabras que contribuyen a la clase 1\n",
    "[(vocab[e[0]], round(e[1],2)) for e in zip(svm.coef_[0].argsort(), sorted(logreg.coef_[0]))][:10]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "id": "9d78bc34",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[('view', 0.01),\n",
       " ('subtypes', 0.01),\n",
       " ('andthe', 0.01),\n",
       " ('smoking', 0.02),\n",
       " ('jiaotong', 0.02),\n",
       " ('steroid', 0.03),\n",
       " ('party', 0.05),\n",
       " ('andor', 0.1),\n",
       " ('ofthe', 0.29),\n",
       " ('thyroid', 0.47)]"
      ]
     },
     "execution_count": 50,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "[(vocab[e[0]],round(e[1],2)) for e in zip(dt.feature_importances_.argsort(), sorted(dt.feature_importances_))][-10:]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "f498b7c7",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Collecting package metadata (current_repodata.json): done\n",
      "Solving environment: failed with initial frozen solve. Retrying with flexible solve.\n",
      "Solving environment: failed with repodata from current_repodata.json, will retry with next repodata source.\n",
      "Collecting package metadata (repodata.json): done\n",
      "Solving environment: failed with initial frozen solve. Retrying with flexible solve.\n",
      "Solving environment: \\ \n",
      "Found conflicts! Looking for incompatible packages.\n",
      "This can take several minutes.  Press CTRL-C to abort.\n",
      "Examining conflict for ffmpeg graphviz anaconda gnutls gettext:  59%|‚ñå| 229/386| "
     ]
    }
   ],
   "source": [
    "! conda install graphviz"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "0f90ff73",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "6c5514fd",
   "metadata": {},
   "outputs": [],
   "source": [
    "En Mac: pip install graphviz\n",
    "En Ubuntu: sudo apt-get install graphviz\n",
    "En Windows: conda install graphviz"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 64,
   "id": "a046b760",
   "metadata": {},
   "outputs": [
    {
     "ename": "ExecutableNotFound",
     "evalue": "failed to execute PosixPath('dot'), make sure the Graphviz executables are on your systems' PATH",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mFileNotFoundError\u001b[0m                         Traceback (most recent call last)",
      "\u001b[0;32m~/opt/anaconda3/lib/python3.9/site-packages/graphviz/backend/execute.py\u001b[0m in \u001b[0;36mrun_check\u001b[0;34m(cmd, input_lines, encoding, quiet, **kwargs)\u001b[0m\n\u001b[1;32m     78\u001b[0m                 \u001b[0mkwargs\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'stdout'\u001b[0m\u001b[0;34m]\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mkwargs\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'stderr'\u001b[0m\u001b[0;34m]\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0msubprocess\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mPIPE\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 79\u001b[0;31m             \u001b[0mproc\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0m_run_input_lines\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mcmd\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0minput_lines\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mkwargs\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mkwargs\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     80\u001b[0m         \u001b[0;32melse\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/opt/anaconda3/lib/python3.9/site-packages/graphviz/backend/execute.py\u001b[0m in \u001b[0;36m_run_input_lines\u001b[0;34m(cmd, input_lines, kwargs)\u001b[0m\n\u001b[1;32m     98\u001b[0m \u001b[0;32mdef\u001b[0m \u001b[0m_run_input_lines\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mcmd\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0minput_lines\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m*\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mkwargs\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 99\u001b[0;31m     \u001b[0mpopen\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0msubprocess\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mPopen\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mcmd\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mstdin\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0msubprocess\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mPIPE\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mkwargs\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    100\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/opt/anaconda3/lib/python3.9/subprocess.py\u001b[0m in \u001b[0;36m__init__\u001b[0;34m(self, args, bufsize, executable, stdin, stdout, stderr, preexec_fn, close_fds, shell, cwd, env, universal_newlines, startupinfo, creationflags, restore_signals, start_new_session, pass_fds, user, group, extra_groups, encoding, errors, text, umask)\u001b[0m\n\u001b[1;32m    950\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 951\u001b[0;31m             self._execute_child(args, executable, preexec_fn, close_fds,\n\u001b[0m\u001b[1;32m    952\u001b[0m                                 \u001b[0mpass_fds\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mcwd\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0menv\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/opt/anaconda3/lib/python3.9/subprocess.py\u001b[0m in \u001b[0;36m_execute_child\u001b[0;34m(self, args, executable, preexec_fn, close_fds, pass_fds, cwd, env, startupinfo, creationflags, shell, p2cread, p2cwrite, c2pread, c2pwrite, errread, errwrite, restore_signals, gid, gids, uid, umask, start_new_session)\u001b[0m\n\u001b[1;32m   1820\u001b[0m                         \u001b[0merr_msg\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mos\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mstrerror\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0merrno_num\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 1821\u001b[0;31m                     \u001b[0;32mraise\u001b[0m \u001b[0mchild_exception_type\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0merrno_num\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0merr_msg\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0merr_filename\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   1822\u001b[0m                 \u001b[0;32mraise\u001b[0m \u001b[0mchild_exception_type\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0merr_msg\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mFileNotFoundError\u001b[0m: [Errno 2] No such file or directory: PosixPath('dot')",
      "\nThe above exception was the direct cause of the following exception:\n",
      "\u001b[0;31mExecutableNotFound\u001b[0m                        Traceback (most recent call last)",
      "\u001b[0;32m~/opt/anaconda3/lib/python3.9/site-packages/IPython/core/formatters.py\u001b[0m in \u001b[0;36m__call__\u001b[0;34m(self, obj, include, exclude)\u001b[0m\n\u001b[1;32m    968\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    969\u001b[0m             \u001b[0;32mif\u001b[0m \u001b[0mmethod\u001b[0m \u001b[0;32mis\u001b[0m \u001b[0;32mnot\u001b[0m \u001b[0;32mNone\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 970\u001b[0;31m                 \u001b[0;32mreturn\u001b[0m \u001b[0mmethod\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0minclude\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0minclude\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mexclude\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mexclude\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    971\u001b[0m             \u001b[0;32mreturn\u001b[0m \u001b[0;32mNone\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    972\u001b[0m         \u001b[0;32melse\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/opt/anaconda3/lib/python3.9/site-packages/graphviz/jupyter_integration.py\u001b[0m in \u001b[0;36m_repr_mimebundle_\u001b[0;34m(self, include, exclude, **_)\u001b[0m\n\u001b[1;32m     96\u001b[0m         \u001b[0minclude\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mset\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0minclude\u001b[0m\u001b[0;34m)\u001b[0m \u001b[0;32mif\u001b[0m \u001b[0minclude\u001b[0m \u001b[0;32mis\u001b[0m \u001b[0;32mnot\u001b[0m \u001b[0;32mNone\u001b[0m \u001b[0;32melse\u001b[0m \u001b[0;34m{\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_jupyter_mimetype\u001b[0m\u001b[0;34m}\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     97\u001b[0m         \u001b[0minclude\u001b[0m \u001b[0;34m-=\u001b[0m \u001b[0mset\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mexclude\u001b[0m \u001b[0;32mor\u001b[0m \u001b[0;34m[\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 98\u001b[0;31m         return {mimetype: getattr(self, method_name)()\n\u001b[0m\u001b[1;32m     99\u001b[0m                 \u001b[0;32mfor\u001b[0m \u001b[0mmimetype\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mmethod_name\u001b[0m \u001b[0;32min\u001b[0m \u001b[0mMIME_TYPES\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mitems\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    100\u001b[0m                 if mimetype in include}\n",
      "\u001b[0;32m~/opt/anaconda3/lib/python3.9/site-packages/graphviz/jupyter_integration.py\u001b[0m in \u001b[0;36m<dictcomp>\u001b[0;34m(.0)\u001b[0m\n\u001b[1;32m     96\u001b[0m         \u001b[0minclude\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mset\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0minclude\u001b[0m\u001b[0;34m)\u001b[0m \u001b[0;32mif\u001b[0m \u001b[0minclude\u001b[0m \u001b[0;32mis\u001b[0m \u001b[0;32mnot\u001b[0m \u001b[0;32mNone\u001b[0m \u001b[0;32melse\u001b[0m \u001b[0;34m{\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_jupyter_mimetype\u001b[0m\u001b[0;34m}\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     97\u001b[0m         \u001b[0minclude\u001b[0m \u001b[0;34m-=\u001b[0m \u001b[0mset\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mexclude\u001b[0m \u001b[0;32mor\u001b[0m \u001b[0;34m[\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 98\u001b[0;31m         return {mimetype: getattr(self, method_name)()\n\u001b[0m\u001b[1;32m     99\u001b[0m                 \u001b[0;32mfor\u001b[0m \u001b[0mmimetype\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mmethod_name\u001b[0m \u001b[0;32min\u001b[0m \u001b[0mMIME_TYPES\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mitems\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    100\u001b[0m                 if mimetype in include}\n",
      "\u001b[0;32m~/opt/anaconda3/lib/python3.9/site-packages/graphviz/jupyter_integration.py\u001b[0m in \u001b[0;36m_repr_image_svg_xml\u001b[0;34m(self)\u001b[0m\n\u001b[1;32m    110\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0m_repr_image_svg_xml\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m)\u001b[0m \u001b[0;34m->\u001b[0m \u001b[0mstr\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    111\u001b[0m         \u001b[0;34m\"\"\"Return the rendered graph as SVG string.\"\"\"\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 112\u001b[0;31m         \u001b[0;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mpipe\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mformat\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;34m'svg'\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mencoding\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mSVG_ENCODING\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m",
      "\u001b[0;32m~/opt/anaconda3/lib/python3.9/site-packages/graphviz/piping.py\u001b[0m in \u001b[0;36mpipe\u001b[0;34m(self, format, renderer, formatter, neato_no_op, quiet, engine, encoding)\u001b[0m\n\u001b[1;32m    102\u001b[0m             \u001b[0;34m'<?xml version='\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    103\u001b[0m         \"\"\"\n\u001b[0;32m--> 104\u001b[0;31m         return self._pipe_legacy(format,\n\u001b[0m\u001b[1;32m    105\u001b[0m                                  \u001b[0mrenderer\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mrenderer\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    106\u001b[0m                                  \u001b[0mformatter\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mformatter\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/opt/anaconda3/lib/python3.9/site-packages/graphviz/_tools.py\u001b[0m in \u001b[0;36mwrapper\u001b[0;34m(*args, **kwargs)\u001b[0m\n\u001b[1;32m    169\u001b[0m                               category=category)\n\u001b[1;32m    170\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 171\u001b[0;31m             \u001b[0;32mreturn\u001b[0m \u001b[0mfunc\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m*\u001b[0m\u001b[0margs\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mkwargs\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    172\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    173\u001b[0m         \u001b[0;32mreturn\u001b[0m \u001b[0mwrapper\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/opt/anaconda3/lib/python3.9/site-packages/graphviz/piping.py\u001b[0m in \u001b[0;36m_pipe_legacy\u001b[0;34m(self, format, renderer, formatter, neato_no_op, quiet, engine, encoding)\u001b[0m\n\u001b[1;32m    119\u001b[0m                      \u001b[0mengine\u001b[0m\u001b[0;34m:\u001b[0m \u001b[0mtyping\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mOptional\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0mstr\u001b[0m\u001b[0;34m]\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;32mNone\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    120\u001b[0m                      encoding: typing.Optional[str] = None) -> typing.Union[bytes, str]:\n\u001b[0;32m--> 121\u001b[0;31m         return self._pipe_future(format,\n\u001b[0m\u001b[1;32m    122\u001b[0m                                  \u001b[0mrenderer\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mrenderer\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    123\u001b[0m                                  \u001b[0mformatter\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mformatter\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/opt/anaconda3/lib/python3.9/site-packages/graphviz/piping.py\u001b[0m in \u001b[0;36m_pipe_future\u001b[0;34m(self, format, renderer, formatter, neato_no_op, quiet, engine, encoding)\u001b[0m\n\u001b[1;32m    147\u001b[0m             \u001b[0;32mif\u001b[0m \u001b[0mcodecs\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mlookup\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mencoding\u001b[0m\u001b[0;34m)\u001b[0m \u001b[0;32mis\u001b[0m \u001b[0mcodecs\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mlookup\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mencoding\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    148\u001b[0m                 \u001b[0;31m# common case: both stdin and stdout need the same encoding\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 149\u001b[0;31m                 \u001b[0;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_pipe_lines_string\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m*\u001b[0m\u001b[0margs\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mencoding\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mencoding\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mkwargs\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    150\u001b[0m             \u001b[0;32mtry\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    151\u001b[0m                 \u001b[0mraw\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_pipe_lines\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m*\u001b[0m\u001b[0margs\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0minput_encoding\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mencoding\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mkwargs\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/opt/anaconda3/lib/python3.9/site-packages/graphviz/backend/piping.py\u001b[0m in \u001b[0;36mpipe_lines_string\u001b[0;34m(engine, format, input_lines, encoding, renderer, formatter, neato_no_op, quiet)\u001b[0m\n\u001b[1;32m    210\u001b[0m     \u001b[0mkwargs\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;34m{\u001b[0m\u001b[0;34m'input_lines'\u001b[0m\u001b[0;34m:\u001b[0m \u001b[0minput_lines\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m'encoding'\u001b[0m\u001b[0;34m:\u001b[0m \u001b[0mencoding\u001b[0m\u001b[0;34m}\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    211\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 212\u001b[0;31m     \u001b[0mproc\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mexecute\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mrun_check\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mcmd\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mcapture_output\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;32mTrue\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mquiet\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mquiet\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mkwargs\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    213\u001b[0m     \u001b[0;32mreturn\u001b[0m \u001b[0mproc\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mstdout\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/opt/anaconda3/lib/python3.9/site-packages/graphviz/backend/execute.py\u001b[0m in \u001b[0;36mrun_check\u001b[0;34m(cmd, input_lines, encoding, quiet, **kwargs)\u001b[0m\n\u001b[1;32m     82\u001b[0m     \u001b[0;32mexcept\u001b[0m \u001b[0mOSError\u001b[0m \u001b[0;32mas\u001b[0m \u001b[0me\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     83\u001b[0m         \u001b[0;32mif\u001b[0m \u001b[0me\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0merrno\u001b[0m \u001b[0;34m==\u001b[0m \u001b[0merrno\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mENOENT\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 84\u001b[0;31m             \u001b[0;32mraise\u001b[0m \u001b[0mExecutableNotFound\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mcmd\u001b[0m\u001b[0;34m)\u001b[0m \u001b[0;32mfrom\u001b[0m \u001b[0me\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     85\u001b[0m         \u001b[0;32mraise\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     86\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mExecutableNotFound\u001b[0m: failed to execute PosixPath('dot'), make sure the Graphviz executables are on your systems' PATH"
     ]
    },
    {
     "data": {
      "text/plain": [
       "<graphviz.sources.Source at 0x7f94015a0a90>"
      ]
     },
     "execution_count": 64,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import graphviz \n",
    "from sklearn.tree import export_graphviz\n",
    "\n",
    "viz = export_graphviz(dt, \n",
    "                      out_file=None, # ¬øSalvar en el computador?\n",
    "                    max_depth=5, # M√°xima profundidad\n",
    "                      feature_names=list(dict(sorted(tfidf_vect.vocabulary_.items())).keys()), # Palabras\n",
    "                      class_names = ['basura', 'leg√≠timo'],\n",
    "                      rounded=True, filled=True\n",
    "                     )\n",
    "\n",
    "graph = graphviz.Source(viz)\n",
    "graph"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 57,
   "id": "e3e4f0c3",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>cancer_type</th>\n",
       "      <th>note</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Thyroid_Cancer</td>\n",
       "      <td>Incidence Differences Between First PrimaryCan...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Thyroid_Cancer</td>\n",
       "      <td>Deoxyshikonin Inhibits Viability andGlycolysis...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Thyroid_Cancer</td>\n",
       "      <td>Aprospective blinded multicenter studyDear edi...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Lung_Cancer</td>\n",
       "      <td>\"Thirty micrograms of WCL were resolved by SDS...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Lung_Cancer</td>\n",
       "      <td>\"These studies demonstrate that RRM1 could be ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>995</th>\n",
       "      <td>Thyroid_Cancer</td>\n",
       "      <td>\"Type  and  diabetes confer an increased risk ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>996</th>\n",
       "      <td>Lung_Cancer</td>\n",
       "      <td>\"Complete surgical excision was impossible due...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>997</th>\n",
       "      <td>Lung_Cancer</td>\n",
       "      <td>\"atment groups.For each subgroupthe sum of the...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>998</th>\n",
       "      <td>Lung_Cancer</td>\n",
       "      <td>\"Lung cancer is the most common malignancy and...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>999</th>\n",
       "      <td>Lung_Cancer</td>\n",
       "      <td>\"inhibitors by causing immediate and nearly co...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1000 rows √ó 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        cancer_type                                               note\n",
       "0    Thyroid_Cancer  Incidence Differences Between First PrimaryCan...\n",
       "1    Thyroid_Cancer  Deoxyshikonin Inhibits Viability andGlycolysis...\n",
       "2    Thyroid_Cancer  Aprospective blinded multicenter studyDear edi...\n",
       "3       Lung_Cancer  \"Thirty micrograms of WCL were resolved by SDS...\n",
       "4       Lung_Cancer  \"These studies demonstrate that RRM1 could be ...\n",
       "..              ...                                                ...\n",
       "995  Thyroid_Cancer  \"Type  and  diabetes confer an increased risk ...\n",
       "996     Lung_Cancer  \"Complete surgical excision was impossible due...\n",
       "997     Lung_Cancer  \"atment groups.For each subgroupthe sum of the...\n",
       "998     Lung_Cancer  \"Lung cancer is the most common malignancy and...\n",
       "999     Lung_Cancer  \"inhibitors by causing immediate and nearly co...\n",
       "\n",
       "[1000 rows x 2 columns]"
      ]
     },
     "execution_count": 57,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 58,
   "id": "fd0e1d71",
   "metadata": {},
   "outputs": [],
   "source": [
    "df.to_csv(\"cancer.csv\", index=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "c6a63421",
   "metadata": {
    "slideshow": {
     "slide_type": "slide"
    }
   },
   "source": [
    "# ‚è™ Today's recap\n",
    "\n",
    "- Classification models\n",
    "    - Logistic regression \n",
    "    - Naive Bayes\n",
    "    - Support Vector Machines (SVM)\n",
    "    - Decision Trees"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "b5c8ffe1",
   "metadata": {
    "slideshow": {
     "slide_type": "slide"
    }
   },
   "source": [
    "<center><img src='img/bye.gif' style='height:250px;'></center> \n",
    "\n",
    "# Next class: Classification models, part 2: coding!\n",
    "# See you next Friday!"
   ]
  }
 ],
 "metadata": {
  "celltoolbar": "Slideshow",
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.7"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
